{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import absolute_import\n",
    "import utils\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from pandas.api.types import is_categorical_dtype , is_numeric_dtype\n",
    "import plotly.graph_objects as go\n",
    "from utils import _get_hover_text\n",
    "import pickle\n",
    "import dash\n",
    "import dash_core_components as dcc\n",
    "import dash_html_components as html\n",
    "from jupyter_dash import JupyterDash\n",
    "import os\n",
    "import sys\n",
    "from bioservices.kegg import KEGG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Main functions that draw Manhattan plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "SUGGESTIVE_LINE_LABEL = \"suggestive line\"\n",
    "GENOMEWIDE_LINE_LABEL = \"genomewide line\"\n",
    "\n",
    "\n",
    "def ManhattanPlot(\n",
    "        dataframe,\n",
    "        chrm=\"CHR\",\n",
    "        bp=\"BP\",\n",
    "        p=\"P\",\n",
    "        snp=\"SNP\",\n",
    "        gene=\"GENE\",\n",
    "        annotation=None,\n",
    "        logp=True,\n",
    "        title=\"Manhattan Plot\",\n",
    "        showgrid=True,\n",
    "        xlabel=None,\n",
    "        ylabel='-log10(p)',\n",
    "        point_size=5,\n",
    "        showlegend=True,\n",
    "        col=None,\n",
    "        suggestiveline_value= -np.log10(1e-8), #160\n",
    "        suggestiveline_color='#636efa',\n",
    "        suggestiveline_width=1,\n",
    "        genomewideline_value=  -np.log10(5e-8), #100, #\n",
    "        genomewideline_color='#EF553B',\n",
    "        genomewideline_width=1,\n",
    "        highlight=True,\n",
    "        highlight_color=\"red\",\n",
    "):\n",
    "    \"\"\"Returns a figure for a manhattan plot.\n",
    "Keyword arguments:\n",
    "- dataframe (dataframe; required): A pandas dataframe which must contain at\n",
    "    least the following three columns:\n",
    "            - the chromosome number\n",
    "            - genomic base-pair position\n",
    "            - a numeric quantity to plot such as a p-value or zscore\n",
    "- chrm (string; default 'CHR'): A string denoting the column name for\n",
    "    the chromosome. This column must be float or integer. Minimum\n",
    "    number of chromosomes required is 1. If you have X, Y, or MT\n",
    "    chromosomes, be sure to renumber these 23, 24, 25, etc.\n",
    "- bp (string; default 'BP'): A string denoting the column name for the\n",
    "    chromosomal position.\n",
    "- p (string; default 'P'): A string denoting the column name for the\n",
    "    float quantity to be plotted on the y-axis. This column must be\n",
    "    numeric. It does not have to be a p-value. It can be any numeric\n",
    "    quantity such as peak heights, Bayes factors, test statistics. If\n",
    "    it is not a p-value, make sure to set logp = False.\n",
    "- snp (string; default 'SNP'): A string denoting the column name for\n",
    "    the SNP names (e.g., rs number). More generally, this column could\n",
    "    be anything that identifies each point being plotted. For example,\n",
    "    in an Epigenomewide association study (EWAS), this could be the\n",
    "    probe name or cg number. This column should be a character. This\n",
    "    argument is optional, however it is necessary to specify it if you\n",
    "    want to highlight points on the plot, using the highlight argument\n",
    "    in the figure method.\n",
    "- gene (string; default 'GENE'): A string denoting the column name for\n",
    "    the GENE names. This column could be a string or a float. More\n",
    "    generally, it could be any annotation information that you want\n",
    "    to include in the plot.\n",
    "- annotation (string; optional): A string denoting the column to use\n",
    "    as annotations. This column could be a string or a float. It\n",
    "    could be any annotation information that you want to include in\n",
    "    the plot (e.g., zscore, effect size, minor allele frequency).\n",
    "- logp (bool; optional): If True, the -log10 of the p-value is\n",
    "    plotted. It isn't very useful to plot raw p-values; however,\n",
    "    plotting the raw value could be useful for other genome-wide plots\n",
    "    (e.g., peak heights, Bayes factors, test statistics, other\n",
    "    \"scores\", etc.)\n",
    "- title (string; default 'Manhattan Plot'): The title of the graph.\n",
    "- showgrid (bool; default true): Boolean indicating whether gridlines\n",
    "    should be shown.\n",
    "- xlabel (string; optional): Label of the x axis.\n",
    "- ylabel (string; default '-log10(p)'): Label of the y axis.\n",
    "- point_size (number; default 5): Size of the points of the Scatter\n",
    "    plot.\n",
    "- showlegend (bool; default true): Boolean indicating whether legends\n",
    "    should be shown.\n",
    "- col (string; optional): A string representing the color of the\n",
    "    points of the scatter plot. Can be in any color format accepted by\n",
    "    plotly.graph_objects.\n",
    "- suggestiveline_value (bool | float; default 8): A value which must\n",
    "    be either False to deactivate the option, or a numerical value\n",
    "    corresponding to the p-value at which the line should be drawn.\n",
    "    The line has no influence on the data points.\n",
    "- suggestiveline_color (string; default 'grey'): Color of the suggestive\n",
    "  line.\n",
    "- suggestiveline_width (number; default 2): Width of the suggestive\n",
    "    line.\n",
    "- genomewideline_value (bool | float; default -log10(5e-8)): A boolean\n",
    "    which must be either False to deactivate the option, or a numerical value\n",
    "    corresponding to the p-value above which the data points are\n",
    "    considered significant.\n",
    "- genomewideline_color (string; default 'red'): Color of the genome-wide\n",
    "    line. Can be in any color format accepted by plotly.graph_objects.\n",
    "- genomewideline_width (number; default 1): Width of the genome-wide\n",
    "  line.\n",
    "- highlight (bool; default True): turning on/off the highlighting of\n",
    "    data points considered significant.\n",
    "- highlight_color (string; default 'red'): Color of the data points\n",
    "    highlighted because they are significant. Can be in any color\n",
    "    format accepted by plotly.graph_objects.\n",
    "    # ...\n",
    "    Example 1: Random Manhattan Plot\n",
    "    '''\n",
    "    dataframe = pd.DataFrame(\n",
    "        np.random.randint(0,100,size=(100, 3)),\n",
    "        columns=['P', 'CHR', 'BP'])\n",
    "    fig = create_manhattan(dataframe, title='XYZ Manhattan plot')\n",
    "    plotly.offline.plot(fig, image='png')\n",
    "    '''\n",
    "    \"\"\"\n",
    "\n",
    "    mh = _ManhattanPlot(\n",
    "        dataframe,\n",
    "        chrm=chrm,\n",
    "        bp=bp,\n",
    "        p=p,\n",
    "        snp=snp,\n",
    "        gene=gene,\n",
    "        annotation=annotation,\n",
    "        logp=logp\n",
    "    )\n",
    "\n",
    "    return mh.figure(\n",
    "        title=title,\n",
    "        showgrid=showgrid,\n",
    "        xlabel=xlabel,\n",
    "        ylabel=ylabel,\n",
    "        point_size=point_size,\n",
    "        showlegend=showlegend,\n",
    "        col=col,\n",
    "        suggestiveline_value=suggestiveline_value,\n",
    "        suggestiveline_color=suggestiveline_color,\n",
    "        suggestiveline_width=suggestiveline_width,\n",
    "        genomewideline_value=genomewideline_value,\n",
    "        genomewideline_color=genomewideline_color,\n",
    "        genomewideline_width=genomewideline_width,\n",
    "        highlight=highlight,\n",
    "        highlight_color=highlight_color\n",
    "    )\n",
    "\n",
    "\n",
    "class _ManhattanPlot():\n",
    "\n",
    "    def __init__(\n",
    "            self,\n",
    "            x,\n",
    "            chrm=\"CHR\",\n",
    "            bp=\"BP\",\n",
    "            p=\"P\",\n",
    "            snp=\"SNP\",\n",
    "            gene=\"GENE\",\n",
    "            annotation=None,\n",
    "            logp=True\n",
    "    ):\n",
    "        \"\"\"\n",
    "        Keyword arguments:\n",
    "        - dataframe (dataframe; required): A pandas dataframe which\n",
    "        must contain at least the following three columns:\n",
    "            - the chromosome number\n",
    "            - genomic base-pair position\n",
    "            - a numeric quantity to plot such as a p-value or zscore\n",
    "        - chrm (string; default 'CHR'): A string denoting the column name for the\n",
    "        chromosome.  This column must be float or integer.  Minimum number\n",
    "        of chromosomes required is 1. If you have X, Y, or MT chromosomes,\n",
    "        be sure to renumber these 23, 24, 25, etc.\n",
    "        - bp (string; default 'BP'): A string denoting the column name for the\n",
    "        chromosomal position.\n",
    "        - p (string; default 'P'): A string denoting the column name for the\n",
    "        float quantity to be plotted on the y-axis. This column must be\n",
    "        numeric. This does not have to be a p-value. It can be any\n",
    "        numeric quantity such as peak heights, bayes factors, test\n",
    "        statistics. If it is not a p-value, make sure to set logp = FALSE.\n",
    "        - snp (string; default 'SNP'): A string denoting the column name for the\n",
    "        SNP names (e.g. rs number). More generally, this column could be\n",
    "        anything that identifies each point being plotted. For example, in\n",
    "        an Epigenomewide association study (EWAS) this could be the probe\n",
    "        name or cg number. This column should be a character. This\n",
    "        argument is optional, however it is necessary to specify if you\n",
    "        want to highlight points on the plot using the highlight argument\n",
    "        in the figure method.\n",
    "        - gene (string; default 'GENE'): A string denoting the column name for the\n",
    "        GENE names. This column could be a string or a float. More\n",
    "        generally, it could be any annotation information that you want\n",
    "        to include in the plot.\n",
    "        - annotation (string; optional): A string denoting the column name for\n",
    "        an annotation. This column could be a string or a float.  This\n",
    "        could be any annotation information that you want to include in\n",
    "        the plot (e.g. zscore, effect size, minor allele frequency).\n",
    "        - logp (bool; default True): If True, the -log10 of the p-value is\n",
    "        plotted.  It isn't very useful to plot raw p-values; however,\n",
    "        plotting the raw value could be useful for other genome-wide plots\n",
    "        (e.g., peak heights, Bayes factors, test statistics, other\n",
    "        \"scores\", etc.).\n",
    "        Returns:\n",
    "        - A ManhattanPlot object.\"\"\"\n",
    "\n",
    "        # checking the validity of the arguments\n",
    "\n",
    "        # Make sure you have chrm, bp and p columns and that they are of\n",
    "        # numeric type\n",
    "        if chrm not in x.columns.values:\n",
    "            raise KeyError(\"Column %s not found in 'x' data.frame\" % chrm)\n",
    "        else:\n",
    "            if not is_numeric_dtype(x[chrm].dtype):\n",
    "                raise TypeError(\"%s column should be numeric. Do you have \"\n",
    "                                \"'X', 'Y', 'MT', etc? If so change to \"\n",
    "                                \"numbers and try again.\" % chrm)\n",
    "\n",
    "        if bp not in x.columns.values:\n",
    "            raise KeyError(\"Column %s not found in 'x' data.frame\" % bp)\n",
    "        else:\n",
    "            if not is_numeric_dtype(x[bp].dtype):\n",
    "                raise TypeError(\"%s column should be numeric type\" % bp)\n",
    "\n",
    "        if p not in x.columns.values:\n",
    "            raise KeyError(\"Column %s not found in 'x' data.frame\" % p)\n",
    "        else:\n",
    "            if not is_numeric_dtype(x[p].dtype):\n",
    "                raise TypeError(\"%s column should be numeric type\" % p)\n",
    "\n",
    "        # Create a new DataFrame with columns named after chrm, bp, and p.\n",
    "        self.data = pd.DataFrame(data=x[[chrm, bp, p]])\n",
    "\n",
    "        if snp is not None:\n",
    "            if snp not in x.columns.values:\n",
    "                # Warn if you don't have a snp column\n",
    "                raise KeyError(\n",
    "                    \"snp argument specified as %s but column not found in \"\n",
    "                    \"'x' data.frame\" % snp)\n",
    "            else:\n",
    "                # If the input DataFrame has a snp column, add it to the new\n",
    "                # DataFrame\n",
    "                self.data[snp] = x[snp]\n",
    "\n",
    "        if gene is not None:\n",
    "            if gene not in x.columns.values:\n",
    "                # Warn if you don't have a gene column\n",
    "                raise KeyError(\n",
    "                    \"gene argument specified as %s but column not found in \"\n",
    "                    \"'x' data.frame\" % gene)\n",
    "            else:\n",
    "                # If the input DataFrame has a gene column, add it to the new\n",
    "                # DataFrame\n",
    "                self.data[gene] = x[gene]\n",
    "\n",
    "        if annotation is not None:\n",
    "            if annotation not in x.columns.values:\n",
    "                # Warn if you don't have an annotation column\n",
    "                raise KeyError(\n",
    "                    \"annotation argument specified as %s but column not \"\n",
    "                    \"found in 'x' data.frame\" % annotation\n",
    "                )\n",
    "            else:\n",
    "                # If the input DataFrame has a gene column, add it to the new\n",
    "                # DataFrame\n",
    "                self.data[annotation] = x[annotation]\n",
    "\n",
    "        self.xlabel = \"\"\n",
    "        self.ticks = []\n",
    "        self.ticksLabels = []\n",
    "        self.nChr = len(x[chrm].unique())\n",
    "        self.chrName = chrm\n",
    "        self.pName = p\n",
    "        self.snpName = snp\n",
    "        self.geneName = gene\n",
    "        self.annotationName = annotation\n",
    "        self.logp = logp\n",
    "\n",
    "        # Set positions, ticks, and labels for plotting\n",
    "\n",
    "        self.index = 'INDEX'\n",
    "        self.pos = 'POSITION'\n",
    "\n",
    "        # Fixes the bug where one chromosome is missing by adding a sequential\n",
    "        # index column.\n",
    "        idx = 0\n",
    "        for i in self.data[chrm].unique():\n",
    "            idx = idx + 1\n",
    "            self.data.loc[self.data[chrm] == i, self.index] = int(idx)\n",
    "        # Set the type to be the same as provided for chrm column\n",
    "        self.data[self.index] = \\\n",
    "            self.data[self.index].astype(self.data[chrm].dtype)\n",
    "\n",
    "        # This section sets up positions and ticks. Ticks should be placed in\n",
    "        # the middle of a chromosome. The new pos column is added that keeps\n",
    "        # a running sum of the positions of each successive chromosome.\n",
    "        # For example:\n",
    "        # chrm bp pos\n",
    "        # 1   1  1\n",
    "        # 1   2  2\n",
    "        # 2   1  3\n",
    "        # 2   2  4\n",
    "        # 3   1  5\n",
    "\n",
    "        if self.nChr == 1:\n",
    "            # For a single chromosome\n",
    "            self.data[self.pos] = self.data[bp]\n",
    "            self.ticks.append(int(len(self.data[self.pos]) / 2.) + 1)\n",
    "            self.xlabel = \"Chromosome %s position\" % (self.data[chrm].unique())\n",
    "            self.ticksLabels = self.ticks\n",
    "        else:\n",
    "            # For multiple chromosomes\n",
    "            lastbase = 0\n",
    "            for i in self.data[self.index].unique():\n",
    "                if i == 1:\n",
    "                    self.data.loc[self.data[self.index] == i, self.pos] = \\\n",
    "                        self.data.loc[self.data[self.index] == i, bp].values\n",
    "                else:\n",
    "                    prevbp = self.data.loc[self.data[self.index] == i - 1, bp]\n",
    "                    # Shift the basepair position by the largest bp of the\n",
    "                    # current chromosome\n",
    "                    lastbase = lastbase + prevbp.iat[-1]\n",
    "\n",
    "                    self.data.loc[self.data[self.index] == i, self.pos] = \\\n",
    "                        self.data.loc[self.data[self.index] == i, bp].values \\\n",
    "                        + lastbase\n",
    "\n",
    "                tmin = min(self.data.loc[self.data[self.index] == i, self.pos])\n",
    "                tmax = max(self.data.loc[self.data[self.index] == i, self.pos])\n",
    "                self.ticks.append(int((tmin + tmax) / 2.) + 1)\n",
    "\n",
    "            self.xlabel = 'Chromosome'\n",
    "            self.data[self.pos] = self.data[self.pos].astype(\n",
    "                self.data[bp].dtype)\n",
    "\n",
    "#             if self.nChr > 10:  # To avoid crowded labels\n",
    "#                 self.ticksLabels = [\n",
    "#                     t if np.mod(int(t), 2)  # Only every two ticks\n",
    "#                     else ''\n",
    "#                     for t in self.data[chrm].unique()\n",
    "#                 ]\n",
    "#             else:\n",
    "            self.ticksLabels = self.data[chrm].unique()  # All the ticks\n",
    "\n",
    "    def figure(\n",
    "            self,\n",
    "            title=\"Manhattan Plot\",\n",
    "            showgrid=True,\n",
    "            xlabel=None,\n",
    "            ylabel='-log10(p)',\n",
    "            point_size=3,\n",
    "            showlegend=True,\n",
    "            col=None,\n",
    "            suggestiveline_value= -np.log10(1e-8), #160, \n",
    "            suggestiveline_color='blue',\n",
    "            suggestiveline_width=1,\n",
    "            genomewideline_value= -np.log10(1e-6), #100,#\n",
    "            genomewideline_color='red',\n",
    "            genomewideline_width=1,\n",
    "            highlight=True,\n",
    "            highlight_color=\"red\",\n",
    "    ):\n",
    "        \"\"\"Keyword arguments:\n",
    "    - title (string; default 'Manhattan Plot'): The title of the\n",
    "        graph.\n",
    "    - showgrid (bool; default True): Boolean indicating whether\n",
    "        gridlines should be shown.\n",
    "    - xlabel (string; optional): Label of the x axis.\n",
    "    - ylabel (string; default '-log10(p)'): Label of the y axis.\n",
    "    - point_size (number; default 5): Size of the points of the\n",
    "        scatter plot.\n",
    "    - showlegend (bool; default True): Boolean indicating whether\n",
    "        legends should be shown.\n",
    "    - col (string; optional): A string representing the color of the\n",
    "        points of the Scatter plot. Can be in any color format\n",
    "        accepted by plotly.graph_objects.\n",
    "    - suggestiveline_value (bool | float; default 8): A value which\n",
    "        must be either False to deactivate the option, or a numerical value\n",
    "        corresponding to the p-value at which the line should be\n",
    "        drawn. The line has no influence on the data points.\n",
    "    - suggestiveline_color (string; default 'grey'): Color of the\n",
    "        suggestive line.\n",
    "    - suggestiveline_width (number; default 2): Width of the\n",
    "        suggestive line.\n",
    "    - genomewideline_value (bool | float; default -log10(5e-8)): A\n",
    "        boolean which must be either False to deactivate the option, or a\n",
    "        numerical value corresponding to the p-value above which the\n",
    "        data points are considered significant.\n",
    "    - genomewideline_color (string; default 'red'): Color of the\n",
    "        genome-wide line. Can be in any color format accepted by\n",
    "        plotly.graph_objects.\n",
    "    - genomewideline_width (number; default 1): Width of the genome\n",
    "      wide line.\n",
    "    - highlight (bool; default True): Whether to turn on or off the\n",
    "        highlighting of data points considered significant.\n",
    "    - highlight_color (string; default 'red'): Color of the data\n",
    "        points highlighted because they are significant. Can be in any\n",
    "        color format accepted by plotly.graph_objects.\n",
    "    Returns:\n",
    "    - A figure formatted for plotly.graph_objects.\n",
    "        \"\"\"\n",
    "\n",
    "        xmin = min(self.data[self.pos].values) # min pos value along x axis\n",
    "        xmax = max(self.data[self.pos].values) # Max Pos value along x axis\n",
    "\n",
    "        horizontallines = []\n",
    "\n",
    "        if suggestiveline_value:\n",
    "            suggestiveline = go.layout.Shape(\n",
    "                name=SUGGESTIVE_LINE_LABEL,\n",
    "                type=\"line\",\n",
    "                fillcolor=suggestiveline_color,\n",
    "                line=dict(\n",
    "                    color=suggestiveline_color,\n",
    "                    width=suggestiveline_width\n",
    "                ),\n",
    "                x0=xmin, x1=xmax, xref=\"x\",\n",
    "                y0=suggestiveline_value, y1=suggestiveline_value, yref=\"y\"\n",
    "            )\n",
    "            horizontallines.append(suggestiveline)\n",
    "\n",
    "        if genomewideline_value:\n",
    "            genomewideline = go.layout.Shape(\n",
    "                name=GENOMEWIDE_LINE_LABEL,\n",
    "                type=\"line\",\n",
    "                fillcolor=genomewideline_color,\n",
    "                line=dict(\n",
    "                    color=genomewideline_color,\n",
    "                    width=genomewideline_width\n",
    "                ),\n",
    "                x0=xmin, x1=xmax, xref=\"x\",\n",
    "                y0=genomewideline_value, y1=genomewideline_value, yref=\"y\"\n",
    "            )\n",
    "            horizontallines.append(genomewideline)\n",
    "       \n",
    "        data_to_plot = []  # List to contain the data traces\n",
    "        tmp = pd.DataFrame()  # Empty DataFrame to contain the highlighted data\n",
    "        pathways_tmp_empty = []# hold unique pathways from the temp dataFrame \n",
    "\n",
    "        if highlight:\n",
    "            if not isinstance(highlight, bool):\n",
    "                if self.snpName not in self.data.columns.values:\n",
    "                    raise KeyError(\n",
    "                        \"snp argument specified for highlight as %s but \"\n",
    "                        \"column not found in the data.frame\" % self.snpName\n",
    "                    )\n",
    "            else:\n",
    "                if not genomewideline_value:\n",
    "                    raise Warning(\n",
    "                        \"The genomewideline_value you entered is not a \"\n",
    "                        \"positive value, or False, you cannot set highlight \"\n",
    "                        \"to True in that case.\")\n",
    "                tmp = self.data\n",
    "\n",
    "                # Sort the p-values (or -log10(p-values) above the line\n",
    "                if genomewideline_value:\n",
    "                    if self.logp:\n",
    "                        tmp = tmp.loc[-np.log10(tmp[self.pName])\n",
    "                                      > genomewideline_value]\n",
    "                    else:\n",
    "                        tmp = tmp.loc[tmp[self.pName] > genomewideline_value]\n",
    "\n",
    "                highlight_hover_text = _get_hover_text(\n",
    "                    tmp,\n",
    "                    snpname=self.snpName,\n",
    "                    genename=self.geneName,\n",
    "                    annotationname=self.annotationName\n",
    "                )\n",
    "\n",
    "                if not tmp.empty:\n",
    "                    #### Containing the significnat ploting points \n",
    "                    data_to_plot.append(\n",
    "                        go.Scattergl(\n",
    "                            x=tmp[self.pos].values,\n",
    "                            y=-np.log10(tmp[self.pName].values) if self.logp\n",
    "                            else tmp[self.pName].values,\n",
    "                            mode= \"markers\",\n",
    "                            text=highlight_hover_text,\n",
    "                            marker=dict(\n",
    "                                color=highlight_color,\n",
    "                                size=point_size\n",
    "                            ),\n",
    "                            name=\"Point(s) of interest\"\n",
    "                        )\n",
    "                    )\n",
    "\n",
    "        # Remove the highlighted data from the DataFrame if not empty\n",
    "        if tmp.empty:\n",
    "            data = self.data\n",
    "        else:\n",
    "            data = self.data.drop(self.data.index[tmp.index])\n",
    "\n",
    "        if self.nChr == 1:\n",
    "\n",
    "            if col is None:\n",
    "                col = ['black']\n",
    "\n",
    "            # If single chromosome, ticks and labels automatic.\n",
    "            layout = go.Layout(\n",
    "                title=title,\n",
    "                xaxis={\n",
    "                    'title': self.xlabel if xlabel is None else xlabel,\n",
    "                    'showgrid': showgrid,\n",
    "                    'range': [xmin, xmax],\n",
    "                },\n",
    "                yaxis={'title': ylabel},\n",
    "                hovermode='closest'\n",
    "            )\n",
    "\n",
    "            hover_text = _get_hover_text(\n",
    "                data,\n",
    "                snpname=self.snpName,\n",
    "                genename=self.geneName,\n",
    "                annotationname=self.annotationName\n",
    "            )\n",
    "\n",
    "            data_to_plot.append(\n",
    "                go.Scattergl(\n",
    "                    x=data[self.pos].values,\n",
    "                    y=-np.log10(data[self.pName].values) if self.logp\n",
    "                    else data[self.pName].values,\n",
    "                    mode=\"markers\",\n",
    "                    showlegend=showlegend,\n",
    "                    marker={\n",
    "                        'color': col[0],\n",
    "                        'size': point_size,\n",
    "                        'name': \"chr%i\" % data[self.chrName].unique()\n",
    "                    },\n",
    "                    text=hover_text\n",
    "                )\n",
    "            )\n",
    "        else:\n",
    "            # if multiple chrms, use the ticks and labels you created above.\n",
    "            layout = go.Layout(\n",
    "                title=title,\n",
    "                xaxis={\n",
    "                    'title': self.xlabel if xlabel is None else xlabel,\n",
    "                    'showgrid': showgrid,\n",
    "                    'range': [xmin, xmax],\n",
    "                    'tickmode': \"array\",\n",
    "                    'tickvals': self.ticks,\n",
    "                    'ticktext': self.ticksLabels,\n",
    "                    'ticks': \"outside\"\n",
    "                },\n",
    "                yaxis={'title': ylabel},\n",
    "                hovermode='closest'\n",
    "            )\n",
    "            # Color variable\n",
    "            icol = 0\n",
    "            if col is None:\n",
    "                col = [\n",
    "                    'black' if np.mod(i, 2)\n",
    "                    else 'grey' for i in range(self.nChr)\n",
    "                ]\n",
    "\n",
    "            for i in data[self.index].unique():\n",
    "\n",
    "                tmp = data[data[self.index] == i]\n",
    "\n",
    "                chromo = tmp[self.chrName].unique()  # Get chromosome name\n",
    "\n",
    "                hover_text = _get_hover_text(\n",
    "                    data,\n",
    "                    snpname=self.snpName,\n",
    "                    genename=self.geneName,\n",
    "                    annotationname=self.annotationName\n",
    "                )\n",
    "\n",
    "                data_to_plot.append(\n",
    "                    go.Scattergl(\n",
    "                        x=tmp[self.pos].values,\n",
    "                        y=-np.log10(tmp[self.pName].values) if self.logp\n",
    "                        else tmp[self.pName].values,\n",
    "                        mode=\"markers\",\n",
    "                        showlegend=showlegend,\n",
    "                        name=\"Chr%i\" % chromo,\n",
    "                        marker={\n",
    "                            'color': col[icol],\n",
    "                            'size': point_size\n",
    "                        },\n",
    "                        text=hover_text\n",
    "                    )\n",
    "                )\n",
    "\n",
    "                icol = icol + 1\n",
    "\n",
    "        layout.shapes = horizontallines\n",
    "\n",
    "        return go.Figure(data=data_to_plot, layout=layout)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Testing datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv( 'https://raw.githubusercontent.com/plotly/dash-bio-docs-files/master/' +\n",
    "                 'manhattan_data.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>P</th>\n",
       "      <th>ZSCORE</th>\n",
       "      <th>EFFECTSIZE</th>\n",
       "      <th>DISTANCE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>14412.000000</td>\n",
       "      <td>1.441200e+04</td>\n",
       "      <td>1.441200e+04</td>\n",
       "      <td>14412.000000</td>\n",
       "      <td>14412.000000</td>\n",
       "      <td>1.441200e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>9.248820</td>\n",
       "      <td>7.827018e+07</td>\n",
       "      <td>4.959331e-01</td>\n",
       "      <td>0.813815</td>\n",
       "      <td>0.000756</td>\n",
       "      <td>8.997008e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>6.285682</td>\n",
       "      <td>5.643974e+07</td>\n",
       "      <td>2.882760e-01</td>\n",
       "      <td>0.639327</td>\n",
       "      <td>0.110560</td>\n",
       "      <td>1.969981e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>9.222000e+04</td>\n",
       "      <td>6.750100e-10</td>\n",
       "      <td>0.000100</td>\n",
       "      <td>-1.787300</td>\n",
       "      <td>-1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>4.000000</td>\n",
       "      <td>3.159827e+07</td>\n",
       "      <td>2.483034e-01</td>\n",
       "      <td>0.327250</td>\n",
       "      <td>-0.066925</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>8.000000</td>\n",
       "      <td>6.837940e+07</td>\n",
       "      <td>4.944957e-01</td>\n",
       "      <td>0.683200</td>\n",
       "      <td>0.000200</td>\n",
       "      <td>2.783500e+03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>14.000000</td>\n",
       "      <td>1.154268e+08</td>\n",
       "      <td>7.434780e-01</td>\n",
       "      <td>1.154475</td>\n",
       "      <td>0.067500</td>\n",
       "      <td>8.004100e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>23.000000</td>\n",
       "      <td>2.465864e+08</td>\n",
       "      <td>9.999448e-01</td>\n",
       "      <td>6.171800</td>\n",
       "      <td>2.654900</td>\n",
       "      <td>1.968847e+06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                CHR            BP             P        ZSCORE    EFFECTSIZE  \\\n",
       "count  14412.000000  1.441200e+04  1.441200e+04  14412.000000  14412.000000   \n",
       "mean       9.248820  7.827018e+07  4.959331e-01      0.813815      0.000756   \n",
       "std        6.285682  5.643974e+07  2.882760e-01      0.639327      0.110560   \n",
       "min        1.000000  9.222000e+04  6.750100e-10      0.000100     -1.787300   \n",
       "25%        4.000000  3.159827e+07  2.483034e-01      0.327250     -0.066925   \n",
       "50%        8.000000  6.837940e+07  4.944957e-01      0.683200      0.000200   \n",
       "75%       14.000000  1.154268e+08  7.434780e-01      1.154475      0.067500   \n",
       "max       23.000000  2.465864e+08  9.999448e-01      6.171800      2.654900   \n",
       "\n",
       "           DISTANCE  \n",
       "count  1.441200e+04  \n",
       "mean   8.997008e+04  \n",
       "std    1.969981e+05  \n",
       "min   -1.000000e+00  \n",
       "25%    0.000000e+00  \n",
       "50%    2.783500e+03  \n",
       "75%    8.004100e+04  \n",
       "max    1.968847e+06  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "geneList = df['GENE'].tolist()\n",
    "geneList = list(dict.fromkeys(geneList))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "#fig = ManhattanPlot(df, title='Manhattan plot')\n",
    "#fig.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![](Man_plot.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reading the GWAS Datasets to a DF "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "GWAS_DF2 = pd.read_excel('GWASData\\GWAS\\hum0014.v3.T2DM-2.v1.xlsx', keep_default_na=False, na_values=[\"\"]) \n",
    "GWAS_rename_DF2 = GWAS_DF2.rename(columns={\"#SNPID\": \"SNP\",\n",
    "                        \"chr\": \"CHR\",\n",
    "                        \"chrloc　(hg18)\":\"BP\",\n",
    "                       \"ptrend\":\"P\",\n",
    "                        \"maf_total\":\"annotation\"\n",
    "                       }, errors=\"raise\")\n",
    "\n",
    "with open('GWAS_rename_DF2.pickle', 'wb') as GWAS_rename_DF_Obj2: # Creating pickle file object\n",
    "    pickle.dump(GWAS_rename_DF2,GWAS_rename_DF_Obj2) # Dumping the object into the pickle file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "GWAS_DF2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Deseralising the pickle objec from directory"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('GWAS_rename_DF2.pickle', 'rb') as GWAS_rename_DF_Obj2:\n",
    "\n",
    "    GWAS_rename_DF_Ob2 = pickle.load(GWAS_rename_DF_Obj2)\n",
    "\n",
    "    GWAS_rename_DF_Obj2.close()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2237896</td>\n",
       "      <td>683</td>\n",
       "      <td>2631</td>\n",
       "      <td>2332</td>\n",
       "      <td>0</td>\n",
       "      <td>3359</td>\n",
       "      <td>9351</td>\n",
       "      <td>6696</td>\n",
       "      <td>14</td>\n",
       "      <td>2.640000e-30</td>\n",
       "      <td>1.289090</td>\n",
       "      <td>1.234070</td>\n",
       "      <td>1.346570</td>\n",
       "      <td>0.353967</td>\n",
       "      <td>0.414021</td>\n",
       "      <td>0.400487</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>2815016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2299620</td>\n",
       "      <td>745</td>\n",
       "      <td>2686</td>\n",
       "      <td>2180</td>\n",
       "      <td>35</td>\n",
       "      <td>3516</td>\n",
       "      <td>9448</td>\n",
       "      <td>6294</td>\n",
       "      <td>162</td>\n",
       "      <td>4.070000e-26</td>\n",
       "      <td>1.263870</td>\n",
       "      <td>1.210050</td>\n",
       "      <td>1.320080</td>\n",
       "      <td>0.372126</td>\n",
       "      <td>0.427874</td>\n",
       "      <td>0.415296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>2814871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2237892</td>\n",
       "      <td>653</td>\n",
       "      <td>2599</td>\n",
       "      <td>2393</td>\n",
       "      <td>1</td>\n",
       "      <td>3043</td>\n",
       "      <td>9333</td>\n",
       "      <td>7039</td>\n",
       "      <td>5</td>\n",
       "      <td>5.960000e-23</td>\n",
       "      <td>1.247430</td>\n",
       "      <td>1.193710</td>\n",
       "      <td>1.303560</td>\n",
       "      <td>0.345881</td>\n",
       "      <td>0.397090</td>\n",
       "      <td>0.385555</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>2796327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2383208</td>\n",
       "      <td>2163</td>\n",
       "      <td>2663</td>\n",
       "      <td>819</td>\n",
       "      <td>1</td>\n",
       "      <td>6400</td>\n",
       "      <td>9416</td>\n",
       "      <td>3601</td>\n",
       "      <td>3</td>\n",
       "      <td>7.440000e-19</td>\n",
       "      <td>0.823873</td>\n",
       "      <td>0.789288</td>\n",
       "      <td>0.859973</td>\n",
       "      <td>0.380957</td>\n",
       "      <td>0.427924</td>\n",
       "      <td>0.417345</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>9</td>\n",
       "      <td>22122076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs234853</td>\n",
       "      <td>2068</td>\n",
       "      <td>2743</td>\n",
       "      <td>827</td>\n",
       "      <td>8</td>\n",
       "      <td>6166</td>\n",
       "      <td>9582</td>\n",
       "      <td>3637</td>\n",
       "      <td>35</td>\n",
       "      <td>1.740000e-17</td>\n",
       "      <td>0.829544</td>\n",
       "      <td>0.794549</td>\n",
       "      <td>0.866081</td>\n",
       "      <td>0.389943</td>\n",
       "      <td>0.434769</td>\n",
       "      <td>0.424669</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>2807404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479083</th>\n",
       "      <td>rs4244872</td>\n",
       "      <td>838</td>\n",
       "      <td>2669</td>\n",
       "      <td>2139</td>\n",
       "      <td>0</td>\n",
       "      <td>2908</td>\n",
       "      <td>9126</td>\n",
       "      <td>7382</td>\n",
       "      <td>4</td>\n",
       "      <td>9.999920e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.957967</td>\n",
       "      <td>1.043880</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>33684341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479084</th>\n",
       "      <td>rs4693013</td>\n",
       "      <td>2640</td>\n",
       "      <td>2470</td>\n",
       "      <td>534</td>\n",
       "      <td>2</td>\n",
       "      <td>9165</td>\n",
       "      <td>8335</td>\n",
       "      <td>1919</td>\n",
       "      <td>1</td>\n",
       "      <td>9.999930e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.955791</td>\n",
       "      <td>1.046250</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>83134726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479085</th>\n",
       "      <td>rs6988928</td>\n",
       "      <td>83</td>\n",
       "      <td>1142</td>\n",
       "      <td>4421</td>\n",
       "      <td>0</td>\n",
       "      <td>276</td>\n",
       "      <td>3947</td>\n",
       "      <td>15197</td>\n",
       "      <td>0</td>\n",
       "      <td>9.999940e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.936889</td>\n",
       "      <td>1.067360</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>13781245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479086</th>\n",
       "      <td>rs372143</td>\n",
       "      <td>4341</td>\n",
       "      <td>1215</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>14916</td>\n",
       "      <td>4208</td>\n",
       "      <td>295</td>\n",
       "      <td>1</td>\n",
       "      <td>9.999970e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.938331</td>\n",
       "      <td>1.065720</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>14</td>\n",
       "      <td>67657091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479087</th>\n",
       "      <td>rs9928840</td>\n",
       "      <td>373</td>\n",
       "      <td>2077</td>\n",
       "      <td>3196</td>\n",
       "      <td>0</td>\n",
       "      <td>1203</td>\n",
       "      <td>7304</td>\n",
       "      <td>10913</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.952805</td>\n",
       "      <td>1.049530</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>16</td>\n",
       "      <td>77172531</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>479088 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0       rs2237896     683    2631    2332       0    3359    9351    6696   \n",
       "1       rs2299620     745    2686    2180      35    3516    9448    6294   \n",
       "2       rs2237892     653    2599    2393       1    3043    9333    7039   \n",
       "3       rs2383208    2163    2663     819       1    6400    9416    3601   \n",
       "4        rs234853    2068    2743     827       8    6166    9582    3637   \n",
       "...           ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "479083  rs4244872     838    2669    2139       0    2908    9126    7382   \n",
       "479084  rs4693013    2640    2470     534       2    9165    8335    1919   \n",
       "479085  rs6988928      83    1142    4421       0     276    3947   15197   \n",
       "479086   rs372143    4341    1215      90       0   14916    4208     295   \n",
       "479087  rs9928840     373    2077    3196       0    1203    7304   10913   \n",
       "\n",
       "        ctrl??             P  OR_logistic  OR_lower  OR_upper  maf_case  \\\n",
       "0           14  2.640000e-30     1.289090  1.234070  1.346570  0.353967   \n",
       "1          162  4.070000e-26     1.263870  1.210050  1.320080  0.372126   \n",
       "2            5  5.960000e-23     1.247430  1.193710  1.303560  0.345881   \n",
       "3            3  7.440000e-19     0.823873  0.789288  0.859973  0.380957   \n",
       "4           35  1.740000e-17     0.829544  0.794549  0.866081  0.389943   \n",
       "...        ...           ...          ...       ...       ...       ...   \n",
       "479083       4  9.999920e-01     1.000000  0.957967  1.043880  0.384786   \n",
       "479084       1  9.999930e-01     1.000000  0.955791  1.046250  0.313430   \n",
       "479085       0  9.999940e-01     1.000000  0.936889  1.067360  0.115834   \n",
       "479086       1  9.999970e-01     1.000000  0.938331  1.065720  0.123539   \n",
       "479087       0  1.000000e+00     1.000000  0.952805  1.049530  0.250000   \n",
       "\n",
       "        maf_ctrl  annotation alleleA alleleB  CHR        BP  \n",
       "0       0.414021    0.400487       A       G   11   2815016  \n",
       "1       0.427874    0.415296       A       G   11   2814871  \n",
       "2       0.397090    0.385555       T       C   11   2796327  \n",
       "3       0.427924    0.417345       A       G    9  22122076  \n",
       "4       0.434769    0.424669       T       C   11   2807404  \n",
       "...          ...         ...     ...     ...  ...       ...  \n",
       "479083  0.384786    0.384786       T       C   12  33684341  \n",
       "479084  0.313430    0.313430       A       G    4  83134726  \n",
       "479085  0.115834    0.115834       A       G    8  13781245  \n",
       "479086  0.123539    0.123539       A       G   14  67657091  \n",
       "479087  0.250000    0.250000       A       G   16  77172531  \n",
       "\n",
       "[479088 rows x 20 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GWAS_rename_DF_Ob2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>4.790880e+05</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>479088.000000</td>\n",
       "      <td>4.790880e+05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>1793.843400</td>\n",
       "      <td>1859.663166</td>\n",
       "      <td>1991.130588</td>\n",
       "      <td>1.362846</td>\n",
       "      <td>6167.959431</td>\n",
       "      <td>6400.690276</td>\n",
       "      <td>6846.888221</td>\n",
       "      <td>4.462072</td>\n",
       "      <td>4.867605e-01</td>\n",
       "      <td>1.000542</td>\n",
       "      <td>0.942809</td>\n",
       "      <td>1.062603</td>\n",
       "      <td>0.246531</td>\n",
       "      <td>0.246535</td>\n",
       "      <td>0.246539</td>\n",
       "      <td>8.783136</td>\n",
       "      <td>7.764898e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1667.189797</td>\n",
       "      <td>815.539372</td>\n",
       "      <td>1705.398125</td>\n",
       "      <td>3.816454</td>\n",
       "      <td>5734.408601</td>\n",
       "      <td>2805.966913</td>\n",
       "      <td>5866.219741</td>\n",
       "      <td>12.771642</td>\n",
       "      <td>2.918356e-01</td>\n",
       "      <td>0.034712</td>\n",
       "      <td>0.039922</td>\n",
       "      <td>0.049094</td>\n",
       "      <td>0.141741</td>\n",
       "      <td>0.141734</td>\n",
       "      <td>0.141732</td>\n",
       "      <td>5.780515</td>\n",
       "      <td>5.671348e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>81.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>374.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.640000e-30</td>\n",
       "      <td>0.657528</td>\n",
       "      <td>0.518163</td>\n",
       "      <td>0.767680</td>\n",
       "      <td>0.007173</td>\n",
       "      <td>0.010015</td>\n",
       "      <td>0.009575</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.124400e+04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>289.000000</td>\n",
       "      <td>1208.000000</td>\n",
       "      <td>394.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>992.000000</td>\n",
       "      <td>4159.000000</td>\n",
       "      <td>1355.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.312057e-01</td>\n",
       "      <td>0.980746</td>\n",
       "      <td>0.926618</td>\n",
       "      <td>1.033610</td>\n",
       "      <td>0.122630</td>\n",
       "      <td>0.122760</td>\n",
       "      <td>0.122736</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>3.056896e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>1262.000000</td>\n",
       "      <td>2058.000000</td>\n",
       "      <td>1586.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4337.000000</td>\n",
       "      <td>7082.000000</td>\n",
       "      <td>5450.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.820170e-01</td>\n",
       "      <td>0.999938</td>\n",
       "      <td>0.948604</td>\n",
       "      <td>1.054060</td>\n",
       "      <td>0.241808</td>\n",
       "      <td>0.241798</td>\n",
       "      <td>0.241820</td>\n",
       "      <td>8.000000</td>\n",
       "      <td>6.789811e+07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>3079.000000</td>\n",
       "      <td>2610.000000</td>\n",
       "      <td>3405.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>10584.000000</td>\n",
       "      <td>8988.000000</td>\n",
       "      <td>11704.000000</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>7.395060e-01</td>\n",
       "      <td>1.019520</td>\n",
       "      <td>0.967411</td>\n",
       "      <td>1.079010</td>\n",
       "      <td>0.368402</td>\n",
       "      <td>0.368317</td>\n",
       "      <td>0.368412</td>\n",
       "      <td>13.000000</td>\n",
       "      <td>1.143118e+08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>5565.000000</td>\n",
       "      <td>2955.000000</td>\n",
       "      <td>5565.000000</td>\n",
       "      <td>56.000000</td>\n",
       "      <td>19037.000000</td>\n",
       "      <td>9957.000000</td>\n",
       "      <td>19037.000000</td>\n",
       "      <td>194.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.520850</td>\n",
       "      <td>1.303830</td>\n",
       "      <td>1.929890</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>22.000000</td>\n",
       "      <td>2.471773e+08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              case11         case12         case22         case??  \\\n",
       "count  479088.000000  479088.000000  479088.000000  479088.000000   \n",
       "mean     1793.843400    1859.663166    1991.130588       1.362846   \n",
       "std      1667.189797     815.539372    1705.398125       3.816454   \n",
       "min         0.000000      81.000000       0.000000       0.000000   \n",
       "25%       289.000000    1208.000000     394.000000       0.000000   \n",
       "50%      1262.000000    2058.000000    1586.000000       0.000000   \n",
       "75%      3079.000000    2610.000000    3405.000000       1.000000   \n",
       "max      5565.000000    2955.000000    5565.000000      56.000000   \n",
       "\n",
       "              ctrl11         ctrl12         ctrl22         ctrl??  \\\n",
       "count  479088.000000  479088.000000  479088.000000  479088.000000   \n",
       "mean     6167.959431    6400.690276    6846.888221       4.462072   \n",
       "std      5734.408601    2805.966913    5866.219741      12.771642   \n",
       "min         0.000000     374.000000       0.000000       0.000000   \n",
       "25%       992.000000    4159.000000    1355.000000       0.000000   \n",
       "50%      4337.000000    7082.000000    5450.000000       1.000000   \n",
       "75%     10584.000000    8988.000000   11704.000000       3.000000   \n",
       "max     19037.000000    9957.000000   19037.000000     194.000000   \n",
       "\n",
       "                  P    OR_logistic       OR_lower       OR_upper  \\\n",
       "count  4.790880e+05  479088.000000  479088.000000  479088.000000   \n",
       "mean   4.867605e-01       1.000542       0.942809       1.062603   \n",
       "std    2.918356e-01       0.034712       0.039922       0.049094   \n",
       "min    2.640000e-30       0.657528       0.518163       0.767680   \n",
       "25%    2.312057e-01       0.980746       0.926618       1.033610   \n",
       "50%    4.820170e-01       0.999938       0.948604       1.054060   \n",
       "75%    7.395060e-01       1.019520       0.967411       1.079010   \n",
       "max    1.000000e+00       1.520850       1.303830       1.929890   \n",
       "\n",
       "            maf_case       maf_ctrl     annotation            CHR  \\\n",
       "count  479088.000000  479088.000000  479088.000000  479088.000000   \n",
       "mean        0.246531       0.246535       0.246539       8.783136   \n",
       "std         0.141741       0.141734       0.141732       5.780515   \n",
       "min         0.007173       0.010015       0.009575       1.000000   \n",
       "25%         0.122630       0.122760       0.122736       4.000000   \n",
       "50%         0.241808       0.241798       0.241820       8.000000   \n",
       "75%         0.368402       0.368317       0.368412      13.000000   \n",
       "max         0.500000       0.500000       0.500000      22.000000   \n",
       "\n",
       "                 BP  \n",
       "count  4.790880e+05  \n",
       "mean   7.764898e+07  \n",
       "std    5.671348e+07  \n",
       "min    1.124400e+04  \n",
       "25%    3.056896e+07  \n",
       "50%    6.789811e+07  \n",
       "75%    1.143118e+08  \n",
       "max    2.471773e+08  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "GWAS_rename_DF_Ob2.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Removing all rows with Null value of SNPS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [SNP, case11, case12, case22, case??, ctrl11, ctrl12, ctrl22, ctrl??, P, OR_logistic, OR_lower, OR_upper, maf_case, maf_ctrl, annotation, alleleA, alleleB, CHR, BP]\n",
       "Index: []"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#GWAS_rename_DF.isnull().sum()\n",
    "#Creaties a series of of null values in SNPS columns  \n",
    "#gapminder_no_NA = gapminder[gapminder.year.notnull()]\n",
    "GWAS_rename_DF_bool_series = GWAS_rename_DF_Ob2[GWAS_rename_DF_Ob2['SNP'].isnull()] \n",
    "# filtering data  \n",
    "# displaying data only with Gender = NaN  \n",
    "#GWAS_rename_DF_Ob2[GWAS_rename_DF_bool_series] \n",
    "GWAS_rename_DF_bool_series"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2237896</td>\n",
       "      <td>683</td>\n",
       "      <td>2631</td>\n",
       "      <td>2332</td>\n",
       "      <td>0</td>\n",
       "      <td>3359</td>\n",
       "      <td>9351</td>\n",
       "      <td>6696</td>\n",
       "      <td>14</td>\n",
       "      <td>2.640000e-30</td>\n",
       "      <td>1.289090</td>\n",
       "      <td>1.234070</td>\n",
       "      <td>1.346570</td>\n",
       "      <td>0.353967</td>\n",
       "      <td>0.414021</td>\n",
       "      <td>0.400487</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>2815016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2299620</td>\n",
       "      <td>745</td>\n",
       "      <td>2686</td>\n",
       "      <td>2180</td>\n",
       "      <td>35</td>\n",
       "      <td>3516</td>\n",
       "      <td>9448</td>\n",
       "      <td>6294</td>\n",
       "      <td>162</td>\n",
       "      <td>4.070000e-26</td>\n",
       "      <td>1.263870</td>\n",
       "      <td>1.210050</td>\n",
       "      <td>1.320080</td>\n",
       "      <td>0.372126</td>\n",
       "      <td>0.427874</td>\n",
       "      <td>0.415296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>2814871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2237892</td>\n",
       "      <td>653</td>\n",
       "      <td>2599</td>\n",
       "      <td>2393</td>\n",
       "      <td>1</td>\n",
       "      <td>3043</td>\n",
       "      <td>9333</td>\n",
       "      <td>7039</td>\n",
       "      <td>5</td>\n",
       "      <td>5.960000e-23</td>\n",
       "      <td>1.247430</td>\n",
       "      <td>1.193710</td>\n",
       "      <td>1.303560</td>\n",
       "      <td>0.345881</td>\n",
       "      <td>0.397090</td>\n",
       "      <td>0.385555</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>2796327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2383208</td>\n",
       "      <td>2163</td>\n",
       "      <td>2663</td>\n",
       "      <td>819</td>\n",
       "      <td>1</td>\n",
       "      <td>6400</td>\n",
       "      <td>9416</td>\n",
       "      <td>3601</td>\n",
       "      <td>3</td>\n",
       "      <td>7.440000e-19</td>\n",
       "      <td>0.823873</td>\n",
       "      <td>0.789288</td>\n",
       "      <td>0.859973</td>\n",
       "      <td>0.380957</td>\n",
       "      <td>0.427924</td>\n",
       "      <td>0.417345</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>9</td>\n",
       "      <td>22122076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs234853</td>\n",
       "      <td>2068</td>\n",
       "      <td>2743</td>\n",
       "      <td>827</td>\n",
       "      <td>8</td>\n",
       "      <td>6166</td>\n",
       "      <td>9582</td>\n",
       "      <td>3637</td>\n",
       "      <td>35</td>\n",
       "      <td>1.740000e-17</td>\n",
       "      <td>0.829544</td>\n",
       "      <td>0.794549</td>\n",
       "      <td>0.866081</td>\n",
       "      <td>0.389943</td>\n",
       "      <td>0.434769</td>\n",
       "      <td>0.424669</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>2807404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479083</th>\n",
       "      <td>rs4244872</td>\n",
       "      <td>838</td>\n",
       "      <td>2669</td>\n",
       "      <td>2139</td>\n",
       "      <td>0</td>\n",
       "      <td>2908</td>\n",
       "      <td>9126</td>\n",
       "      <td>7382</td>\n",
       "      <td>4</td>\n",
       "      <td>9.999920e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.957967</td>\n",
       "      <td>1.043880</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>33684341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479084</th>\n",
       "      <td>rs4693013</td>\n",
       "      <td>2640</td>\n",
       "      <td>2470</td>\n",
       "      <td>534</td>\n",
       "      <td>2</td>\n",
       "      <td>9165</td>\n",
       "      <td>8335</td>\n",
       "      <td>1919</td>\n",
       "      <td>1</td>\n",
       "      <td>9.999930e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.955791</td>\n",
       "      <td>1.046250</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>83134726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479085</th>\n",
       "      <td>rs6988928</td>\n",
       "      <td>83</td>\n",
       "      <td>1142</td>\n",
       "      <td>4421</td>\n",
       "      <td>0</td>\n",
       "      <td>276</td>\n",
       "      <td>3947</td>\n",
       "      <td>15197</td>\n",
       "      <td>0</td>\n",
       "      <td>9.999940e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.936889</td>\n",
       "      <td>1.067360</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>13781245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479086</th>\n",
       "      <td>rs372143</td>\n",
       "      <td>4341</td>\n",
       "      <td>1215</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>14916</td>\n",
       "      <td>4208</td>\n",
       "      <td>295</td>\n",
       "      <td>1</td>\n",
       "      <td>9.999970e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.938331</td>\n",
       "      <td>1.065720</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>14</td>\n",
       "      <td>67657091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479087</th>\n",
       "      <td>rs9928840</td>\n",
       "      <td>373</td>\n",
       "      <td>2077</td>\n",
       "      <td>3196</td>\n",
       "      <td>0</td>\n",
       "      <td>1203</td>\n",
       "      <td>7304</td>\n",
       "      <td>10913</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.952805</td>\n",
       "      <td>1.049530</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>16</td>\n",
       "      <td>77172531</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>479088 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0       rs2237896     683    2631    2332       0    3359    9351    6696   \n",
       "1       rs2299620     745    2686    2180      35    3516    9448    6294   \n",
       "2       rs2237892     653    2599    2393       1    3043    9333    7039   \n",
       "3       rs2383208    2163    2663     819       1    6400    9416    3601   \n",
       "4        rs234853    2068    2743     827       8    6166    9582    3637   \n",
       "...           ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "479083  rs4244872     838    2669    2139       0    2908    9126    7382   \n",
       "479084  rs4693013    2640    2470     534       2    9165    8335    1919   \n",
       "479085  rs6988928      83    1142    4421       0     276    3947   15197   \n",
       "479086   rs372143    4341    1215      90       0   14916    4208     295   \n",
       "479087  rs9928840     373    2077    3196       0    1203    7304   10913   \n",
       "\n",
       "        ctrl??             P  OR_logistic  OR_lower  OR_upper  maf_case  \\\n",
       "0           14  2.640000e-30     1.289090  1.234070  1.346570  0.353967   \n",
       "1          162  4.070000e-26     1.263870  1.210050  1.320080  0.372126   \n",
       "2            5  5.960000e-23     1.247430  1.193710  1.303560  0.345881   \n",
       "3            3  7.440000e-19     0.823873  0.789288  0.859973  0.380957   \n",
       "4           35  1.740000e-17     0.829544  0.794549  0.866081  0.389943   \n",
       "...        ...           ...          ...       ...       ...       ...   \n",
       "479083       4  9.999920e-01     1.000000  0.957967  1.043880  0.384786   \n",
       "479084       1  9.999930e-01     1.000000  0.955791  1.046250  0.313430   \n",
       "479085       0  9.999940e-01     1.000000  0.936889  1.067360  0.115834   \n",
       "479086       1  9.999970e-01     1.000000  0.938331  1.065720  0.123539   \n",
       "479087       0  1.000000e+00     1.000000  0.952805  1.049530  0.250000   \n",
       "\n",
       "        maf_ctrl  annotation alleleA alleleB  CHR        BP  \n",
       "0       0.414021    0.400487       A       G   11   2815016  \n",
       "1       0.427874    0.415296       A       G   11   2814871  \n",
       "2       0.397090    0.385555       T       C   11   2796327  \n",
       "3       0.427924    0.417345       A       G    9  22122076  \n",
       "4       0.434769    0.424669       T       C   11   2807404  \n",
       "...          ...         ...     ...     ...  ...       ...  \n",
       "479083  0.384786    0.384786       T       C   12  33684341  \n",
       "479084  0.313430    0.313430       A       G    4  83134726  \n",
       "479085  0.115834    0.115834       A       G    8  13781245  \n",
       "479086  0.123539    0.123539       A       G   14  67657091  \n",
       "479087  0.250000    0.250000       A       G   16  77172531  \n",
       "\n",
       "[479088 rows x 20 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# creating bool series True for NaN values  \n",
    "GWAS_rename_DF_Ob2_bool_series = pd.notnull(GWAS_rename_DF_Ob2[\"SNP\"])  \n",
    "    \n",
    "# filtering data  \n",
    "# displayind data only with Gender = Not NaN  \n",
    "GWAS_rename_DF_Ob2 = GWAS_rename_DF_Ob2[GWAS_rename_DF_Ob2_bool_series]  \n",
    "GWAS_rename_DF_Ob2 = GWAS_rename_DF_Ob2.dropna()\n",
    "GWAS_rename_DF_Ob2 =GWAS_rename_DF_Ob2.drop_duplicates(['SNP'], keep = 'first')\n",
    "GWAS_rename_DF_Ob2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23046</th>\n",
       "      <td>rs7302233</td>\n",
       "      <td>4169</td>\n",
       "      <td>1358</td>\n",
       "      <td>119</td>\n",
       "      <td>0</td>\n",
       "      <td>14052</td>\n",
       "      <td>4942</td>\n",
       "      <td>425</td>\n",
       "      <td>1</td>\n",
       "      <td>0.039738</td>\n",
       "      <td>0.939128</td>\n",
       "      <td>0.884556</td>\n",
       "      <td>0.997067</td>\n",
       "      <td>0.141339</td>\n",
       "      <td>0.149132</td>\n",
       "      <td>0.147377</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>80976062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30344</th>\n",
       "      <td>rs1323271</td>\n",
       "      <td>898</td>\n",
       "      <td>2692</td>\n",
       "      <td>2055</td>\n",
       "      <td>1</td>\n",
       "      <td>3256</td>\n",
       "      <td>9320</td>\n",
       "      <td>6840</td>\n",
       "      <td>4</td>\n",
       "      <td>0.053046</td>\n",
       "      <td>1.043010</td>\n",
       "      <td>0.999444</td>\n",
       "      <td>1.088470</td>\n",
       "      <td>0.397520</td>\n",
       "      <td>0.407705</td>\n",
       "      <td>0.405411</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>9</td>\n",
       "      <td>834487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33436</th>\n",
       "      <td>rs12056091</td>\n",
       "      <td>1457</td>\n",
       "      <td>2854</td>\n",
       "      <td>1335</td>\n",
       "      <td>0</td>\n",
       "      <td>5269</td>\n",
       "      <td>9693</td>\n",
       "      <td>4458</td>\n",
       "      <td>0</td>\n",
       "      <td>0.058940</td>\n",
       "      <td>1.041280</td>\n",
       "      <td>0.998471</td>\n",
       "      <td>1.085930</td>\n",
       "      <td>0.489196</td>\n",
       "      <td>0.479119</td>\n",
       "      <td>0.481389</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>7</td>\n",
       "      <td>96596607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40155</th>\n",
       "      <td>rs11082439</td>\n",
       "      <td>117</td>\n",
       "      <td>1284</td>\n",
       "      <td>4244</td>\n",
       "      <td>1</td>\n",
       "      <td>321</td>\n",
       "      <td>4328</td>\n",
       "      <td>14769</td>\n",
       "      <td>2</td>\n",
       "      <td>0.071858</td>\n",
       "      <td>0.945065</td>\n",
       "      <td>0.888668</td>\n",
       "      <td>1.005040</td>\n",
       "      <td>0.134455</td>\n",
       "      <td>0.127974</td>\n",
       "      <td>0.129434</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>18</td>\n",
       "      <td>41124301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40160</th>\n",
       "      <td>500301</td>\n",
       "      <td>146</td>\n",
       "      <td>1506</td>\n",
       "      <td>3994</td>\n",
       "      <td>0</td>\n",
       "      <td>482</td>\n",
       "      <td>4949</td>\n",
       "      <td>13988</td>\n",
       "      <td>1</td>\n",
       "      <td>0.071864</td>\n",
       "      <td>0.948827</td>\n",
       "      <td>0.896072</td>\n",
       "      <td>1.004690</td>\n",
       "      <td>0.159228</td>\n",
       "      <td>0.152248</td>\n",
       "      <td>0.153820</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>224098852</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475385</th>\n",
       "      <td>rs944269</td>\n",
       "      <td>3907</td>\n",
       "      <td>1570</td>\n",
       "      <td>168</td>\n",
       "      <td>1</td>\n",
       "      <td>13467</td>\n",
       "      <td>5344</td>\n",
       "      <td>607</td>\n",
       "      <td>2</td>\n",
       "      <td>0.991710</td>\n",
       "      <td>0.999706</td>\n",
       "      <td>0.945764</td>\n",
       "      <td>1.056730</td>\n",
       "      <td>0.168822</td>\n",
       "      <td>0.168864</td>\n",
       "      <td>0.168854</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>20</td>\n",
       "      <td>59647962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475498</th>\n",
       "      <td>rs10757447</td>\n",
       "      <td>4011</td>\n",
       "      <td>1474</td>\n",
       "      <td>161</td>\n",
       "      <td>0</td>\n",
       "      <td>13718</td>\n",
       "      <td>5228</td>\n",
       "      <td>474</td>\n",
       "      <td>0</td>\n",
       "      <td>0.991976</td>\n",
       "      <td>1.000290</td>\n",
       "      <td>0.944600</td>\n",
       "      <td>1.059270</td>\n",
       "      <td>0.159051</td>\n",
       "      <td>0.159011</td>\n",
       "      <td>0.159020</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>9</td>\n",
       "      <td>23720654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475668</th>\n",
       "      <td>rs6942837</td>\n",
       "      <td>520</td>\n",
       "      <td>2368</td>\n",
       "      <td>2758</td>\n",
       "      <td>0</td>\n",
       "      <td>1801</td>\n",
       "      <td>8122</td>\n",
       "      <td>9497</td>\n",
       "      <td>0</td>\n",
       "      <td>0.992359</td>\n",
       "      <td>1.000220</td>\n",
       "      <td>0.955743</td>\n",
       "      <td>1.046770</td>\n",
       "      <td>0.301807</td>\n",
       "      <td>0.301854</td>\n",
       "      <td>0.301843</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>7</td>\n",
       "      <td>47737034</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476312</th>\n",
       "      <td>rs6768455</td>\n",
       "      <td>188</td>\n",
       "      <td>1717</td>\n",
       "      <td>3739</td>\n",
       "      <td>2</td>\n",
       "      <td>690</td>\n",
       "      <td>5820</td>\n",
       "      <td>12909</td>\n",
       "      <td>1</td>\n",
       "      <td>0.993736</td>\n",
       "      <td>0.999784</td>\n",
       "      <td>0.947409</td>\n",
       "      <td>1.055060</td>\n",
       "      <td>0.185418</td>\n",
       "      <td>0.185385</td>\n",
       "      <td>0.185393</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3</td>\n",
       "      <td>85123339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476610</th>\n",
       "      <td>rs7025476</td>\n",
       "      <td>640</td>\n",
       "      <td>2493</td>\n",
       "      <td>2513</td>\n",
       "      <td>0</td>\n",
       "      <td>2201</td>\n",
       "      <td>8575</td>\n",
       "      <td>8641</td>\n",
       "      <td>3</td>\n",
       "      <td>0.994392</td>\n",
       "      <td>1.000160</td>\n",
       "      <td>0.956859</td>\n",
       "      <td>1.045420</td>\n",
       "      <td>0.334130</td>\n",
       "      <td>0.334166</td>\n",
       "      <td>0.334158</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>9</td>\n",
       "      <td>32715773</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>624 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "23046    rs7302233    4169    1358     119       0   14052    4942     425   \n",
       "30344    rs1323271     898    2692    2055       1    3256    9320    6840   \n",
       "33436   rs12056091    1457    2854    1335       0    5269    9693    4458   \n",
       "40155   rs11082439     117    1284    4244       1     321    4328   14769   \n",
       "40160       500301     146    1506    3994       0     482    4949   13988   \n",
       "...            ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "475385    rs944269    3907    1570     168       1   13467    5344     607   \n",
       "475498  rs10757447    4011    1474     161       0   13718    5228     474   \n",
       "475668   rs6942837     520    2368    2758       0    1801    8122    9497   \n",
       "476312   rs6768455     188    1717    3739       2     690    5820   12909   \n",
       "476610   rs7025476     640    2493    2513       0    2201    8575    8641   \n",
       "\n",
       "        ctrl??         P  OR_logistic  OR_lower  OR_upper  maf_case  maf_ctrl  \\\n",
       "23046        1  0.039738     0.939128  0.884556  0.997067  0.141339  0.149132   \n",
       "30344        4  0.053046     1.043010  0.999444  1.088470  0.397520  0.407705   \n",
       "33436        0  0.058940     1.041280  0.998471  1.085930  0.489196  0.479119   \n",
       "40155        2  0.071858     0.945065  0.888668  1.005040  0.134455  0.127974   \n",
       "40160        1  0.071864     0.948827  0.896072  1.004690  0.159228  0.152248   \n",
       "...        ...       ...          ...       ...       ...       ...       ...   \n",
       "475385       2  0.991710     0.999706  0.945764  1.056730  0.168822  0.168864   \n",
       "475498       0  0.991976     1.000290  0.944600  1.059270  0.159051  0.159011   \n",
       "475668       0  0.992359     1.000220  0.955743  1.046770  0.301807  0.301854   \n",
       "476312       1  0.993736     0.999784  0.947409  1.055060  0.185418  0.185385   \n",
       "476610       3  0.994392     1.000160  0.956859  1.045420  0.334130  0.334166   \n",
       "\n",
       "        annotation alleleA alleleB  CHR         BP  \n",
       "23046     0.147377       A       C   12   80976062  \n",
       "30344     0.405411       T       C    9     834487  \n",
       "33436     0.481389       T       C    7   96596607  \n",
       "40155     0.129434       T       C   18   41124301  \n",
       "40160     0.153820       A       G    1  224098852  \n",
       "...            ...     ...     ...  ...        ...  \n",
       "475385    0.168854       T       C   20   59647962  \n",
       "475498    0.159020       A       G    9   23720654  \n",
       "475668    0.301843       T       C    7   47737034  \n",
       "476312    0.185393       A       G    3   85123339  \n",
       "476610    0.334158       A       C    9   32715773  \n",
       "\n",
       "[624 rows x 20 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicateRowsDF2 = GWAS_rename_DF_Ob2[GWAS_rename_DF_Ob2.duplicated(['BP'])]\n",
    "duplicateRowsDF2_droped = GWAS_rename_DF_Ob2.drop_duplicates(subset='BP', keep=\"first\")\n",
    "#df = df.drop_duplicates('column_name', keep='last')\n",
    "duplicateRowsDF2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2237896</td>\n",
       "      <td>683</td>\n",
       "      <td>2631</td>\n",
       "      <td>2332</td>\n",
       "      <td>0</td>\n",
       "      <td>3359</td>\n",
       "      <td>9351</td>\n",
       "      <td>6696</td>\n",
       "      <td>14</td>\n",
       "      <td>2.640000e-30</td>\n",
       "      <td>1.289090</td>\n",
       "      <td>1.234070</td>\n",
       "      <td>1.346570</td>\n",
       "      <td>0.353967</td>\n",
       "      <td>0.414021</td>\n",
       "      <td>0.400487</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>2815016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2299620</td>\n",
       "      <td>745</td>\n",
       "      <td>2686</td>\n",
       "      <td>2180</td>\n",
       "      <td>35</td>\n",
       "      <td>3516</td>\n",
       "      <td>9448</td>\n",
       "      <td>6294</td>\n",
       "      <td>162</td>\n",
       "      <td>4.070000e-26</td>\n",
       "      <td>1.263870</td>\n",
       "      <td>1.210050</td>\n",
       "      <td>1.320080</td>\n",
       "      <td>0.372126</td>\n",
       "      <td>0.427874</td>\n",
       "      <td>0.415296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>2814871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2237892</td>\n",
       "      <td>653</td>\n",
       "      <td>2599</td>\n",
       "      <td>2393</td>\n",
       "      <td>1</td>\n",
       "      <td>3043</td>\n",
       "      <td>9333</td>\n",
       "      <td>7039</td>\n",
       "      <td>5</td>\n",
       "      <td>5.960000e-23</td>\n",
       "      <td>1.247430</td>\n",
       "      <td>1.193710</td>\n",
       "      <td>1.303560</td>\n",
       "      <td>0.345881</td>\n",
       "      <td>0.397090</td>\n",
       "      <td>0.385555</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>2796327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2383208</td>\n",
       "      <td>2163</td>\n",
       "      <td>2663</td>\n",
       "      <td>819</td>\n",
       "      <td>1</td>\n",
       "      <td>6400</td>\n",
       "      <td>9416</td>\n",
       "      <td>3601</td>\n",
       "      <td>3</td>\n",
       "      <td>7.440000e-19</td>\n",
       "      <td>0.823873</td>\n",
       "      <td>0.789288</td>\n",
       "      <td>0.859973</td>\n",
       "      <td>0.380957</td>\n",
       "      <td>0.427924</td>\n",
       "      <td>0.417345</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>9</td>\n",
       "      <td>22122076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs234853</td>\n",
       "      <td>2068</td>\n",
       "      <td>2743</td>\n",
       "      <td>827</td>\n",
       "      <td>8</td>\n",
       "      <td>6166</td>\n",
       "      <td>9582</td>\n",
       "      <td>3637</td>\n",
       "      <td>35</td>\n",
       "      <td>1.740000e-17</td>\n",
       "      <td>0.829544</td>\n",
       "      <td>0.794549</td>\n",
       "      <td>0.866081</td>\n",
       "      <td>0.389943</td>\n",
       "      <td>0.434769</td>\n",
       "      <td>0.424669</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>2807404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479083</th>\n",
       "      <td>rs4244872</td>\n",
       "      <td>838</td>\n",
       "      <td>2669</td>\n",
       "      <td>2139</td>\n",
       "      <td>0</td>\n",
       "      <td>2908</td>\n",
       "      <td>9126</td>\n",
       "      <td>7382</td>\n",
       "      <td>4</td>\n",
       "      <td>9.999920e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.957967</td>\n",
       "      <td>1.043880</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>0.384786</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>33684341</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479084</th>\n",
       "      <td>rs4693013</td>\n",
       "      <td>2640</td>\n",
       "      <td>2470</td>\n",
       "      <td>534</td>\n",
       "      <td>2</td>\n",
       "      <td>9165</td>\n",
       "      <td>8335</td>\n",
       "      <td>1919</td>\n",
       "      <td>1</td>\n",
       "      <td>9.999930e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.955791</td>\n",
       "      <td>1.046250</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>0.313430</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>83134726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479085</th>\n",
       "      <td>rs6988928</td>\n",
       "      <td>83</td>\n",
       "      <td>1142</td>\n",
       "      <td>4421</td>\n",
       "      <td>0</td>\n",
       "      <td>276</td>\n",
       "      <td>3947</td>\n",
       "      <td>15197</td>\n",
       "      <td>0</td>\n",
       "      <td>9.999940e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.936889</td>\n",
       "      <td>1.067360</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>0.115834</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>8</td>\n",
       "      <td>13781245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479086</th>\n",
       "      <td>rs372143</td>\n",
       "      <td>4341</td>\n",
       "      <td>1215</td>\n",
       "      <td>90</td>\n",
       "      <td>0</td>\n",
       "      <td>14916</td>\n",
       "      <td>4208</td>\n",
       "      <td>295</td>\n",
       "      <td>1</td>\n",
       "      <td>9.999970e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.938331</td>\n",
       "      <td>1.065720</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>0.123539</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>14</td>\n",
       "      <td>67657091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>479087</th>\n",
       "      <td>rs9928840</td>\n",
       "      <td>373</td>\n",
       "      <td>2077</td>\n",
       "      <td>3196</td>\n",
       "      <td>0</td>\n",
       "      <td>1203</td>\n",
       "      <td>7304</td>\n",
       "      <td>10913</td>\n",
       "      <td>0</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.952805</td>\n",
       "      <td>1.049530</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>16</td>\n",
       "      <td>77172531</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>478464 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0       rs2237896     683    2631    2332       0    3359    9351    6696   \n",
       "1       rs2299620     745    2686    2180      35    3516    9448    6294   \n",
       "2       rs2237892     653    2599    2393       1    3043    9333    7039   \n",
       "3       rs2383208    2163    2663     819       1    6400    9416    3601   \n",
       "4        rs234853    2068    2743     827       8    6166    9582    3637   \n",
       "...           ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "479083  rs4244872     838    2669    2139       0    2908    9126    7382   \n",
       "479084  rs4693013    2640    2470     534       2    9165    8335    1919   \n",
       "479085  rs6988928      83    1142    4421       0     276    3947   15197   \n",
       "479086   rs372143    4341    1215      90       0   14916    4208     295   \n",
       "479087  rs9928840     373    2077    3196       0    1203    7304   10913   \n",
       "\n",
       "        ctrl??             P  OR_logistic  OR_lower  OR_upper  maf_case  \\\n",
       "0           14  2.640000e-30     1.289090  1.234070  1.346570  0.353967   \n",
       "1          162  4.070000e-26     1.263870  1.210050  1.320080  0.372126   \n",
       "2            5  5.960000e-23     1.247430  1.193710  1.303560  0.345881   \n",
       "3            3  7.440000e-19     0.823873  0.789288  0.859973  0.380957   \n",
       "4           35  1.740000e-17     0.829544  0.794549  0.866081  0.389943   \n",
       "...        ...           ...          ...       ...       ...       ...   \n",
       "479083       4  9.999920e-01     1.000000  0.957967  1.043880  0.384786   \n",
       "479084       1  9.999930e-01     1.000000  0.955791  1.046250  0.313430   \n",
       "479085       0  9.999940e-01     1.000000  0.936889  1.067360  0.115834   \n",
       "479086       1  9.999970e-01     1.000000  0.938331  1.065720  0.123539   \n",
       "479087       0  1.000000e+00     1.000000  0.952805  1.049530  0.250000   \n",
       "\n",
       "        maf_ctrl  annotation alleleA alleleB  CHR        BP  \n",
       "0       0.414021    0.400487       A       G   11   2815016  \n",
       "1       0.427874    0.415296       A       G   11   2814871  \n",
       "2       0.397090    0.385555       T       C   11   2796327  \n",
       "3       0.427924    0.417345       A       G    9  22122076  \n",
       "4       0.434769    0.424669       T       C   11   2807404  \n",
       "...          ...         ...     ...     ...  ...       ...  \n",
       "479083  0.384786    0.384786       T       C   12  33684341  \n",
       "479084  0.313430    0.313430       A       G    4  83134726  \n",
       "479085  0.115834    0.115834       A       G    8  13781245  \n",
       "479086  0.123539    0.123539       A       G   14  67657091  \n",
       "479087  0.250000    0.250000       A       G   16  77172531  \n",
       "\n",
       "[478464 rows x 20 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "duplicateRowsDF2_droped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Retrieving rs SNPS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>310870</th>\n",
       "      <td>rs10159099</td>\n",
       "      <td>3552</td>\n",
       "      <td>1849</td>\n",
       "      <td>245</td>\n",
       "      <td>0</td>\n",
       "      <td>12268</td>\n",
       "      <td>6337</td>\n",
       "      <td>814</td>\n",
       "      <td>1</td>\n",
       "      <td>0.634300</td>\n",
       "      <td>1.012640</td>\n",
       "      <td>0.961567</td>\n",
       "      <td>1.06642</td>\n",
       "      <td>0.207138</td>\n",
       "      <td>0.205083</td>\n",
       "      <td>0.205546</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>56573324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38939</th>\n",
       "      <td>rs4518905</td>\n",
       "      <td>965</td>\n",
       "      <td>2711</td>\n",
       "      <td>1970</td>\n",
       "      <td>0</td>\n",
       "      <td>3523</td>\n",
       "      <td>9291</td>\n",
       "      <td>6605</td>\n",
       "      <td>1</td>\n",
       "      <td>0.069637</td>\n",
       "      <td>1.039860</td>\n",
       "      <td>0.996864</td>\n",
       "      <td>1.08470</td>\n",
       "      <td>0.410999</td>\n",
       "      <td>0.420645</td>\n",
       "      <td>0.418472</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>226164963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172262</th>\n",
       "      <td>rs4657733</td>\n",
       "      <td>382</td>\n",
       "      <td>2167</td>\n",
       "      <td>3097</td>\n",
       "      <td>0</td>\n",
       "      <td>1345</td>\n",
       "      <td>7566</td>\n",
       "      <td>10509</td>\n",
       "      <td>0</td>\n",
       "      <td>0.339431</td>\n",
       "      <td>1.023560</td>\n",
       "      <td>0.975807</td>\n",
       "      <td>1.07365</td>\n",
       "      <td>0.259564</td>\n",
       "      <td>0.264058</td>\n",
       "      <td>0.263046</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>166353058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>172254</th>\n",
       "      <td>rs1337444</td>\n",
       "      <td>206</td>\n",
       "      <td>1689</td>\n",
       "      <td>3751</td>\n",
       "      <td>0</td>\n",
       "      <td>644</td>\n",
       "      <td>5785</td>\n",
       "      <td>12991</td>\n",
       "      <td>0</td>\n",
       "      <td>0.339413</td>\n",
       "      <td>0.974084</td>\n",
       "      <td>0.922996</td>\n",
       "      <td>1.02800</td>\n",
       "      <td>0.186061</td>\n",
       "      <td>0.182106</td>\n",
       "      <td>0.182997</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>163856159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352610</th>\n",
       "      <td>rs761430</td>\n",
       "      <td>38</td>\n",
       "      <td>854</td>\n",
       "      <td>4754</td>\n",
       "      <td>0</td>\n",
       "      <td>148</td>\n",
       "      <td>2943</td>\n",
       "      <td>16327</td>\n",
       "      <td>2</td>\n",
       "      <td>0.724795</td>\n",
       "      <td>1.013720</td>\n",
       "      <td>0.939649</td>\n",
       "      <td>1.09364</td>\n",
       "      <td>0.082359</td>\n",
       "      <td>0.083402</td>\n",
       "      <td>0.083167</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>29551361</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439591</th>\n",
       "      <td>rs4819804</td>\n",
       "      <td>383</td>\n",
       "      <td>2192</td>\n",
       "      <td>3071</td>\n",
       "      <td>0</td>\n",
       "      <td>1380</td>\n",
       "      <td>7394</td>\n",
       "      <td>10645</td>\n",
       "      <td>1</td>\n",
       "      <td>0.913927</td>\n",
       "      <td>0.997388</td>\n",
       "      <td>0.951188</td>\n",
       "      <td>1.04583</td>\n",
       "      <td>0.261955</td>\n",
       "      <td>0.261445</td>\n",
       "      <td>0.261560</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>17912378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214011</th>\n",
       "      <td>rs2041607</td>\n",
       "      <td>1087</td>\n",
       "      <td>2756</td>\n",
       "      <td>1803</td>\n",
       "      <td>0</td>\n",
       "      <td>3845</td>\n",
       "      <td>9429</td>\n",
       "      <td>6143</td>\n",
       "      <td>3</td>\n",
       "      <td>0.428184</td>\n",
       "      <td>1.017110</td>\n",
       "      <td>0.975310</td>\n",
       "      <td>1.06070</td>\n",
       "      <td>0.436592</td>\n",
       "      <td>0.440825</td>\n",
       "      <td>0.439872</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>15830515</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214030</th>\n",
       "      <td>rs1034420</td>\n",
       "      <td>87</td>\n",
       "      <td>1182</td>\n",
       "      <td>4377</td>\n",
       "      <td>0</td>\n",
       "      <td>261</td>\n",
       "      <td>4036</td>\n",
       "      <td>15123</td>\n",
       "      <td>0</td>\n",
       "      <td>0.428221</td>\n",
       "      <td>0.974218</td>\n",
       "      <td>0.913254</td>\n",
       "      <td>1.03925</td>\n",
       "      <td>0.120085</td>\n",
       "      <td>0.117353</td>\n",
       "      <td>0.117969</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>33660152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>439408</th>\n",
       "      <td>rs5760680</td>\n",
       "      <td>34</td>\n",
       "      <td>783</td>\n",
       "      <td>4825</td>\n",
       "      <td>4</td>\n",
       "      <td>104</td>\n",
       "      <td>2709</td>\n",
       "      <td>16605</td>\n",
       "      <td>2</td>\n",
       "      <td>0.913555</td>\n",
       "      <td>0.995607</td>\n",
       "      <td>0.919526</td>\n",
       "      <td>1.07798</td>\n",
       "      <td>0.075416</td>\n",
       "      <td>0.075111</td>\n",
       "      <td>0.075180</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>23541793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201871</th>\n",
       "      <td>rs5747297</td>\n",
       "      <td>4150</td>\n",
       "      <td>1381</td>\n",
       "      <td>107</td>\n",
       "      <td>8</td>\n",
       "      <td>14197</td>\n",
       "      <td>4775</td>\n",
       "      <td>416</td>\n",
       "      <td>32</td>\n",
       "      <td>0.402097</td>\n",
       "      <td>0.974674</td>\n",
       "      <td>0.917908</td>\n",
       "      <td>1.03495</td>\n",
       "      <td>0.141451</td>\n",
       "      <td>0.144600</td>\n",
       "      <td>0.143890</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>16495392</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>478426 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "310870  rs10159099    3552    1849     245       0   12268    6337     814   \n",
       "38939    rs4518905     965    2711    1970       0    3523    9291    6605   \n",
       "172262   rs4657733     382    2167    3097       0    1345    7566   10509   \n",
       "172254   rs1337444     206    1689    3751       0     644    5785   12991   \n",
       "352610    rs761430      38     854    4754       0     148    2943   16327   \n",
       "...            ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "439591   rs4819804     383    2192    3071       0    1380    7394   10645   \n",
       "214011   rs2041607    1087    2756    1803       0    3845    9429    6143   \n",
       "214030   rs1034420      87    1182    4377       0     261    4036   15123   \n",
       "439408   rs5760680      34     783    4825       4     104    2709   16605   \n",
       "201871   rs5747297    4150    1381     107       8   14197    4775     416   \n",
       "\n",
       "        ctrl??         P  OR_logistic  OR_lower  OR_upper  maf_case  maf_ctrl  \\\n",
       "310870       1  0.634300     1.012640  0.961567   1.06642  0.207138  0.205083   \n",
       "38939        1  0.069637     1.039860  0.996864   1.08470  0.410999  0.420645   \n",
       "172262       0  0.339431     1.023560  0.975807   1.07365  0.259564  0.264058   \n",
       "172254       0  0.339413     0.974084  0.922996   1.02800  0.186061  0.182106   \n",
       "352610       2  0.724795     1.013720  0.939649   1.09364  0.082359  0.083402   \n",
       "...        ...       ...          ...       ...       ...       ...       ...   \n",
       "439591       1  0.913927     0.997388  0.951188   1.04583  0.261955  0.261445   \n",
       "214011       3  0.428184     1.017110  0.975310   1.06070  0.436592  0.440825   \n",
       "214030       0  0.428221     0.974218  0.913254   1.03925  0.120085  0.117353   \n",
       "439408       2  0.913555     0.995607  0.919526   1.07798  0.075416  0.075111   \n",
       "201871      32  0.402097     0.974674  0.917908   1.03495  0.141451  0.144600   \n",
       "\n",
       "        annotation alleleA alleleB  CHR         BP  \n",
       "310870    0.205546       T       C    1   56573324  \n",
       "38939     0.418472       A       G    1  226164963  \n",
       "172262    0.263046       A       G    1  166353058  \n",
       "172254    0.182997       A       G    1  163856159  \n",
       "352610    0.083167       T       G    1   29551361  \n",
       "...            ...     ...     ...  ...        ...  \n",
       "439591    0.261560       A       G   22   17912378  \n",
       "214011    0.439872       A       G   22   15830515  \n",
       "214030    0.117969       T       G   22   33660152  \n",
       "439408    0.075180       T       C   22   23541793  \n",
       "201871    0.143890       A       G   22   16495392  \n",
       "\n",
       "[478426 rows x 20 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#GWAS_rename_DF.loc[GWAS_rename_DF.str.startswith('rs', na=False)]\n",
    "duplicateRowsDF2_droped_rs = duplicateRowsDF2_droped[duplicateRowsDF2_droped['SNP'].str.match('rs', na = False)]\n",
    "#duplicateRowsDF2_droped_rs\n",
    "#sorting by chromosomes\n",
    "duplicateRowsDF2_droped_rs_sorted_DF=duplicateRowsDF2_droped_rs.sort_values(by=['CHR'])\n",
    "duplicateRowsDF2_droped_rs_sorted_DF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### retrieving SNPS without rs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>200519</td>\n",
       "      <td>637</td>\n",
       "      <td>2433</td>\n",
       "      <td>2576</td>\n",
       "      <td>0</td>\n",
       "      <td>2266</td>\n",
       "      <td>8679</td>\n",
       "      <td>8475</td>\n",
       "      <td>0</td>\n",
       "      <td>0.019560</td>\n",
       "      <td>1.054250</td>\n",
       "      <td>1.008510</td>\n",
       "      <td>1.10206</td>\n",
       "      <td>0.328286</td>\n",
       "      <td>0.340139</td>\n",
       "      <td>0.337469</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>33274012</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>500694</td>\n",
       "      <td>2845</td>\n",
       "      <td>2315</td>\n",
       "      <td>486</td>\n",
       "      <td>0</td>\n",
       "      <td>10075</td>\n",
       "      <td>7785</td>\n",
       "      <td>1555</td>\n",
       "      <td>5</td>\n",
       "      <td>0.029647</td>\n",
       "      <td>1.052470</td>\n",
       "      <td>1.005060</td>\n",
       "      <td>1.10212</td>\n",
       "      <td>0.291091</td>\n",
       "      <td>0.280582</td>\n",
       "      <td>0.282950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>16</td>\n",
       "      <td>86433796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>200521</td>\n",
       "      <td>1691</td>\n",
       "      <td>2772</td>\n",
       "      <td>1183</td>\n",
       "      <td>0</td>\n",
       "      <td>5566</td>\n",
       "      <td>9637</td>\n",
       "      <td>4217</td>\n",
       "      <td>0</td>\n",
       "      <td>0.054923</td>\n",
       "      <td>0.959727</td>\n",
       "      <td>0.920275</td>\n",
       "      <td>1.00087</td>\n",
       "      <td>0.455012</td>\n",
       "      <td>0.465268</td>\n",
       "      <td>0.462958</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>33281976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>200262</td>\n",
       "      <td>2643</td>\n",
       "      <td>2418</td>\n",
       "      <td>585</td>\n",
       "      <td>0</td>\n",
       "      <td>9302</td>\n",
       "      <td>8207</td>\n",
       "      <td>1910</td>\n",
       "      <td>1</td>\n",
       "      <td>0.104876</td>\n",
       "      <td>1.037770</td>\n",
       "      <td>0.992292</td>\n",
       "      <td>1.08534</td>\n",
       "      <td>0.317747</td>\n",
       "      <td>0.309671</td>\n",
       "      <td>0.311490</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>20866804</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>200124</td>\n",
       "      <td>2896</td>\n",
       "      <td>2293</td>\n",
       "      <td>457</td>\n",
       "      <td>0</td>\n",
       "      <td>9773</td>\n",
       "      <td>7956</td>\n",
       "      <td>1689</td>\n",
       "      <td>2</td>\n",
       "      <td>0.107534</td>\n",
       "      <td>0.962757</td>\n",
       "      <td>0.919267</td>\n",
       "      <td>1.00831</td>\n",
       "      <td>0.284006</td>\n",
       "      <td>0.291843</td>\n",
       "      <td>0.290077</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>7</td>\n",
       "      <td>150564216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>200260</td>\n",
       "      <td>1204</td>\n",
       "      <td>2800</td>\n",
       "      <td>1639</td>\n",
       "      <td>3</td>\n",
       "      <td>4335</td>\n",
       "      <td>9566</td>\n",
       "      <td>5508</td>\n",
       "      <td>11</td>\n",
       "      <td>0.120311</td>\n",
       "      <td>1.033730</td>\n",
       "      <td>0.991355</td>\n",
       "      <td>1.07792</td>\n",
       "      <td>0.461457</td>\n",
       "      <td>0.469782</td>\n",
       "      <td>0.467907</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>20865406</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>500556</td>\n",
       "      <td>596</td>\n",
       "      <td>2401</td>\n",
       "      <td>2644</td>\n",
       "      <td>5</td>\n",
       "      <td>2086</td>\n",
       "      <td>8489</td>\n",
       "      <td>8840</td>\n",
       "      <td>5</td>\n",
       "      <td>0.131119</td>\n",
       "      <td>1.035070</td>\n",
       "      <td>0.989771</td>\n",
       "      <td>1.08244</td>\n",
       "      <td>0.318472</td>\n",
       "      <td>0.326062</td>\n",
       "      <td>0.324353</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>17</td>\n",
       "      <td>17350246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>200070</td>\n",
       "      <td>1532</td>\n",
       "      <td>2791</td>\n",
       "      <td>1321</td>\n",
       "      <td>2</td>\n",
       "      <td>5152</td>\n",
       "      <td>9611</td>\n",
       "      <td>4656</td>\n",
       "      <td>1</td>\n",
       "      <td>0.270142</td>\n",
       "      <td>0.976788</td>\n",
       "      <td>0.936852</td>\n",
       "      <td>1.01843</td>\n",
       "      <td>0.481308</td>\n",
       "      <td>0.487229</td>\n",
       "      <td>0.485896</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>16</td>\n",
       "      <td>16174099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>400630</td>\n",
       "      <td>4362</td>\n",
       "      <td>1200</td>\n",
       "      <td>84</td>\n",
       "      <td>0</td>\n",
       "      <td>15144</td>\n",
       "      <td>3994</td>\n",
       "      <td>282</td>\n",
       "      <td>0</td>\n",
       "      <td>0.273082</td>\n",
       "      <td>1.036570</td>\n",
       "      <td>0.972083</td>\n",
       "      <td>1.10533</td>\n",
       "      <td>0.121148</td>\n",
       "      <td>0.117353</td>\n",
       "      <td>0.118208</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>11</td>\n",
       "      <td>118530141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>500715</td>\n",
       "      <td>3545</td>\n",
       "      <td>1867</td>\n",
       "      <td>234</td>\n",
       "      <td>0</td>\n",
       "      <td>12424</td>\n",
       "      <td>6117</td>\n",
       "      <td>875</td>\n",
       "      <td>4</td>\n",
       "      <td>0.334218</td>\n",
       "      <td>1.025640</td>\n",
       "      <td>0.974269</td>\n",
       "      <td>1.07972</td>\n",
       "      <td>0.206784</td>\n",
       "      <td>0.202591</td>\n",
       "      <td>0.203535</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>14</td>\n",
       "      <td>22723148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>200006</td>\n",
       "      <td>362</td>\n",
       "      <td>2122</td>\n",
       "      <td>3162</td>\n",
       "      <td>0</td>\n",
       "      <td>1223</td>\n",
       "      <td>7195</td>\n",
       "      <td>10998</td>\n",
       "      <td>4</td>\n",
       "      <td>0.417341</td>\n",
       "      <td>0.980266</td>\n",
       "      <td>0.934169</td>\n",
       "      <td>1.02864</td>\n",
       "      <td>0.252037</td>\n",
       "      <td>0.248275</td>\n",
       "      <td>0.249122</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>9</td>\n",
       "      <td>139046223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>401753</td>\n",
       "      <td>431</td>\n",
       "      <td>2223</td>\n",
       "      <td>2988</td>\n",
       "      <td>4</td>\n",
       "      <td>1468</td>\n",
       "      <td>7535</td>\n",
       "      <td>10398</td>\n",
       "      <td>19</td>\n",
       "      <td>0.459375</td>\n",
       "      <td>0.982499</td>\n",
       "      <td>0.937604</td>\n",
       "      <td>1.02954</td>\n",
       "      <td>0.273396</td>\n",
       "      <td>0.269857</td>\n",
       "      <td>0.270654</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>6</td>\n",
       "      <td>32913568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>500437</td>\n",
       "      <td>2784</td>\n",
       "      <td>2376</td>\n",
       "      <td>484</td>\n",
       "      <td>2</td>\n",
       "      <td>9538</td>\n",
       "      <td>8113</td>\n",
       "      <td>1766</td>\n",
       "      <td>3</td>\n",
       "      <td>0.459739</td>\n",
       "      <td>0.982863</td>\n",
       "      <td>0.938839</td>\n",
       "      <td>1.02895</td>\n",
       "      <td>0.296244</td>\n",
       "      <td>0.299866</td>\n",
       "      <td>0.299050</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>18330763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>400863</td>\n",
       "      <td>638</td>\n",
       "      <td>2553</td>\n",
       "      <td>2455</td>\n",
       "      <td>0</td>\n",
       "      <td>2203</td>\n",
       "      <td>8623</td>\n",
       "      <td>8592</td>\n",
       "      <td>2</td>\n",
       "      <td>0.476035</td>\n",
       "      <td>0.984033</td>\n",
       "      <td>0.941423</td>\n",
       "      <td>1.02857</td>\n",
       "      <td>0.339090</td>\n",
       "      <td>0.335488</td>\n",
       "      <td>0.336299</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>20938161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>402616</td>\n",
       "      <td>1925</td>\n",
       "      <td>2733</td>\n",
       "      <td>986</td>\n",
       "      <td>2</td>\n",
       "      <td>6561</td>\n",
       "      <td>9381</td>\n",
       "      <td>3474</td>\n",
       "      <td>4</td>\n",
       "      <td>0.486117</td>\n",
       "      <td>0.985068</td>\n",
       "      <td>0.944233</td>\n",
       "      <td>1.02767</td>\n",
       "      <td>0.416814</td>\n",
       "      <td>0.420504</td>\n",
       "      <td>0.419673</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>9</td>\n",
       "      <td>74709191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>200261</td>\n",
       "      <td>1209</td>\n",
       "      <td>2758</td>\n",
       "      <td>1679</td>\n",
       "      <td>0</td>\n",
       "      <td>4040</td>\n",
       "      <td>9580</td>\n",
       "      <td>5797</td>\n",
       "      <td>3</td>\n",
       "      <td>0.498048</td>\n",
       "      <td>0.985618</td>\n",
       "      <td>0.945171</td>\n",
       "      <td>1.02780</td>\n",
       "      <td>0.458378</td>\n",
       "      <td>0.454756</td>\n",
       "      <td>0.455572</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>20841103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>200091</td>\n",
       "      <td>242</td>\n",
       "      <td>1946</td>\n",
       "      <td>3458</td>\n",
       "      <td>0</td>\n",
       "      <td>912</td>\n",
       "      <td>6419</td>\n",
       "      <td>12088</td>\n",
       "      <td>1</td>\n",
       "      <td>0.500327</td>\n",
       "      <td>0.982636</td>\n",
       "      <td>0.933834</td>\n",
       "      <td>1.03399</td>\n",
       "      <td>0.215197</td>\n",
       "      <td>0.212241</td>\n",
       "      <td>0.212906</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>16</td>\n",
       "      <td>16188655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>200053</td>\n",
       "      <td>4512</td>\n",
       "      <td>1069</td>\n",
       "      <td>63</td>\n",
       "      <td>2</td>\n",
       "      <td>15464</td>\n",
       "      <td>3715</td>\n",
       "      <td>239</td>\n",
       "      <td>2</td>\n",
       "      <td>0.526678</td>\n",
       "      <td>0.978334</td>\n",
       "      <td>0.914188</td>\n",
       "      <td>1.04698</td>\n",
       "      <td>0.105865</td>\n",
       "      <td>0.107967</td>\n",
       "      <td>0.107493</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>219783289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>200240</td>\n",
       "      <td>4447</td>\n",
       "      <td>1106</td>\n",
       "      <td>90</td>\n",
       "      <td>3</td>\n",
       "      <td>15176</td>\n",
       "      <td>3975</td>\n",
       "      <td>267</td>\n",
       "      <td>2</td>\n",
       "      <td>0.529908</td>\n",
       "      <td>0.979207</td>\n",
       "      <td>0.917065</td>\n",
       "      <td>1.04556</td>\n",
       "      <td>0.113946</td>\n",
       "      <td>0.116104</td>\n",
       "      <td>0.115618</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>20850186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>200052</td>\n",
       "      <td>63</td>\n",
       "      <td>1069</td>\n",
       "      <td>4511</td>\n",
       "      <td>3</td>\n",
       "      <td>240</td>\n",
       "      <td>3713</td>\n",
       "      <td>15467</td>\n",
       "      <td>0</td>\n",
       "      <td>0.532673</td>\n",
       "      <td>1.021820</td>\n",
       "      <td>0.954834</td>\n",
       "      <td>1.09350</td>\n",
       "      <td>0.105883</td>\n",
       "      <td>0.107956</td>\n",
       "      <td>0.107489</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>219783037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>200258</td>\n",
       "      <td>2704</td>\n",
       "      <td>2407</td>\n",
       "      <td>532</td>\n",
       "      <td>3</td>\n",
       "      <td>9276</td>\n",
       "      <td>8228</td>\n",
       "      <td>1913</td>\n",
       "      <td>3</td>\n",
       "      <td>0.565974</td>\n",
       "      <td>0.986844</td>\n",
       "      <td>0.943211</td>\n",
       "      <td>1.03250</td>\n",
       "      <td>0.307549</td>\n",
       "      <td>0.310398</td>\n",
       "      <td>0.309757</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>20869344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>200047</td>\n",
       "      <td>4509</td>\n",
       "      <td>1075</td>\n",
       "      <td>61</td>\n",
       "      <td>1</td>\n",
       "      <td>15457</td>\n",
       "      <td>3713</td>\n",
       "      <td>238</td>\n",
       "      <td>12</td>\n",
       "      <td>0.567692</td>\n",
       "      <td>0.980420</td>\n",
       "      <td>0.916131</td>\n",
       "      <td>1.04922</td>\n",
       "      <td>0.106023</td>\n",
       "      <td>0.107919</td>\n",
       "      <td>0.107492</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>2</td>\n",
       "      <td>219793146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>200050</td>\n",
       "      <td>61</td>\n",
       "      <td>1076</td>\n",
       "      <td>4507</td>\n",
       "      <td>2</td>\n",
       "      <td>237</td>\n",
       "      <td>3721</td>\n",
       "      <td>15460</td>\n",
       "      <td>2</td>\n",
       "      <td>0.569451</td>\n",
       "      <td>1.019880</td>\n",
       "      <td>0.953006</td>\n",
       "      <td>1.09144</td>\n",
       "      <td>0.106130</td>\n",
       "      <td>0.108018</td>\n",
       "      <td>0.107593</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>2</td>\n",
       "      <td>219797929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>401070</td>\n",
       "      <td>3850</td>\n",
       "      <td>1636</td>\n",
       "      <td>160</td>\n",
       "      <td>0</td>\n",
       "      <td>13368</td>\n",
       "      <td>5458</td>\n",
       "      <td>590</td>\n",
       "      <td>4</td>\n",
       "      <td>0.572502</td>\n",
       "      <td>1.016060</td>\n",
       "      <td>0.961364</td>\n",
       "      <td>1.07387</td>\n",
       "      <td>0.173220</td>\n",
       "      <td>0.170941</td>\n",
       "      <td>0.171455</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>4</td>\n",
       "      <td>100552989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>200003</td>\n",
       "      <td>3715</td>\n",
       "      <td>1720</td>\n",
       "      <td>211</td>\n",
       "      <td>0</td>\n",
       "      <td>12831</td>\n",
       "      <td>5901</td>\n",
       "      <td>688</td>\n",
       "      <td>0</td>\n",
       "      <td>0.577046</td>\n",
       "      <td>1.015310</td>\n",
       "      <td>0.962511</td>\n",
       "      <td>1.07101</td>\n",
       "      <td>0.189692</td>\n",
       "      <td>0.187358</td>\n",
       "      <td>0.187884</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>9</td>\n",
       "      <td>139026180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>200199</td>\n",
       "      <td>1145</td>\n",
       "      <td>2788</td>\n",
       "      <td>1712</td>\n",
       "      <td>1</td>\n",
       "      <td>3995</td>\n",
       "      <td>9588</td>\n",
       "      <td>5835</td>\n",
       "      <td>2</td>\n",
       "      <td>0.593792</td>\n",
       "      <td>1.011510</td>\n",
       "      <td>0.969854</td>\n",
       "      <td>1.05495</td>\n",
       "      <td>0.449779</td>\n",
       "      <td>0.452621</td>\n",
       "      <td>0.451981</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>40847933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>401232</td>\n",
       "      <td>4033</td>\n",
       "      <td>1461</td>\n",
       "      <td>152</td>\n",
       "      <td>0</td>\n",
       "      <td>13744</td>\n",
       "      <td>5198</td>\n",
       "      <td>476</td>\n",
       "      <td>2</td>\n",
       "      <td>0.598663</td>\n",
       "      <td>0.984649</td>\n",
       "      <td>0.929527</td>\n",
       "      <td>1.04304</td>\n",
       "      <td>0.156305</td>\n",
       "      <td>0.158358</td>\n",
       "      <td>0.157896</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>163890075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>400826</td>\n",
       "      <td>2751</td>\n",
       "      <td>2358</td>\n",
       "      <td>535</td>\n",
       "      <td>2</td>\n",
       "      <td>9505</td>\n",
       "      <td>8126</td>\n",
       "      <td>1787</td>\n",
       "      <td>2</td>\n",
       "      <td>0.623502</td>\n",
       "      <td>1.011440</td>\n",
       "      <td>0.966521</td>\n",
       "      <td>1.05845</td>\n",
       "      <td>0.303685</td>\n",
       "      <td>0.301267</td>\n",
       "      <td>0.301812</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>12</td>\n",
       "      <td>20905215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>401327</td>\n",
       "      <td>1438</td>\n",
       "      <td>2846</td>\n",
       "      <td>1361</td>\n",
       "      <td>1</td>\n",
       "      <td>5062</td>\n",
       "      <td>9651</td>\n",
       "      <td>4706</td>\n",
       "      <td>1</td>\n",
       "      <td>0.661124</td>\n",
       "      <td>1.009410</td>\n",
       "      <td>0.968021</td>\n",
       "      <td>1.05256</td>\n",
       "      <td>0.493180</td>\n",
       "      <td>0.490834</td>\n",
       "      <td>0.491362</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>140007520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>401326</td>\n",
       "      <td>1438</td>\n",
       "      <td>2846</td>\n",
       "      <td>1361</td>\n",
       "      <td>1</td>\n",
       "      <td>5062</td>\n",
       "      <td>9650</td>\n",
       "      <td>4707</td>\n",
       "      <td>1</td>\n",
       "      <td>0.664628</td>\n",
       "      <td>1.009300</td>\n",
       "      <td>0.967922</td>\n",
       "      <td>1.05245</td>\n",
       "      <td>0.493180</td>\n",
       "      <td>0.490859</td>\n",
       "      <td>0.491382</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>140007608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>401754</td>\n",
       "      <td>1886</td>\n",
       "      <td>2688</td>\n",
       "      <td>1071</td>\n",
       "      <td>1</td>\n",
       "      <td>6392</td>\n",
       "      <td>9372</td>\n",
       "      <td>3653</td>\n",
       "      <td>3</td>\n",
       "      <td>0.756322</td>\n",
       "      <td>0.993376</td>\n",
       "      <td>0.952537</td>\n",
       "      <td>1.03596</td>\n",
       "      <td>0.427812</td>\n",
       "      <td>0.429469</td>\n",
       "      <td>0.429096</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>6</td>\n",
       "      <td>32913147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>500770</td>\n",
       "      <td>2012</td>\n",
       "      <td>2702</td>\n",
       "      <td>931</td>\n",
       "      <td>1</td>\n",
       "      <td>6848</td>\n",
       "      <td>9384</td>\n",
       "      <td>3185</td>\n",
       "      <td>3</td>\n",
       "      <td>0.786169</td>\n",
       "      <td>0.994106</td>\n",
       "      <td>0.952547</td>\n",
       "      <td>1.03748</td>\n",
       "      <td>0.404252</td>\n",
       "      <td>0.405675</td>\n",
       "      <td>0.405355</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>11</td>\n",
       "      <td>74543755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>200087</td>\n",
       "      <td>3949</td>\n",
       "      <td>1541</td>\n",
       "      <td>151</td>\n",
       "      <td>5</td>\n",
       "      <td>13617</td>\n",
       "      <td>5252</td>\n",
       "      <td>523</td>\n",
       "      <td>28</td>\n",
       "      <td>0.806018</td>\n",
       "      <td>1.007120</td>\n",
       "      <td>0.951664</td>\n",
       "      <td>1.06581</td>\n",
       "      <td>0.163358</td>\n",
       "      <td>0.162387</td>\n",
       "      <td>0.162605</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>16</td>\n",
       "      <td>16153665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>200078</td>\n",
       "      <td>311</td>\n",
       "      <td>1930</td>\n",
       "      <td>3405</td>\n",
       "      <td>0</td>\n",
       "      <td>1013</td>\n",
       "      <td>6791</td>\n",
       "      <td>11610</td>\n",
       "      <td>6</td>\n",
       "      <td>0.810526</td>\n",
       "      <td>1.006120</td>\n",
       "      <td>0.957158</td>\n",
       "      <td>1.05759</td>\n",
       "      <td>0.226001</td>\n",
       "      <td>0.227078</td>\n",
       "      <td>0.226836</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>16</td>\n",
       "      <td>16194115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>200096</td>\n",
       "      <td>1686</td>\n",
       "      <td>2791</td>\n",
       "      <td>1169</td>\n",
       "      <td>0</td>\n",
       "      <td>5855</td>\n",
       "      <td>9524</td>\n",
       "      <td>4039</td>\n",
       "      <td>2</td>\n",
       "      <td>0.855114</td>\n",
       "      <td>1.003910</td>\n",
       "      <td>0.962711</td>\n",
       "      <td>1.04688</td>\n",
       "      <td>0.454215</td>\n",
       "      <td>0.453239</td>\n",
       "      <td>0.453459</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>16</td>\n",
       "      <td>16195546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>500717</td>\n",
       "      <td>4034</td>\n",
       "      <td>1489</td>\n",
       "      <td>123</td>\n",
       "      <td>0</td>\n",
       "      <td>13961</td>\n",
       "      <td>4971</td>\n",
       "      <td>488</td>\n",
       "      <td>0</td>\n",
       "      <td>0.890248</td>\n",
       "      <td>1.004090</td>\n",
       "      <td>0.947575</td>\n",
       "      <td>1.06397</td>\n",
       "      <td>0.153649</td>\n",
       "      <td>0.153115</td>\n",
       "      <td>0.153235</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>14</td>\n",
       "      <td>22719874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>200623</td>\n",
       "      <td>2813</td>\n",
       "      <td>2340</td>\n",
       "      <td>493</td>\n",
       "      <td>0</td>\n",
       "      <td>9731</td>\n",
       "      <td>7922</td>\n",
       "      <td>1767</td>\n",
       "      <td>0</td>\n",
       "      <td>0.933674</td>\n",
       "      <td>0.998065</td>\n",
       "      <td>0.953562</td>\n",
       "      <td>1.04464</td>\n",
       "      <td>0.294545</td>\n",
       "      <td>0.294954</td>\n",
       "      <td>0.294862</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>4</td>\n",
       "      <td>70350714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>401050</td>\n",
       "      <td>319</td>\n",
       "      <td>2036</td>\n",
       "      <td>3291</td>\n",
       "      <td>0</td>\n",
       "      <td>1105</td>\n",
       "      <td>6977</td>\n",
       "      <td>11337</td>\n",
       "      <td>1</td>\n",
       "      <td>0.954800</td>\n",
       "      <td>0.998579</td>\n",
       "      <td>0.950654</td>\n",
       "      <td>1.04892</td>\n",
       "      <td>0.236805</td>\n",
       "      <td>0.236547</td>\n",
       "      <td>0.236605</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4</td>\n",
       "      <td>100575480</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  ctrl??  \\\n",
       "0   200519     637    2433    2576       0    2266    8679    8475       0   \n",
       "1   500694    2845    2315     486       0   10075    7785    1555       5   \n",
       "2   200521    1691    2772    1183       0    5566    9637    4217       0   \n",
       "3   200262    2643    2418     585       0    9302    8207    1910       1   \n",
       "4   200124    2896    2293     457       0    9773    7956    1689       2   \n",
       "5   200260    1204    2800    1639       3    4335    9566    5508      11   \n",
       "6   500556     596    2401    2644       5    2086    8489    8840       5   \n",
       "7   200070    1532    2791    1321       2    5152    9611    4656       1   \n",
       "8   400630    4362    1200      84       0   15144    3994     282       0   \n",
       "9   500715    3545    1867     234       0   12424    6117     875       4   \n",
       "10  200006     362    2122    3162       0    1223    7195   10998       4   \n",
       "11  401753     431    2223    2988       4    1468    7535   10398      19   \n",
       "12  500437    2784    2376     484       2    9538    8113    1766       3   \n",
       "13  400863     638    2553    2455       0    2203    8623    8592       2   \n",
       "14  402616    1925    2733     986       2    6561    9381    3474       4   \n",
       "15  200261    1209    2758    1679       0    4040    9580    5797       3   \n",
       "16  200091     242    1946    3458       0     912    6419   12088       1   \n",
       "17  200053    4512    1069      63       2   15464    3715     239       2   \n",
       "18  200240    4447    1106      90       3   15176    3975     267       2   \n",
       "19  200052      63    1069    4511       3     240    3713   15467       0   \n",
       "20  200258    2704    2407     532       3    9276    8228    1913       3   \n",
       "21  200047    4509    1075      61       1   15457    3713     238      12   \n",
       "22  200050      61    1076    4507       2     237    3721   15460       2   \n",
       "23  401070    3850    1636     160       0   13368    5458     590       4   \n",
       "24  200003    3715    1720     211       0   12831    5901     688       0   \n",
       "25  200199    1145    2788    1712       1    3995    9588    5835       2   \n",
       "26  401232    4033    1461     152       0   13744    5198     476       2   \n",
       "27  400826    2751    2358     535       2    9505    8126    1787       2   \n",
       "28  401327    1438    2846    1361       1    5062    9651    4706       1   \n",
       "29  401326    1438    2846    1361       1    5062    9650    4707       1   \n",
       "30  401754    1886    2688    1071       1    6392    9372    3653       3   \n",
       "31  500770    2012    2702     931       1    6848    9384    3185       3   \n",
       "32  200087    3949    1541     151       5   13617    5252     523      28   \n",
       "33  200078     311    1930    3405       0    1013    6791   11610       6   \n",
       "34  200096    1686    2791    1169       0    5855    9524    4039       2   \n",
       "35  500717    4034    1489     123       0   13961    4971     488       0   \n",
       "36  200623    2813    2340     493       0    9731    7922    1767       0   \n",
       "37  401050     319    2036    3291       0    1105    6977   11337       1   \n",
       "\n",
       "           P  OR_logistic  OR_lower  OR_upper  maf_case  maf_ctrl  annotation  \\\n",
       "0   0.019560     1.054250  1.008510   1.10206  0.328286  0.340139    0.337469   \n",
       "1   0.029647     1.052470  1.005060   1.10212  0.291091  0.280582    0.282950   \n",
       "2   0.054923     0.959727  0.920275   1.00087  0.455012  0.465268    0.462958   \n",
       "3   0.104876     1.037770  0.992292   1.08534  0.317747  0.309671    0.311490   \n",
       "4   0.107534     0.962757  0.919267   1.00831  0.284006  0.291843    0.290077   \n",
       "5   0.120311     1.033730  0.991355   1.07792  0.461457  0.469782    0.467907   \n",
       "6   0.131119     1.035070  0.989771   1.08244  0.318472  0.326062    0.324353   \n",
       "7   0.270142     0.976788  0.936852   1.01843  0.481308  0.487229    0.485896   \n",
       "8   0.273082     1.036570  0.972083   1.10533  0.121148  0.117353    0.118208   \n",
       "9   0.334218     1.025640  0.974269   1.07972  0.206784  0.202591    0.203535   \n",
       "10  0.417341     0.980266  0.934169   1.02864  0.252037  0.248275    0.249122   \n",
       "11  0.459375     0.982499  0.937604   1.02954  0.273396  0.269857    0.270654   \n",
       "12  0.459739     0.982863  0.938839   1.02895  0.296244  0.299866    0.299050   \n",
       "13  0.476035     0.984033  0.941423   1.02857  0.339090  0.335488    0.336299   \n",
       "14  0.486117     0.985068  0.944233   1.02767  0.416814  0.420504    0.419673   \n",
       "15  0.498048     0.985618  0.945171   1.02780  0.458378  0.454756    0.455572   \n",
       "16  0.500327     0.982636  0.933834   1.03399  0.215197  0.212241    0.212906   \n",
       "17  0.526678     0.978334  0.914188   1.04698  0.105865  0.107967    0.107493   \n",
       "18  0.529908     0.979207  0.917065   1.04556  0.113946  0.116104    0.115618   \n",
       "19  0.532673     1.021820  0.954834   1.09350  0.105883  0.107956    0.107489   \n",
       "20  0.565974     0.986844  0.943211   1.03250  0.307549  0.310398    0.309757   \n",
       "21  0.567692     0.980420  0.916131   1.04922  0.106023  0.107919    0.107492   \n",
       "22  0.569451     1.019880  0.953006   1.09144  0.106130  0.108018    0.107593   \n",
       "23  0.572502     1.016060  0.961364   1.07387  0.173220  0.170941    0.171455   \n",
       "24  0.577046     1.015310  0.962511   1.07101  0.189692  0.187358    0.187884   \n",
       "25  0.593792     1.011510  0.969854   1.05495  0.449779  0.452621    0.451981   \n",
       "26  0.598663     0.984649  0.929527   1.04304  0.156305  0.158358    0.157896   \n",
       "27  0.623502     1.011440  0.966521   1.05845  0.303685  0.301267    0.301812   \n",
       "28  0.661124     1.009410  0.968021   1.05256  0.493180  0.490834    0.491362   \n",
       "29  0.664628     1.009300  0.967922   1.05245  0.493180  0.490859    0.491382   \n",
       "30  0.756322     0.993376  0.952537   1.03596  0.427812  0.429469    0.429096   \n",
       "31  0.786169     0.994106  0.952547   1.03748  0.404252  0.405675    0.405355   \n",
       "32  0.806018     1.007120  0.951664   1.06581  0.163358  0.162387    0.162605   \n",
       "33  0.810526     1.006120  0.957158   1.05759  0.226001  0.227078    0.226836   \n",
       "34  0.855114     1.003910  0.962711   1.04688  0.454215  0.453239    0.453459   \n",
       "35  0.890248     1.004090  0.947575   1.06397  0.153649  0.153115    0.153235   \n",
       "36  0.933674     0.998065  0.953562   1.04464  0.294545  0.294954    0.294862   \n",
       "37  0.954800     0.998579  0.950654   1.04892  0.236805  0.236547    0.236605   \n",
       "\n",
       "   alleleA alleleB  CHR         BP  \n",
       "0        T       C    6   33274012  \n",
       "1        T       C   16   86433796  \n",
       "2        T       C    6   33281976  \n",
       "3        T       C    1   20866804  \n",
       "4        T       C    7  150564216  \n",
       "5        G       C    1   20865406  \n",
       "6        A       G   17   17350246  \n",
       "7        G       C   16   16174099  \n",
       "8        A       G   11  118530141  \n",
       "9        A       C   14   22723148  \n",
       "10       A       G    9  139046223  \n",
       "11       G       C    6   32913568  \n",
       "12       A       C   22   18330763  \n",
       "13       A       C   12   20938161  \n",
       "14       G       C    9   74709191  \n",
       "15       A       G    1   20841103  \n",
       "16       A       C   16   16188655  \n",
       "17       T       C    2  219783289  \n",
       "18       C       G    1   20850186  \n",
       "19       T       A    2  219783037  \n",
       "20       A       C    1   20869344  \n",
       "21       T       C    2  219793146  \n",
       "22       C       G    2  219797929  \n",
       "23       G       C    4  100552989  \n",
       "24       T       C    9  139026180  \n",
       "25       G       C   22   40847933  \n",
       "26       A       G    1  163890075  \n",
       "27       T       C   12   20905215  \n",
       "28       T       C    5  140007520  \n",
       "29       T       C    5  140007608  \n",
       "30       A       G    6   32913147  \n",
       "31       T       C   11   74543755  \n",
       "32       T       G   16   16153665  \n",
       "33       C       G   16   16194115  \n",
       "34       A       C   16   16195546  \n",
       "35       T       G   14   22719874  \n",
       "36       T       C    4   70350714  \n",
       "37       A       G    4  100575480  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# That dosent strats with rs\n",
    "#df[~df['col'].str.startswith('t')]\n",
    "#GWAS_sorted_DF_Not_rs = GWAS_sorted_DF[~GWAS_sorted_DF['SNP'].str.match('rs')]\n",
    "duplicateRowsDF2_droped_sorted_DF_Not_rs = duplicateRowsDF2_droped[~duplicateRowsDF2_droped['SNP'].str.startswith('rs',na = False)].reset_index(drop = True)\n",
    "duplicateRowsDF2_droped_sorted_DF_Not_rs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sample/Trim the Datasets to be used"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4970420</td>\n",
       "      <td>3640</td>\n",
       "      <td>1790</td>\n",
       "      <td>212</td>\n",
       "      <td>4</td>\n",
       "      <td>12501</td>\n",
       "      <td>6165</td>\n",
       "      <td>732</td>\n",
       "      <td>22</td>\n",
       "      <td>0.917874</td>\n",
       "      <td>0.997219</td>\n",
       "      <td>0.945806</td>\n",
       "      <td>1.05143</td>\n",
       "      <td>0.196207</td>\n",
       "      <td>0.196644</td>\n",
       "      <td>0.196546</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>1096336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs13303016</td>\n",
       "      <td>3472</td>\n",
       "      <td>1925</td>\n",
       "      <td>247</td>\n",
       "      <td>2</td>\n",
       "      <td>11988</td>\n",
       "      <td>6577</td>\n",
       "      <td>850</td>\n",
       "      <td>5</td>\n",
       "      <td>0.793916</td>\n",
       "      <td>1.006870</td>\n",
       "      <td>0.956453</td>\n",
       "      <td>1.05995</td>\n",
       "      <td>0.214298</td>\n",
       "      <td>0.213160</td>\n",
       "      <td>0.213416</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1936451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2254669</td>\n",
       "      <td>26</td>\n",
       "      <td>664</td>\n",
       "      <td>4956</td>\n",
       "      <td>0</td>\n",
       "      <td>76</td>\n",
       "      <td>2403</td>\n",
       "      <td>16941</td>\n",
       "      <td>0</td>\n",
       "      <td>0.367653</td>\n",
       "      <td>1.040220</td>\n",
       "      <td>0.954697</td>\n",
       "      <td>1.13340</td>\n",
       "      <td>0.063408</td>\n",
       "      <td>0.065783</td>\n",
       "      <td>0.065248</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1995968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs424079</td>\n",
       "      <td>3536</td>\n",
       "      <td>1861</td>\n",
       "      <td>245</td>\n",
       "      <td>4</td>\n",
       "      <td>11987</td>\n",
       "      <td>6479</td>\n",
       "      <td>927</td>\n",
       "      <td>27</td>\n",
       "      <td>0.139649</td>\n",
       "      <td>0.962079</td>\n",
       "      <td>0.913970</td>\n",
       "      <td>1.01272</td>\n",
       "      <td>0.208348</td>\n",
       "      <td>0.214846</td>\n",
       "      <td>0.213381</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>2061200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7512482</td>\n",
       "      <td>4637</td>\n",
       "      <td>951</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>15912</td>\n",
       "      <td>3322</td>\n",
       "      <td>185</td>\n",
       "      <td>1</td>\n",
       "      <td>0.856273</td>\n",
       "      <td>0.993432</td>\n",
       "      <td>0.925055</td>\n",
       "      <td>1.06686</td>\n",
       "      <td>0.094492</td>\n",
       "      <td>0.095061</td>\n",
       "      <td>0.094933</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>2136826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47838</th>\n",
       "      <td>rs11703226</td>\n",
       "      <td>2494</td>\n",
       "      <td>2543</td>\n",
       "      <td>603</td>\n",
       "      <td>6</td>\n",
       "      <td>8787</td>\n",
       "      <td>8502</td>\n",
       "      <td>2098</td>\n",
       "      <td>33</td>\n",
       "      <td>0.332329</td>\n",
       "      <td>1.022280</td>\n",
       "      <td>0.977734</td>\n",
       "      <td>1.06886</td>\n",
       "      <td>0.332358</td>\n",
       "      <td>0.327487</td>\n",
       "      <td>0.328585</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49000551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47839</th>\n",
       "      <td>rs5771242</td>\n",
       "      <td>208</td>\n",
       "      <td>1644</td>\n",
       "      <td>3793</td>\n",
       "      <td>1</td>\n",
       "      <td>632</td>\n",
       "      <td>5913</td>\n",
       "      <td>12869</td>\n",
       "      <td>6</td>\n",
       "      <td>0.565587</td>\n",
       "      <td>1.016040</td>\n",
       "      <td>0.962358</td>\n",
       "      <td>1.07271</td>\n",
       "      <td>0.182462</td>\n",
       "      <td>0.184841</td>\n",
       "      <td>0.184305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>49003996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47840</th>\n",
       "      <td>rs131816</td>\n",
       "      <td>5172</td>\n",
       "      <td>461</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>17706</td>\n",
       "      <td>1657</td>\n",
       "      <td>52</td>\n",
       "      <td>5</td>\n",
       "      <td>0.318233</td>\n",
       "      <td>0.949418</td>\n",
       "      <td>0.857404</td>\n",
       "      <td>1.05131</td>\n",
       "      <td>0.043128</td>\n",
       "      <td>0.045352</td>\n",
       "      <td>0.044851</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49304328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47841</th>\n",
       "      <td>rs470119</td>\n",
       "      <td>362</td>\n",
       "      <td>2173</td>\n",
       "      <td>3111</td>\n",
       "      <td>0</td>\n",
       "      <td>1359</td>\n",
       "      <td>7428</td>\n",
       "      <td>10632</td>\n",
       "      <td>1</td>\n",
       "      <td>0.319039</td>\n",
       "      <td>1.024600</td>\n",
       "      <td>0.976775</td>\n",
       "      <td>1.07476</td>\n",
       "      <td>0.256553</td>\n",
       "      <td>0.261239</td>\n",
       "      <td>0.260184</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49313780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47842</th>\n",
       "      <td>rs9616802</td>\n",
       "      <td>1559</td>\n",
       "      <td>2792</td>\n",
       "      <td>1293</td>\n",
       "      <td>2</td>\n",
       "      <td>5279</td>\n",
       "      <td>9570</td>\n",
       "      <td>4570</td>\n",
       "      <td>1</td>\n",
       "      <td>0.323190</td>\n",
       "      <td>0.979196</td>\n",
       "      <td>0.939195</td>\n",
       "      <td>1.02090</td>\n",
       "      <td>0.476435</td>\n",
       "      <td>0.481745</td>\n",
       "      <td>0.480549</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>49419027</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47843 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0       rs4970420    3640    1790     212       4   12501    6165     732   \n",
       "1      rs13303016    3472    1925     247       2   11988    6577     850   \n",
       "2       rs2254669      26     664    4956       0      76    2403   16941   \n",
       "3        rs424079    3536    1861     245       4   11987    6479     927   \n",
       "4       rs7512482    4637     951      58       0   15912    3322     185   \n",
       "...           ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "47838  rs11703226    2494    2543     603       6    8787    8502    2098   \n",
       "47839   rs5771242     208    1644    3793       1     632    5913   12869   \n",
       "47840    rs131816    5172     461      13       0   17706    1657      52   \n",
       "47841    rs470119     362    2173    3111       0    1359    7428   10632   \n",
       "47842   rs9616802    1559    2792    1293       2    5279    9570    4570   \n",
       "\n",
       "       ctrl??         P  OR_logistic  OR_lower  OR_upper  maf_case  maf_ctrl  \\\n",
       "0          22  0.917874     0.997219  0.945806   1.05143  0.196207  0.196644   \n",
       "1           5  0.793916     1.006870  0.956453   1.05995  0.214298  0.213160   \n",
       "2           0  0.367653     1.040220  0.954697   1.13340  0.063408  0.065783   \n",
       "3          27  0.139649     0.962079  0.913970   1.01272  0.208348  0.214846   \n",
       "4           1  0.856273     0.993432  0.925055   1.06686  0.094492  0.095061   \n",
       "...       ...       ...          ...       ...       ...       ...       ...   \n",
       "47838      33  0.332329     1.022280  0.977734   1.06886  0.332358  0.327487   \n",
       "47839       6  0.565587     1.016040  0.962358   1.07271  0.182462  0.184841   \n",
       "47840       5  0.318233     0.949418  0.857404   1.05131  0.043128  0.045352   \n",
       "47841       1  0.319039     1.024600  0.976775   1.07476  0.256553  0.261239   \n",
       "47842       1  0.323190     0.979196  0.939195   1.02090  0.476435  0.481745   \n",
       "\n",
       "       annotation alleleA alleleB  CHR        BP  \n",
       "0        0.196546       T       C    1   1096336  \n",
       "1        0.213416       A       G    1   1936451  \n",
       "2        0.065248       A       G    1   1995968  \n",
       "3        0.213381       A       C    1   2061200  \n",
       "4        0.094933       T       C    1   2136826  \n",
       "...           ...     ...     ...  ...       ...  \n",
       "47838    0.328585       T       C   22  49000551  \n",
       "47839    0.184305       A       G   22  49003996  \n",
       "47840    0.044851       T       C   22  49304328  \n",
       "47841    0.260184       T       C   22  49313780  \n",
       "47842    0.480549       A       G   22  49419027  \n",
       "\n",
       "[47843 rows x 20 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DATASET_smaple = duplicateRowsDF2_droped_rs_sorted_DF.sample(n=None, frac=0.1, replace=False, weights=None, random_state=1, axis=0)\n",
    "#DATASET_smaple\n",
    "SampleData = DATASET_smaple.groupby(['CHR','BP']).apply(lambda u:u[:])#.reset_index(drop =True) #.head(30000))\n",
    "#SampleData1 = DATASET_smaple.groupby(['CHR','BP']).apply(lambda u:u)\n",
    "SampleData1 = SampleData.droplevel(['BP','CHR']).reset_index(drop =True)\n",
    "#SampleData1 = SampleData.sort_values(by =\"CHR\").reset_index(drop =True)\n",
    "# SampleData1\n",
    "# SampleData1 = SampleData.droplevel('CHR').reset_index(drop =True)\n",
    "# SampleData1\n",
    "SampleData1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "groupbyDF = DATASET_smaple.groupby(['CHR'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHR</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "      <td>3759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "      <td>3928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "      <td>3355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "      <td>2940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "      <td>3065</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "      <td>3304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "      <td>2668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "      <td>2712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "      <td>2392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "      <td>2535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "      <td>2391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "      <td>2367</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "      <td>1822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "      <td>1625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "      <td>1425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "      <td>1437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "      <td>1236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "      <td>1443</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "      <td>843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "      <td>1219</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "      <td>699</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "      <td>678</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  ctrl??  \\\n",
       "CHR                                                                         \n",
       "1    3759    3759    3759    3759    3759    3759    3759    3759    3759   \n",
       "2    3928    3928    3928    3928    3928    3928    3928    3928    3928   \n",
       "3    3355    3355    3355    3355    3355    3355    3355    3355    3355   \n",
       "4    2940    2940    2940    2940    2940    2940    2940    2940    2940   \n",
       "5    3065    3065    3065    3065    3065    3065    3065    3065    3065   \n",
       "6    3304    3304    3304    3304    3304    3304    3304    3304    3304   \n",
       "7    2668    2668    2668    2668    2668    2668    2668    2668    2668   \n",
       "8    2712    2712    2712    2712    2712    2712    2712    2712    2712   \n",
       "9    2392    2392    2392    2392    2392    2392    2392    2392    2392   \n",
       "10   2535    2535    2535    2535    2535    2535    2535    2535    2535   \n",
       "11   2391    2391    2391    2391    2391    2391    2391    2391    2391   \n",
       "12   2367    2367    2367    2367    2367    2367    2367    2367    2367   \n",
       "13   1822    1822    1822    1822    1822    1822    1822    1822    1822   \n",
       "14   1625    1625    1625    1625    1625    1625    1625    1625    1625   \n",
       "15   1425    1425    1425    1425    1425    1425    1425    1425    1425   \n",
       "16   1437    1437    1437    1437    1437    1437    1437    1437    1437   \n",
       "17   1236    1236    1236    1236    1236    1236    1236    1236    1236   \n",
       "18   1443    1443    1443    1443    1443    1443    1443    1443    1443   \n",
       "19    843     843     843     843     843     843     843     843     843   \n",
       "20   1219    1219    1219    1219    1219    1219    1219    1219    1219   \n",
       "21    699     699     699     699     699     699     699     699     699   \n",
       "22    678     678     678     678     678     678     678     678     678   \n",
       "\n",
       "        P  OR_logistic  OR_lower  OR_upper  maf_case  maf_ctrl  annotation  \\\n",
       "CHR                                                                          \n",
       "1    3759         3759      3759      3759      3759      3759        3759   \n",
       "2    3928         3928      3928      3928      3928      3928        3928   \n",
       "3    3355         3355      3355      3355      3355      3355        3355   \n",
       "4    2940         2940      2940      2940      2940      2940        2940   \n",
       "5    3065         3065      3065      3065      3065      3065        3065   \n",
       "6    3304         3304      3304      3304      3304      3304        3304   \n",
       "7    2668         2668      2668      2668      2668      2668        2668   \n",
       "8    2712         2712      2712      2712      2712      2712        2712   \n",
       "9    2392         2392      2392      2392      2392      2392        2392   \n",
       "10   2535         2535      2535      2535      2535      2535        2535   \n",
       "11   2391         2391      2391      2391      2391      2391        2391   \n",
       "12   2367         2367      2367      2367      2367      2367        2367   \n",
       "13   1822         1822      1822      1822      1822      1822        1822   \n",
       "14   1625         1625      1625      1625      1625      1625        1625   \n",
       "15   1425         1425      1425      1425      1425      1425        1425   \n",
       "16   1437         1437      1437      1437      1437      1437        1437   \n",
       "17   1236         1236      1236      1236      1236      1236        1236   \n",
       "18   1443         1443      1443      1443      1443      1443        1443   \n",
       "19    843          843       843       843       843       843         843   \n",
       "20   1219         1219      1219      1219      1219      1219        1219   \n",
       "21    699          699       699       699       699       699         699   \n",
       "22    678          678       678       678       678       678         678   \n",
       "\n",
       "     alleleA  alleleB    BP  \n",
       "CHR                          \n",
       "1       3759     3759  3759  \n",
       "2       3928     3928  3928  \n",
       "3       3355     3355  3355  \n",
       "4       2940     2940  2940  \n",
       "5       3065     3065  3065  \n",
       "6       3304     3304  3304  \n",
       "7       2668     2668  2668  \n",
       "8       2712     2712  2712  \n",
       "9       2392     2392  2392  \n",
       "10      2535     2535  2535  \n",
       "11      2391     2391  2391  \n",
       "12      2367     2367  2367  \n",
       "13      1822     1822  1822  \n",
       "14      1625     1625  1625  \n",
       "15      1425     1425  1425  \n",
       "16      1437     1437  1437  \n",
       "17      1236     1236  1236  \n",
       "18      1443     1443  1443  \n",
       "19       843      843   843  \n",
       "20      1219     1219  1219  \n",
       "21       699      699   699  \n",
       "22       678      678   678  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "groupbyDF.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "#fig2 = ManhattanPlot(SampleData1, title='Manhattan plot', annotation = 'annotation', gene= None)\n",
    "#fig2.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Selecting the significant SNPS from the sampled datasest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>OR_logistic</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">1</th>\n",
       "      <th>3755615</th>\n",
       "      <th>2382</th>\n",
       "      <td>rs4648347</td>\n",
       "      <td>1777</td>\n",
       "      <td>2766</td>\n",
       "      <td>1103</td>\n",
       "      <td>0</td>\n",
       "      <td>6407</td>\n",
       "      <td>9531</td>\n",
       "      <td>3479</td>\n",
       "      <td>3</td>\n",
       "      <td>0.002954</td>\n",
       "      <td>1.066250</td>\n",
       "      <td>1.022080</td>\n",
       "      <td>1.112330</td>\n",
       "      <td>0.440312</td>\n",
       "      <td>0.424602</td>\n",
       "      <td>0.428141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>3755615</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11737218</th>\n",
       "      <th>2074</th>\n",
       "      <td>rs1133398</td>\n",
       "      <td>2321</td>\n",
       "      <td>2601</td>\n",
       "      <td>721</td>\n",
       "      <td>3</td>\n",
       "      <td>8400</td>\n",
       "      <td>8711</td>\n",
       "      <td>2299</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002519</td>\n",
       "      <td>1.069770</td>\n",
       "      <td>1.023960</td>\n",
       "      <td>1.117640</td>\n",
       "      <td>0.358231</td>\n",
       "      <td>0.342839</td>\n",
       "      <td>0.346306</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>11737218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21688247</th>\n",
       "      <th>2124</th>\n",
       "      <td>rs904927</td>\n",
       "      <td>191</td>\n",
       "      <td>1579</td>\n",
       "      <td>3876</td>\n",
       "      <td>0</td>\n",
       "      <td>702</td>\n",
       "      <td>5828</td>\n",
       "      <td>12889</td>\n",
       "      <td>1</td>\n",
       "      <td>0.002591</td>\n",
       "      <td>1.087590</td>\n",
       "      <td>1.029750</td>\n",
       "      <td>1.148680</td>\n",
       "      <td>0.173663</td>\n",
       "      <td>0.186209</td>\n",
       "      <td>0.183383</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>21688247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28084237</th>\n",
       "      <th>1179</th>\n",
       "      <td>rs1467464</td>\n",
       "      <td>1965</td>\n",
       "      <td>2713</td>\n",
       "      <td>968</td>\n",
       "      <td>0</td>\n",
       "      <td>6389</td>\n",
       "      <td>9400</td>\n",
       "      <td>3627</td>\n",
       "      <td>4</td>\n",
       "      <td>0.001238</td>\n",
       "      <td>0.932670</td>\n",
       "      <td>0.894036</td>\n",
       "      <td>0.972974</td>\n",
       "      <td>0.411707</td>\n",
       "      <td>0.428873</td>\n",
       "      <td>0.425006</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>28084237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33019027</th>\n",
       "      <th>463</th>\n",
       "      <td>rs2282293</td>\n",
       "      <td>4029</td>\n",
       "      <td>1486</td>\n",
       "      <td>131</td>\n",
       "      <td>0</td>\n",
       "      <td>14340</td>\n",
       "      <td>4675</td>\n",
       "      <td>405</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000310</td>\n",
       "      <td>1.113240</td>\n",
       "      <td>1.050170</td>\n",
       "      <td>1.180110</td>\n",
       "      <td>0.154800</td>\n",
       "      <td>0.141220</td>\n",
       "      <td>0.144279</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>33019027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"5\" valign=\"top\">22</th>\n",
       "      <th>33742833</th>\n",
       "      <th>2867</th>\n",
       "      <td>rs2008519</td>\n",
       "      <td>2661</td>\n",
       "      <td>2460</td>\n",
       "      <td>524</td>\n",
       "      <td>1</td>\n",
       "      <td>8850</td>\n",
       "      <td>8505</td>\n",
       "      <td>2061</td>\n",
       "      <td>4</td>\n",
       "      <td>0.003776</td>\n",
       "      <td>0.935371</td>\n",
       "      <td>0.894015</td>\n",
       "      <td>0.978641</td>\n",
       "      <td>0.310717</td>\n",
       "      <td>0.325170</td>\n",
       "      <td>0.321915</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>33742833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35839033</th>\n",
       "      <th>1803</th>\n",
       "      <td>rs11914132</td>\n",
       "      <td>111</td>\n",
       "      <td>1435</td>\n",
       "      <td>4100</td>\n",
       "      <td>0</td>\n",
       "      <td>353</td>\n",
       "      <td>4555</td>\n",
       "      <td>14512</td>\n",
       "      <td>0</td>\n",
       "      <td>0.002154</td>\n",
       "      <td>0.910620</td>\n",
       "      <td>0.857731</td>\n",
       "      <td>0.966770</td>\n",
       "      <td>0.146741</td>\n",
       "      <td>0.135453</td>\n",
       "      <td>0.137996</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>35839033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46513031</th>\n",
       "      <th>254</th>\n",
       "      <td>rs132198</td>\n",
       "      <td>465</td>\n",
       "      <td>2318</td>\n",
       "      <td>2863</td>\n",
       "      <td>0</td>\n",
       "      <td>1845</td>\n",
       "      <td>8220</td>\n",
       "      <td>9355</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000112</td>\n",
       "      <td>1.094980</td>\n",
       "      <td>1.045710</td>\n",
       "      <td>1.146580</td>\n",
       "      <td>0.287637</td>\n",
       "      <td>0.306643</td>\n",
       "      <td>0.302362</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>46513031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46547970</th>\n",
       "      <th>868</th>\n",
       "      <td>rs4315633</td>\n",
       "      <td>567</td>\n",
       "      <td>2419</td>\n",
       "      <td>2660</td>\n",
       "      <td>0</td>\n",
       "      <td>2155</td>\n",
       "      <td>8563</td>\n",
       "      <td>8700</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000828</td>\n",
       "      <td>1.079540</td>\n",
       "      <td>1.032160</td>\n",
       "      <td>1.129090</td>\n",
       "      <td>0.314648</td>\n",
       "      <td>0.331471</td>\n",
       "      <td>0.327681</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>46547970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47460737</th>\n",
       "      <th>588</th>\n",
       "      <td>rs5771710</td>\n",
       "      <td>785</td>\n",
       "      <td>2659</td>\n",
       "      <td>2201</td>\n",
       "      <td>1</td>\n",
       "      <td>2523</td>\n",
       "      <td>8803</td>\n",
       "      <td>8093</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000482</td>\n",
       "      <td>0.925913</td>\n",
       "      <td>0.886736</td>\n",
       "      <td>0.966822</td>\n",
       "      <td>0.374579</td>\n",
       "      <td>0.356584</td>\n",
       "      <td>0.360637</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>47460737</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>355 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          SNP  case11  case12  case22  case??  ctrl11  ctrl12  \\\n",
       "CHR BP                                                                          \n",
       "1   3755615  2382   rs4648347    1777    2766    1103       0    6407    9531   \n",
       "    11737218 2074   rs1133398    2321    2601     721       3    8400    8711   \n",
       "    21688247 2124    rs904927     191    1579    3876       0     702    5828   \n",
       "    28084237 1179   rs1467464    1965    2713     968       0    6389    9400   \n",
       "    33019027 463    rs2282293    4029    1486     131       0   14340    4675   \n",
       "...                       ...     ...     ...     ...     ...     ...     ...   \n",
       "22  33742833 2867   rs2008519    2661    2460     524       1    8850    8505   \n",
       "    35839033 1803  rs11914132     111    1435    4100       0     353    4555   \n",
       "    46513031 254     rs132198     465    2318    2863       0    1845    8220   \n",
       "    46547970 868    rs4315633     567    2419    2660       0    2155    8563   \n",
       "    47460737 588    rs5771710     785    2659    2201       1    2523    8803   \n",
       "\n",
       "                   ctrl22  ctrl??         P  OR_logistic  OR_lower  OR_upper  \\\n",
       "CHR BP                                                                         \n",
       "1   3755615  2382    3479       3  0.002954     1.066250  1.022080  1.112330   \n",
       "    11737218 2074    2299      10  0.002519     1.069770  1.023960  1.117640   \n",
       "    21688247 2124   12889       1  0.002591     1.087590  1.029750  1.148680   \n",
       "    28084237 1179    3627       4  0.001238     0.932670  0.894036  0.972974   \n",
       "    33019027 463      405       0  0.000310     1.113240  1.050170  1.180110   \n",
       "...                   ...     ...       ...          ...       ...       ...   \n",
       "22  33742833 2867    2061       4  0.003776     0.935371  0.894015  0.978641   \n",
       "    35839033 1803   14512       0  0.002154     0.910620  0.857731  0.966770   \n",
       "    46513031 254     9355       0  0.000112     1.094980  1.045710  1.146580   \n",
       "    46547970 868     8700       2  0.000828     1.079540  1.032160  1.129090   \n",
       "    47460737 588     8093       1  0.000482     0.925913  0.886736  0.966822   \n",
       "\n",
       "                   maf_case  maf_ctrl  annotation alleleA alleleB  CHR  \\\n",
       "CHR BP                                                                   \n",
       "1   3755615  2382  0.440312  0.424602    0.428141       T       C    1   \n",
       "    11737218 2074  0.358231  0.342839    0.346306       T       C    1   \n",
       "    21688247 2124  0.173663  0.186209    0.183383       T       C    1   \n",
       "    28084237 1179  0.411707  0.428873    0.425006       A       G    1   \n",
       "    33019027 463   0.154800  0.141220    0.144279       T       C    1   \n",
       "...                     ...       ...         ...     ...     ...  ...   \n",
       "22  33742833 2867  0.310717  0.325170    0.321915       T       C   22   \n",
       "    35839033 1803  0.146741  0.135453    0.137996       T       C   22   \n",
       "    46513031 254   0.287637  0.306643    0.302362       A       C   22   \n",
       "    46547970 868   0.314648  0.331471    0.327681       A       G   22   \n",
       "    47460737 588   0.374579  0.356584    0.360637       T       C   22   \n",
       "\n",
       "                         BP  \n",
       "CHR BP                       \n",
       "1   3755615  2382   3755615  \n",
       "    11737218 2074  11737218  \n",
       "    21688247 2124  21688247  \n",
       "    28084237 1179  28084237  \n",
       "    33019027 463   33019027  \n",
       "...                     ...  \n",
       "22  33742833 2867  33742833  \n",
       "    35839033 1803  35839033  \n",
       "    46513031 254   46513031  \n",
       "    46547970 868   46547970  \n",
       "    47460737 588   47460737  \n",
       "\n",
       "[355 rows x 20 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "grouped_dataFrame_sign_snps = SampleData.loc[SampleData['P'] <= 0.005]\n",
    "grouped_dataFrame_sign_snps"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Creating a list of ref significnat SNPS from the Sample Datasets\n",
    "and plicling the object for use in GENE extraction from Ensemble DB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Create the lists of selected snips. \n",
    "rsfsnps = list(grouped_dataFrame_sign_snps['SNP'])\n",
    "#rsfsnps\n",
    "# Creating a pickle object for the list\n",
    "with open('selected_rsf_snps.pickle', 'wb') as selected_rsf_snps_obj: # Creating pickle file object\n",
    "    pickle.dump(rsfsnps,selected_rsf_snps_obj) # Dumping the object into the pickle file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['rs4648347', 'rs1133398', 'rs904927', 'rs1467464', 'rs2282293', 'rs4927309', 'rs3820577', 'rs12116456', 'rs6679774', 'rs1983967', 'rs10494267', 'rs10796927', 'rs10908498', 'rs1148821', 'rs8972', 'rs861581', 'rs549890', 'rs12072845', 'rs7554672', 'rs729386', 'rs7606197', 'rs556902', 'rs548032', 'rs10928585', 'rs12999294', 'rs7586173', 'rs6745954', 'rs13383927', 'rs6435979', 'rs7608896', 'rs1860700', 'rs11674589', 'rs1355070', 'rs1601375', 'rs4488811', 'rs9837421', 'rs259557', 'rs259489', 'rs13098412', 'rs1353322', 'rs6799309', 'rs1798802', 'rs4973937', 'rs7650709', 'rs28758977', 'rs6785239', 'rs17790790', 'rs1159290', 'rs260176', 'rs12639377', 'rs9859887', 'rs9851621', 'rs6805459', 'rs6809037', 'rs6438385', 'rs13074860', 'rs6794836', 'rs1421425', 'rs11929676', 'rs6809208', 'rs6786711', 'rs4130791', 'rs4692128', 'rs17653012', 'rs885443', 'rs13147707', 'rs7654525', 'rs4860532', 'rs4860707', 'rs11735016', 'rs2319691', 'rs11945668', 'rs4695789', 'rs1485936', 'rs4635969', 'rs13167887', 'rs16903506', 'rs16903629', 'rs415144', 'rs12653839', 'rs41461', 'rs163282', 'rs490982', 'rs3111643', 'rs266580', 'rs4704066', 'rs1873425', 'rs13362486', 'rs12719549', 'rs10463584', 'rs9686134', 'rs6870951', 'rs4912622', 'rs10482655', 'rs6579767', 'rs4921133', 'rs1105788', 'rs17061327', 'rs360018', 'rs4701077', 'rs6901207', 'rs9502099', 'rs407447', 'rs2274393', 'rs9357002', 'rs9348432', 'rs9350270', 'rs6928012', 'rs12208378', 'rs4712573', 'rs2028973', 'rs2394423', 'rs35120848', 'rs6934302', 'rs2882406', 'rs4632930', 'rs12660231', 'rs487467', 'rs9385564', 'rs271159', 'rs9483769', 'rs1122537', 'rs9322378', 'rs674882', 'rs10499286', 'rs783137', 'rs12524067', 'rs9346489', 'rs836472', 'rs13224907', 'rs6949976', 'rs7777551', 'rs4720971', 'rs11514706', 'rs929250', 'rs13241670', 'rs10264649', 'rs6466086', 'rs10428921', 'rs9641678', 'rs1206367', 'rs2237731', 'rs4731344', 'rs7811565', 'rs10487490', 'rs6467136', 'rs1557643', 'rs3808081', 'rs12155423', 'rs7836428', 'rs9325882', 'rs13267264', 'rs13250561', 'rs7002486', 'rs2920123', 'rs2304869', 'rs2979896', 'rs567971', 'rs6993988', 'rs16933420', 'rs11987041', 'rs10108298', 'rs2255464', 'rs2961110', 'rs2510831', 'rs1948503', 'rs11782735', 'rs1543157', 'rs7824890', 'rs7004029', 'rs10739044', 'rs1392511', 'rs970346', 'rs7037887', 'rs10756819', 'rs10963522', 'rs934475', 'rs1548375', 'rs10965243', 'rs7871143', 'rs2039580', 'rs13288849', 'rs2275683', 'rs2740476', 'rs4149309', 'rs4742987', 'rs4978988', 'rs12342334', 'rs3814593', 'rs10508438', 'rs17151496', 'rs7092141', 'rs3780962', 'rs597837', 'rs10828441', 'rs10508663', 'rs211327', 'rs2504397', 'rs7076624', 'rs282669', 'rs1340586', 'rs1938539', 'rs7081445', 'rs10509233', 'rs4691', 'rs703974', 'rs1029134', 'rs10881830', 'rs7923837', 'rs4933744', 'rs1048445', 'rs10786637', 'rs1416410', 'rs11193407', 'rs2125071', 'rs7086699', 'rs12255372', 'rs2921965', 'rs978484', 'rs12572903', 'rs11248405', 'rs2366878', 'rs11042978', 'rs2283208', 'rs231917', 'rs163171', 'rs4758132', 'rs10766227', 'rs2615040', 'rs11024102', 'rs11025746', 'rs327036', 'rs1223067', 'rs2434474', 'rs4601790', 'rs754532', 'rs2511224', 'rs603965', 'rs17134821', 'rs17134828', 'rs1892953', 'rs7936562', 'rs4384350', 'rs525057', 'rs4754231', 'rs11602361', 'rs7130091', 'rs11063022', 'rs16930053', 'rs7133442', 'rs11836308', 'rs1574409', 'rs4132612', 'rs10861414', 'rs17017109', 'rs17751083', 'rs2373358', 'rs12581164', 'rs1426369', 'rs11057706', 'rs275769', 'rs10847854', 'rs597425', 'rs6491066', 'rs4371022', 'rs9512150', 'rs10507349', 'rs1475221', 'rs2263207', 'rs576674', 'rs12585743', 'rs2225806', 'rs1888261', 'rs9523299', 'rs9524885', 'rs7986656', 'rs1454368', 'rs12889162', 'rs11157815', 'rs1953352', 'rs8017172', 'rs3742656', 'rs7157940', 'rs1064017', 'rs2173288', 'rs10163076', 'rs4334271', 'rs745299', 'rs4144949', 'rs7181383', 'rs8030981', 'rs11631950', 'rs935214', 'rs10518995', 'rs11855027', 'rs12898253', 'rs4785909', 'rs2531995', 'rs13336060', 'rs9925093', 'rs2226994', 'rs12149256', 'rs2218798', 'rs2397376', 'rs17771867', 'rs4788697', 'rs9937686', 'rs2549773', 'rs8058370', 'rs247400', 'rs8080511', 'rs444207', 'rs8069439', 'rs12952139', 'rs757210', 'rs4430796', 'rs9911630', 'rs8176257', 'rs9912203', 'rs11651885', 'rs2040611', 'rs2286677', 'rs12967286', 'rs16941039', 'rs9965472', 'rs643555', 'rs2635516', 'rs12454650', 'rs334456', 'rs9952696', 'rs1078633', 'rs7257948', 'rs12461534', 'rs10417832', 'rs731672', 'rs10414662', 'rs149820', 'rs6135622', 'rs7275102', 'rs17436308', 'rs6099057', 'rs2825673', 'rs2829735', 'rs6003249', 'rs5996364', 'rs179306', 'rs9306387', 'rs8141797', 'rs9613035', 'rs137500', 'rs2008519', 'rs11914132', 'rs132198', 'rs4315633', 'rs5771710']\n"
     ]
    }
   ],
   "source": [
    "print(rsfsnps)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Reading  CSV  files of extrcated genes from Ensemble DB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_b_data = pd.read_table(\"genes_b_data.csv\", sep = '\\s+')\n",
    "\n",
    "genes_a_data = pd.read_csv('genes_a_data.csv', sep = '\\s+')\n",
    "\n",
    "genes_c_data = pd.read_csv('genes_c_data.csv', sep ='\\s+')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refsnp_id</th>\n",
       "      <th>associated_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs929250</td>\n",
       "      <td>HOTTIP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs1557643</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2237731</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs1206367</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs3808081</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>rs10756819</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>rs13288849</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>rs12342334</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>rs10963522</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>rs10965243</td>\n",
       "      <td>CDKN2B-AS1,DMRTA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      refsnp_id    associated_gene\n",
       "0      rs929250             HOTTIP\n",
       "1     rs1557643                NaN\n",
       "2     rs2237731                NaN\n",
       "3     rs1206367                NaN\n",
       "4     rs3808081                NaN\n",
       "..          ...                ...\n",
       "127  rs10756819                NaN\n",
       "128  rs13288849                NaN\n",
       "129  rs12342334                NaN\n",
       "130  rs10963522                NaN\n",
       "131  rs10965243  CDKN2B-AS1,DMRTA1\n",
       "\n",
       "[132 rows x 2 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_b_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refsnp_id</th>\n",
       "      <th>associated_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs8972</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs259489</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs259557</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs260176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs549890</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>rs7654525</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>rs11945668</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>rs13147707</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>rs17653012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>rs11735016</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      refsnp_id associated_gene\n",
       "0        rs8972             NaN\n",
       "1      rs259489             NaN\n",
       "2      rs259557             NaN\n",
       "3      rs260176             NaN\n",
       "4      rs549890             NaN\n",
       "..          ...             ...\n",
       "127   rs7654525             NaN\n",
       "128  rs11945668             NaN\n",
       "129  rs13147707             NaN\n",
       "130  rs17653012             NaN\n",
       "131  rs11735016             NaN\n",
       "\n",
       "[132 rows x 2 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_a_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>refsnp_id</th>\n",
       "      <th>associated_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs745299</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs935214</td>\n",
       "      <td>RPL28P4,LIPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2173288</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs4334271</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7181383</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>rs12967286</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>rs16941039</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>rs9965472</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>rs9952696</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>rs10847854</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>106 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      refsnp_id associated_gene\n",
       "0      rs745299             NaN\n",
       "1      rs935214    RPL28P4,LIPC\n",
       "2     rs2173288             NaN\n",
       "3     rs4334271             NaN\n",
       "4     rs7181383             NaN\n",
       "..          ...             ...\n",
       "101  rs12967286             NaN\n",
       "102  rs16941039             NaN\n",
       "103   rs9965472             NaN\n",
       "104   rs9952696             NaN\n",
       "105  rs10847854             NaN\n",
       "\n",
       "[106 rows x 2 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_c_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Concatnating the two data frames which contain the genes, and renaming the columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs8972</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs259489</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs259557</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs260176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs549890</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>365</th>\n",
       "      <td>rs12967286</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366</th>\n",
       "      <td>rs16941039</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>rs9965472</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>rs9952696</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>rs10847854</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>370 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            SNP GENE\n",
       "0        rs8972  NaN\n",
       "1      rs259489  NaN\n",
       "2      rs259557  NaN\n",
       "3      rs260176  NaN\n",
       "4      rs549890  NaN\n",
       "..          ...  ...\n",
       "365  rs12967286  NaN\n",
       "366  rs16941039  NaN\n",
       "367   rs9965472  NaN\n",
       "368   rs9952696  NaN\n",
       "369  rs10847854  NaN\n",
       "\n",
       "[370 rows x 2 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_data_merged = pd.concat([genes_a_data, genes_b_data,genes_c_data ], \n",
    "                              axis = 0).rename(columns = {'refsnp_id': 'SNP','associated_gene': 'GENE' }).reset_index(drop = True)\n",
    "genes_data_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### The original genes with nan values dropped"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4488811</td>\n",
       "      <td>SYN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs12116456</td>\n",
       "      <td>ZNF326,BARHL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs9837421</td>\n",
       "      <td>SH3BP5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs6785239</td>\n",
       "      <td>FAM19A4,C3orf64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7554672</td>\n",
       "      <td>LOC101929750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>rs6579767</td>\n",
       "      <td>HMGXB3xEPC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>rs17061327</td>\n",
       "      <td>MRP63P6,CCNG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>rs13167887</td>\n",
       "      <td>LPCAT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>TERT,CLPTM1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>CLPTM1L</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>rs7586173</td>\n",
       "      <td>SESTD1,ZNF385B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>rs7606197</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>rs2394423</td>\n",
       "      <td>HCG22,C6orf15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>rs6928012</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>rs929250</td>\n",
       "      <td>HOTTIP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>rs6467136</td>\n",
       "      <td>PAX4,GCC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>rs13241670</td>\n",
       "      <td>WBSCR28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>rs2366878</td>\n",
       "      <td>C10orf90,DOCK1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>rs7923837</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>rs7923837</td>\n",
       "      <td>HHEX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>rs7923837</td>\n",
       "      <td>HHEX,EXOC6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>rs12255372</td>\n",
       "      <td>TCF7L2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>rs231917</td>\n",
       "      <td>KCNQ1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>rs4758132</td>\n",
       "      <td>OR2AG2,OR2AG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>rs11024102</td>\n",
       "      <td>PLEKHA7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>rs4601790</td>\n",
       "      <td>EHBP1L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>rs6993988</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>rs2740476</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>rs4149309</td>\n",
       "      <td>ABCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>rs10965243</td>\n",
       "      <td>CDKN2B-AS1,DMRTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>rs935214</td>\n",
       "      <td>RPL28P4,LIPC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>rs2531995</td>\n",
       "      <td>ADCY9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>rs2531995</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>rs576674</td>\n",
       "      <td>KL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>rs597425</td>\n",
       "      <td>NUS1P2,SGCG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>rs597425</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>rs6491066</td>\n",
       "      <td>ATP8A2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>rs10507349</td>\n",
       "      <td>RNF6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>rs7157940</td>\n",
       "      <td>LOC122509,FAM14B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>rs3742656</td>\n",
       "      <td>CCDC88C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>rs8017172</td>\n",
       "      <td>RPL13AP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>rs8017172</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>HNF1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>HNF1B,TCF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>EPHA10xHNF1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>ANP32DxHNF1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>intergenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>rs4430796</td>\n",
       "      <td>TCF2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>rs444207</td>\n",
       "      <td>CLEC10A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>rs757210</td>\n",
       "      <td>HNF1B,ACACA,C17orf78,TADA2A,DUSP14,SYNRG,DDX52...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>rs757210</td>\n",
       "      <td>HNF1B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>rs8176257</td>\n",
       "      <td>BRCA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>rs11651885</td>\n",
       "      <td>BCAS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>CABIN1,SUSD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>SUSD2,CABIN1,GGT5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>rs4315633</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP                                               GENE\n",
       "0    rs4488811                                               SYN2\n",
       "1   rs12116456                                      ZNF326,BARHL2\n",
       "2    rs9837421                                             SH3BP5\n",
       "3    rs6785239                                    FAM19A4,C3orf64\n",
       "4    rs7554672                                       LOC101929750\n",
       "5    rs6579767                                        HMGXB3xEPC1\n",
       "6   rs17061327                                      MRP63P6,CCNG1\n",
       "7   rs13167887                                             LPCAT1\n",
       "8    rs4635969                                       TERT,CLPTM1L\n",
       "9    rs4635969                                            CLPTM1L\n",
       "10   rs7586173                                     SESTD1,ZNF385B\n",
       "11   rs7606197                                                 NR\n",
       "12   rs2394423                                      HCG22,C6orf15\n",
       "13   rs6928012                                                 NR\n",
       "14    rs929250                                             HOTTIP\n",
       "15   rs6467136                                          PAX4,GCC1\n",
       "16  rs13241670                                            WBSCR28\n",
       "17   rs2366878                                     C10orf90,DOCK1\n",
       "18   rs7923837                                                 NR\n",
       "19   rs7923837                                               HHEX\n",
       "20   rs7923837                                         HHEX,EXOC6\n",
       "21  rs12255372                                             TCF7L2\n",
       "22    rs231917                                              KCNQ1\n",
       "23   rs4758132                                      OR2AG2,OR2AG1\n",
       "24  rs11024102                                            PLEKHA7\n",
       "25   rs4601790                                            EHBP1L1\n",
       "26   rs6993988                                                 NR\n",
       "27   rs2740476                                                 NR\n",
       "28   rs4149309                                              ABCA1\n",
       "29  rs10965243                                  CDKN2B-AS1,DMRTA1\n",
       "30    rs935214                                       RPL28P4,LIPC\n",
       "31   rs2531995                                              ADCY9\n",
       "32   rs2531995                                                 NR\n",
       "33    rs576674                                                 KL\n",
       "34    rs597425                                        NUS1P2,SGCG\n",
       "35    rs597425                                                 NR\n",
       "36   rs6491066                                             ATP8A2\n",
       "37  rs10507349                                               RNF6\n",
       "38   rs7157940                                   LOC122509,FAM14B\n",
       "39   rs3742656                                            CCDC88C\n",
       "40   rs8017172                                           RPL13AP3\n",
       "41   rs8017172                                                 NR\n",
       "42   rs4430796                                              HNF1B\n",
       "43   rs4430796                                         HNF1B,TCF2\n",
       "44   rs4430796                                       EPHA10xHNF1B\n",
       "45   rs4430796                                       ANP32DxHNF1B\n",
       "46   rs4430796                                                 NR\n",
       "47   rs4430796                                         intergenic\n",
       "48   rs4430796                                               TCF2\n",
       "49    rs444207                                            CLEC10A\n",
       "50    rs757210  HNF1B,ACACA,C17orf78,TADA2A,DUSP14,SYNRG,DDX52...\n",
       "51    rs757210                                              HNF1B\n",
       "52   rs8176257                                              BRCA1\n",
       "53  rs11651885                                              BCAS3\n",
       "54   rs8141797                                       CABIN1,SUSD2\n",
       "55   rs8141797                                  SUSD2,CABIN1,GGT5\n",
       "56   rs4315633                                                 NR"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_data_merged_nanDropped = genes_data_merged[genes_data_merged['GENE'].notna()].reset_index(drop = True)\n",
    "#df = df[df['EPS'].notna()]\n",
    "genes_data_merged_nanDropped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Spliting the Genes with comma str and droping the  rows with nan GENE columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_data_merged_split = genes_data_merged.join(pd.DataFrame(genes_data_merged.GENE.str.split(',', expand=True)\n",
    "                                                              .stack().reset_index(level=1, drop=True)\n",
    "                                                               ,columns=['GENE '])).drop('GENE',1).rename(columns=str.strip).reset_index(drop=True)\n",
    "# spliting GENEs containg 'x', eg: 'ANP32DxHNF1B'\n",
    "genes_data_merged_split = genes_data_merged_split.join(pd.DataFrame(genes_data_merged_split.GENE.str.split('x', expand=True)\n",
    "                                                              .stack().reset_index(level=1, drop=True)\n",
    "                                                              ,columns=['GENE '])).drop('GENE',1).rename(columns=str.strip).reset_index(drop=True)\n",
    "\n",
    "genes_data_merged_split_na_droped = genes_data_merged_split[genes_data_merged_split['GENE'].notna()].reset_index(drop = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs8972</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs259489</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs259557</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs260176</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs549890</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>rs12967286</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>rs16941039</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>400</th>\n",
       "      <td>rs9965472</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>401</th>\n",
       "      <td>rs9952696</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402</th>\n",
       "      <td>rs10847854</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            SNP GENE\n",
       "0        rs8972  NaN\n",
       "1      rs259489  NaN\n",
       "2      rs259557  NaN\n",
       "3      rs260176  NaN\n",
       "4      rs549890  NaN\n",
       "..          ...  ...\n",
       "398  rs12967286  NaN\n",
       "399  rs16941039  NaN\n",
       "400   rs9965472  NaN\n",
       "401   rs9952696  NaN\n",
       "402  rs10847854  NaN\n",
       "\n",
       "[403 rows x 2 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_data_merged_split "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DF with nan values droped "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4488811</td>\n",
       "      <td>SYN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs12116456</td>\n",
       "      <td>ZNF326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs12116456</td>\n",
       "      <td>BARHL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs9837421</td>\n",
       "      <td>SH3BP5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs6785239</td>\n",
       "      <td>FAM19A4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>SUSD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>SUSD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>CABIN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>GGT5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>rs4315633</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP     GENE\n",
       "0    rs4488811     SYN2\n",
       "1   rs12116456   ZNF326\n",
       "2   rs12116456   BARHL2\n",
       "3    rs9837421   SH3BP5\n",
       "4    rs6785239  FAM19A4\n",
       "..         ...      ...\n",
       "85   rs8141797    SUSD2\n",
       "86   rs8141797    SUSD2\n",
       "87   rs8141797   CABIN1\n",
       "88   rs8141797     GGT5\n",
       "89   rs4315633       NR\n",
       "\n",
       "[90 rows x 2 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_data_merged_split_na_droped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge the Genes DF with the original sampled datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "SampleData_Merged_Genes = pd.merge(left =SampleData , right = genes_data_merged,\n",
    "                                   how='left', left_on='SNP', right_on='SNP')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>...</th>\n",
       "      <th>OR_lower</th>\n",
       "      <th>OR_upper</th>\n",
       "      <th>maf_case</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4970420</td>\n",
       "      <td>3640</td>\n",
       "      <td>1790</td>\n",
       "      <td>212</td>\n",
       "      <td>4</td>\n",
       "      <td>12501</td>\n",
       "      <td>6165</td>\n",
       "      <td>732</td>\n",
       "      <td>22</td>\n",
       "      <td>0.917874</td>\n",
       "      <td>...</td>\n",
       "      <td>0.945806</td>\n",
       "      <td>1.05143</td>\n",
       "      <td>0.196207</td>\n",
       "      <td>0.196644</td>\n",
       "      <td>0.196546</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>1096336</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs13303016</td>\n",
       "      <td>3472</td>\n",
       "      <td>1925</td>\n",
       "      <td>247</td>\n",
       "      <td>2</td>\n",
       "      <td>11988</td>\n",
       "      <td>6577</td>\n",
       "      <td>850</td>\n",
       "      <td>5</td>\n",
       "      <td>0.793916</td>\n",
       "      <td>...</td>\n",
       "      <td>0.956453</td>\n",
       "      <td>1.05995</td>\n",
       "      <td>0.214298</td>\n",
       "      <td>0.213160</td>\n",
       "      <td>0.213416</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1936451</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2254669</td>\n",
       "      <td>26</td>\n",
       "      <td>664</td>\n",
       "      <td>4956</td>\n",
       "      <td>0</td>\n",
       "      <td>76</td>\n",
       "      <td>2403</td>\n",
       "      <td>16941</td>\n",
       "      <td>0</td>\n",
       "      <td>0.367653</td>\n",
       "      <td>...</td>\n",
       "      <td>0.954697</td>\n",
       "      <td>1.13340</td>\n",
       "      <td>0.063408</td>\n",
       "      <td>0.065783</td>\n",
       "      <td>0.065248</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1995968</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs424079</td>\n",
       "      <td>3536</td>\n",
       "      <td>1861</td>\n",
       "      <td>245</td>\n",
       "      <td>4</td>\n",
       "      <td>11987</td>\n",
       "      <td>6479</td>\n",
       "      <td>927</td>\n",
       "      <td>27</td>\n",
       "      <td>0.139649</td>\n",
       "      <td>...</td>\n",
       "      <td>0.913970</td>\n",
       "      <td>1.01272</td>\n",
       "      <td>0.208348</td>\n",
       "      <td>0.214846</td>\n",
       "      <td>0.213381</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>2061200</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7512482</td>\n",
       "      <td>4637</td>\n",
       "      <td>951</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>15912</td>\n",
       "      <td>3322</td>\n",
       "      <td>185</td>\n",
       "      <td>1</td>\n",
       "      <td>0.856273</td>\n",
       "      <td>...</td>\n",
       "      <td>0.925055</td>\n",
       "      <td>1.06686</td>\n",
       "      <td>0.094492</td>\n",
       "      <td>0.095061</td>\n",
       "      <td>0.094933</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>2136826</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47854</th>\n",
       "      <td>rs11703226</td>\n",
       "      <td>2494</td>\n",
       "      <td>2543</td>\n",
       "      <td>603</td>\n",
       "      <td>6</td>\n",
       "      <td>8787</td>\n",
       "      <td>8502</td>\n",
       "      <td>2098</td>\n",
       "      <td>33</td>\n",
       "      <td>0.332329</td>\n",
       "      <td>...</td>\n",
       "      <td>0.977734</td>\n",
       "      <td>1.06886</td>\n",
       "      <td>0.332358</td>\n",
       "      <td>0.327487</td>\n",
       "      <td>0.328585</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49000551</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47855</th>\n",
       "      <td>rs5771242</td>\n",
       "      <td>208</td>\n",
       "      <td>1644</td>\n",
       "      <td>3793</td>\n",
       "      <td>1</td>\n",
       "      <td>632</td>\n",
       "      <td>5913</td>\n",
       "      <td>12869</td>\n",
       "      <td>6</td>\n",
       "      <td>0.565587</td>\n",
       "      <td>...</td>\n",
       "      <td>0.962358</td>\n",
       "      <td>1.07271</td>\n",
       "      <td>0.182462</td>\n",
       "      <td>0.184841</td>\n",
       "      <td>0.184305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>49003996</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47856</th>\n",
       "      <td>rs131816</td>\n",
       "      <td>5172</td>\n",
       "      <td>461</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>17706</td>\n",
       "      <td>1657</td>\n",
       "      <td>52</td>\n",
       "      <td>5</td>\n",
       "      <td>0.318233</td>\n",
       "      <td>...</td>\n",
       "      <td>0.857404</td>\n",
       "      <td>1.05131</td>\n",
       "      <td>0.043128</td>\n",
       "      <td>0.045352</td>\n",
       "      <td>0.044851</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49304328</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47857</th>\n",
       "      <td>rs470119</td>\n",
       "      <td>362</td>\n",
       "      <td>2173</td>\n",
       "      <td>3111</td>\n",
       "      <td>0</td>\n",
       "      <td>1359</td>\n",
       "      <td>7428</td>\n",
       "      <td>10632</td>\n",
       "      <td>1</td>\n",
       "      <td>0.319039</td>\n",
       "      <td>...</td>\n",
       "      <td>0.976775</td>\n",
       "      <td>1.07476</td>\n",
       "      <td>0.256553</td>\n",
       "      <td>0.261239</td>\n",
       "      <td>0.260184</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49313780</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47858</th>\n",
       "      <td>rs9616802</td>\n",
       "      <td>1559</td>\n",
       "      <td>2792</td>\n",
       "      <td>1293</td>\n",
       "      <td>2</td>\n",
       "      <td>5279</td>\n",
       "      <td>9570</td>\n",
       "      <td>4570</td>\n",
       "      <td>1</td>\n",
       "      <td>0.323190</td>\n",
       "      <td>...</td>\n",
       "      <td>0.939195</td>\n",
       "      <td>1.02090</td>\n",
       "      <td>0.476435</td>\n",
       "      <td>0.481745</td>\n",
       "      <td>0.480549</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>49419027</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47859 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0       rs4970420    3640    1790     212       4   12501    6165     732   \n",
       "1      rs13303016    3472    1925     247       2   11988    6577     850   \n",
       "2       rs2254669      26     664    4956       0      76    2403   16941   \n",
       "3        rs424079    3536    1861     245       4   11987    6479     927   \n",
       "4       rs7512482    4637     951      58       0   15912    3322     185   \n",
       "...           ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "47854  rs11703226    2494    2543     603       6    8787    8502    2098   \n",
       "47855   rs5771242     208    1644    3793       1     632    5913   12869   \n",
       "47856    rs131816    5172     461      13       0   17706    1657      52   \n",
       "47857    rs470119     362    2173    3111       0    1359    7428   10632   \n",
       "47858   rs9616802    1559    2792    1293       2    5279    9570    4570   \n",
       "\n",
       "       ctrl??         P  ...  OR_lower  OR_upper  maf_case  maf_ctrl  \\\n",
       "0          22  0.917874  ...  0.945806   1.05143  0.196207  0.196644   \n",
       "1           5  0.793916  ...  0.956453   1.05995  0.214298  0.213160   \n",
       "2           0  0.367653  ...  0.954697   1.13340  0.063408  0.065783   \n",
       "3          27  0.139649  ...  0.913970   1.01272  0.208348  0.214846   \n",
       "4           1  0.856273  ...  0.925055   1.06686  0.094492  0.095061   \n",
       "...       ...       ...  ...       ...       ...       ...       ...   \n",
       "47854      33  0.332329  ...  0.977734   1.06886  0.332358  0.327487   \n",
       "47855       6  0.565587  ...  0.962358   1.07271  0.182462  0.184841   \n",
       "47856       5  0.318233  ...  0.857404   1.05131  0.043128  0.045352   \n",
       "47857       1  0.319039  ...  0.976775   1.07476  0.256553  0.261239   \n",
       "47858       1  0.323190  ...  0.939195   1.02090  0.476435  0.481745   \n",
       "\n",
       "       annotation  alleleA alleleB CHR        BP  GENE  \n",
       "0        0.196546        T       C   1   1096336   NaN  \n",
       "1        0.213416        A       G   1   1936451   NaN  \n",
       "2        0.065248        A       G   1   1995968   NaN  \n",
       "3        0.213381        A       C   1   2061200   NaN  \n",
       "4        0.094933        T       C   1   2136826   NaN  \n",
       "...           ...      ...     ...  ..       ...   ...  \n",
       "47854    0.328585        T       C  22  49000551   NaN  \n",
       "47855    0.184305        A       G  22  49003996   NaN  \n",
       "47856    0.044851        T       C  22  49304328   NaN  \n",
       "47857    0.260184        T       C  22  49313780   NaN  \n",
       "47858    0.480549        A       G  22  49419027   NaN  \n",
       "\n",
       "[47859 rows x 21 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SampleData_Merged_Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "#fig3 = ManhattanPlot(SampleData_Merged_Genes, title='Manhattan plot', annotation = 'annotation', gene= 'GENE')\n",
    "#fig3.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "71"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_gene = list(set(genes_data_merged_split_na_droped['GENE']))\n",
    "len(unique_gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Looking for pathway (by genes i.e., IDs or usual name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "k = KEGG()\n",
    "k.organism = \"hsa\"\n",
    "human_Path = k.pathwayIds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "____Done Execution!____\n"
     ]
    }
   ],
   "source": [
    "run_gene_list = unique_gene #genelist_pickl_object\n",
    "\n",
    "    #run_gene_list = ['FADS1', 'FADS2', 'FADS3']\n",
    "pathway_dict = {}    \n",
    "list_pathways = []\n",
    "list_res = []\n",
    "list_entry = []  \n",
    "Genes_WithPath_hit =[] # Genes With Pathway hits \n",
    "\n",
    "for a_gene in run_gene_list:\n",
    "    pathways = k.get_pathway_by_gene(a_gene, \"hsa\")\n",
    "        #print(pathways)\n",
    "        \n",
    "    if pathways != None:\n",
    "        \n",
    "        pathway_dict[a_gene] = pathways\n",
    "        list_pathways.append(pathways)\n",
    "        Genes_WithPath_hit.append(a_gene)\n",
    "        \n",
    "        for a_pathway in pathways.keys():\n",
    "           # search for pathways that contain the required gene Id and relations\n",
    "            res = k.parse_kgml_pathway(a_pathway) \n",
    "            list_res.append(a_pathway)  \n",
    "            #print(res.keys())\n",
    "            \n",
    "            gene_id = None\n",
    "            \n",
    "            for entry in res['entries']:  \n",
    "                \n",
    "                if entry['gene_names'] != None:\n",
    "                    \n",
    "                    if a_gene in entry['gene_names'].split(', '):\n",
    "                        list_entry.append(entry)\n",
    "\n",
    "                           \n",
    "print (\"____Done Execution!____\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Serialise the objects from KEGG Database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "pickle_pathway_Obj = list_pathways\n",
    "\n",
    "pickle_dict_res_Obj = list_res\n",
    "\n",
    "pickle_list_entries = list_entry\n",
    "\n",
    "pickle_pathway_dict= pathway_dict\n",
    "\n",
    "with open('pathways.obj', 'wb') as file_pathway_Obj: # Creating pickle file object\n",
    "    pickle.dump(pickle_pathway_Obj,file_pathway_Obj) # Dumping the object into the pickle file\n",
    "    file_pathway_Obj.close()\n",
    "\n",
    "    # Pathway IDS\n",
    "with open('res.obj','wb') as file_dict_res_Obj:\n",
    "        pickle.dump(pickle_dict_res_Obj,file_dict_res_Obj)\n",
    "        file_dict_res_Obj.close()\n",
    "\n",
    "with open('entries.obj','wb') as file_list_entry_Obj:\n",
    "    pickle.dump(pickle_list_entries,file_list_entry_Obj)\n",
    "    file_list_entry_Obj.close()\n",
    "    \n",
    "with open('pathway_dict.obj','wb') as pathway_dict_Obj:\n",
    "    pickle.dump(pickle_pathway_dict,pathway_dict_Obj)\n",
    "    pathway_dict_Obj.close()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Deseralise the stored pickle files for processing "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('pathways.obj', 'rb') as ds_file_pathway_Obj:\n",
    "\n",
    "    pickle_pathway_Obj = pickle.load(ds_file_pathway_Obj)\n",
    "\n",
    "    ds_file_pathway_Obj.close()\n",
    "\n",
    "with open('res.obj','rb') as ds_file_dict_res_Obj:\n",
    "\n",
    "    pickle_dict_res_Obj = pickle.load(ds_file_dict_res_Obj)\n",
    "\n",
    "    ds_file_dict_res_Obj.close()\n",
    "\n",
    "\n",
    "with open(\"entries.obj\",'rb') as ds_file_list_entry_Obj:\n",
    "\n",
    "    pickle_list_entry_Obj = pickle.load(ds_file_list_entry_Obj)\n",
    "\n",
    "    ds_file_list_entry_Obj.close()\n",
    "\n",
    "with open('pathway_dict.obj','rb') as ds_pathway_dict_Obj:\n",
    "    pickle_pathway_dict_Obj = pickle.load(ds_pathway_dict_Obj)\n",
    "    ds_pathway_dict_Obj.close()\n",
    "\n",
    "#return (pickle_pathway_Obj, pickle_dict_res_Obj, pickle_list_entry_Obj)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Creating the DF from the deseralised objs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HHEX</th>\n",
       "      <th>LPCAT1</th>\n",
       "      <th>TERT</th>\n",
       "      <th>KCNQ1</th>\n",
       "      <th>ADCY9</th>\n",
       "      <th>GGT5</th>\n",
       "      <th>OR2AG1</th>\n",
       "      <th>SGCG</th>\n",
       "      <th>DOCK1</th>\n",
       "      <th>ACACA</th>\n",
       "      <th>SOCS7</th>\n",
       "      <th>KL</th>\n",
       "      <th>TCF7L2</th>\n",
       "      <th>BRCA1</th>\n",
       "      <th>CCNG1</th>\n",
       "      <th>HNF1B</th>\n",
       "      <th>LIPC</th>\n",
       "      <th>ABCA1</th>\n",
       "      <th>PAX4</th>\n",
       "      <th>OR2AG2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>hsa04950</th>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa05202</th>\n",
       "      <td>Transcriptional misregulation in cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa00564</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Glycerophospholipid metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa00565</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Ether lipid metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa01100</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa04115</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p53 signaling pathway</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa00561</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Glycerolipid metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa04979</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cholesterol metabolism</td>\n",
       "      <td>Cholesterol metabolism</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa02010</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>hsa04975</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fat digestion and absorption</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>103 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             HHEX  \\\n",
       "hsa04950     Maturity onset diabetes of the young   \n",
       "hsa05202  Transcriptional misregulation in cancer   \n",
       "hsa00564                                      NaN   \n",
       "hsa00565                                      NaN   \n",
       "hsa01100                                      NaN   \n",
       "...                                           ...   \n",
       "hsa04115                                      NaN   \n",
       "hsa00561                                      NaN   \n",
       "hsa04979                                      NaN   \n",
       "hsa02010                                      NaN   \n",
       "hsa04975                                      NaN   \n",
       "\n",
       "                                  LPCAT1 TERT KCNQ1               ADCY9  \\\n",
       "hsa04950                             NaN  NaN   NaN                 NaN   \n",
       "hsa05202                             NaN  NaN   NaN                 NaN   \n",
       "hsa00564  Glycerophospholipid metabolism  NaN   NaN                 NaN   \n",
       "hsa00565          Ether lipid metabolism  NaN   NaN                 NaN   \n",
       "hsa01100              Metabolic pathways  NaN   NaN  Metabolic pathways   \n",
       "...                                  ...  ...   ...                 ...   \n",
       "hsa04115                             NaN  NaN   NaN                 NaN   \n",
       "hsa00561                             NaN  NaN   NaN                 NaN   \n",
       "hsa04979                             NaN  NaN   NaN                 NaN   \n",
       "hsa02010                             NaN  NaN   NaN                 NaN   \n",
       "hsa04975                             NaN  NaN   NaN                 NaN   \n",
       "\n",
       "                        GGT5 OR2AG1 SGCG DOCK1               ACACA SOCS7  \\\n",
       "hsa04950                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa05202                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa00564                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa00565                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa01100  Metabolic pathways    NaN  NaN   NaN  Metabolic pathways   NaN   \n",
       "...                      ...    ...  ...   ...                 ...   ...   \n",
       "hsa04115                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa00561                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa04979                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa02010                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "hsa04975                 NaN    NaN  NaN   NaN                 NaN   NaN   \n",
       "\n",
       "                          KL TCF7L2 BRCA1                  CCNG1  \\\n",
       "hsa04950                 NaN    NaN   NaN                    NaN   \n",
       "hsa05202                 NaN    NaN   NaN                    NaN   \n",
       "hsa00564                 NaN    NaN   NaN                    NaN   \n",
       "hsa00565                 NaN    NaN   NaN                    NaN   \n",
       "hsa01100  Metabolic pathways    NaN   NaN                    NaN   \n",
       "...                      ...    ...   ...                    ...   \n",
       "hsa04115                 NaN    NaN   NaN  p53 signaling pathway   \n",
       "hsa00561                 NaN    NaN   NaN                    NaN   \n",
       "hsa04979                 NaN    NaN   NaN                    NaN   \n",
       "hsa02010                 NaN    NaN   NaN                    NaN   \n",
       "hsa04975                 NaN    NaN   NaN                    NaN   \n",
       "\n",
       "                                         HNF1B                     LIPC  \\\n",
       "hsa04950  Maturity onset diabetes of the young                      NaN   \n",
       "hsa05202                                   NaN                      NaN   \n",
       "hsa00564                                   NaN                      NaN   \n",
       "hsa00565                                   NaN                      NaN   \n",
       "hsa01100                                   NaN       Metabolic pathways   \n",
       "...                                        ...                      ...   \n",
       "hsa04115                                   NaN                      NaN   \n",
       "hsa00561                                   NaN  Glycerolipid metabolism   \n",
       "hsa04979                                   NaN   Cholesterol metabolism   \n",
       "hsa02010                                   NaN                      NaN   \n",
       "hsa04975                                   NaN                      NaN   \n",
       "\n",
       "                                 ABCA1                                  PAX4  \\\n",
       "hsa04950                           NaN  Maturity onset diabetes of the young   \n",
       "hsa05202                           NaN                                   NaN   \n",
       "hsa00564                           NaN                                   NaN   \n",
       "hsa00565                           NaN                                   NaN   \n",
       "hsa01100                           NaN                                   NaN   \n",
       "...                                ...                                   ...   \n",
       "hsa04115                           NaN                                   NaN   \n",
       "hsa00561                           NaN                                   NaN   \n",
       "hsa04979        Cholesterol metabolism                                   NaN   \n",
       "hsa02010              ABC transporters                                   NaN   \n",
       "hsa04975  Fat digestion and absorption                                   NaN   \n",
       "\n",
       "         OR2AG2  \n",
       "hsa04950    NaN  \n",
       "hsa05202    NaN  \n",
       "hsa00564    NaN  \n",
       "hsa00565    NaN  \n",
       "hsa01100    NaN  \n",
       "...         ...  \n",
       "hsa04115    NaN  \n",
       "hsa00561    NaN  \n",
       "hsa04979    NaN  \n",
       "hsa02010    NaN  \n",
       "hsa04975    NaN  \n",
       "\n",
       "[103 rows x 20 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathways_DF = pd.DataFrame(pathway_dict)\n",
    "# #list_pathways1 = pd.DataFrame(list_pathways).T\n",
    "# new_df = list_pathway.stack().reset_index(level=1, drop=True).to_frame()\n",
    "pathways_DF "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Converting the dictionary of pathways to a DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "Pathway_DF = pd.concat({k: pd.Series(v) for k, v in pathway_dict.items()}).reset_index()\n",
    "Pathway_DF = Pathway_DF.rename(columns ={'level_0':'GENE','level_1':'Pathway_ID',0:'Pathway_Name'})\n",
    "#pd.DataFrame(pathway_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Pathway_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>hsa04950</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>hsa05202</td>\n",
       "      <td>Transcriptional misregulation in cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa00564</td>\n",
       "      <td>Glycerophospholipid metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa00565</td>\n",
       "      <td>Ether lipid metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa01100</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>hsa02010</td>\n",
       "      <td>ABC transporters</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>hsa04975</td>\n",
       "      <td>Fat digestion and absorption</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>hsa04979</td>\n",
       "      <td>Cholesterol metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>PAX4</td>\n",
       "      <td>hsa04950</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>OR2AG2</td>\n",
       "      <td>hsa04740</td>\n",
       "      <td>Olfactory transduction</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE Pathway_ID                             Pathway_Name\n",
       "0      HHEX   hsa04950     Maturity onset diabetes of the young\n",
       "1      HHEX   hsa05202  Transcriptional misregulation in cancer\n",
       "2    LPCAT1   hsa00564           Glycerophospholipid metabolism\n",
       "3    LPCAT1   hsa00565                   Ether lipid metabolism\n",
       "4    LPCAT1   hsa01100                       Metabolic pathways\n",
       "..      ...        ...                                      ...\n",
       "127   ABCA1   hsa02010                         ABC transporters\n",
       "128   ABCA1   hsa04975             Fat digestion and absorption\n",
       "129   ABCA1   hsa04979                   Cholesterol metabolism\n",
       "130    PAX4   hsa04950     Maturity onset diabetes of the young\n",
       "131  OR2AG2   hsa04740                   Olfactory transduction\n",
       "\n",
       "[132 rows x 3 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pathway_DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Pathway_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>hsa04950</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>hsa05202</td>\n",
       "      <td>Transcriptional misregulation in cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa00564</td>\n",
       "      <td>Glycerophospholipid metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa00565</td>\n",
       "      <td>Ether lipid metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa01100</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>hsa02010</td>\n",
       "      <td>ABC transporters</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>hsa04975</td>\n",
       "      <td>Fat digestion and absorption</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>hsa04979</td>\n",
       "      <td>Cholesterol metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>PAX4</td>\n",
       "      <td>hsa04950</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>OR2AG2</td>\n",
       "      <td>hsa04740</td>\n",
       "      <td>Olfactory transduction</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>132 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       GENE Pathway_ID                             Pathway_Name\n",
       "0      HHEX   hsa04950     Maturity onset diabetes of the young\n",
       "1      HHEX   hsa05202  Transcriptional misregulation in cancer\n",
       "2    LPCAT1   hsa00564           Glycerophospholipid metabolism\n",
       "3    LPCAT1   hsa00565                   Ether lipid metabolism\n",
       "4    LPCAT1   hsa01100                       Metabolic pathways\n",
       "..      ...        ...                                      ...\n",
       "127   ABCA1   hsa02010                         ABC transporters\n",
       "128   ABCA1   hsa04975             Fat digestion and absorption\n",
       "129   ABCA1   hsa04979                   Cholesterol metabolism\n",
       "130    PAX4   hsa04950     Maturity onset diabetes of the young\n",
       "131  OR2AG2   hsa04740                   Olfactory transduction\n",
       "\n",
       "[132 rows x 3 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Pathway_DF_Duplicate_removed = Pathway_DF.drop_duplicates(keep= 'first').reset_index(drop =True)\n",
    "Pathway_DF_Duplicate_removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>GENE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4488811</td>\n",
       "      <td>SYN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs12116456</td>\n",
       "      <td>ZNF326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs12116456</td>\n",
       "      <td>BARHL2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs9837421</td>\n",
       "      <td>SH3BP5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs6785239</td>\n",
       "      <td>FAM19A4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>SUSD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>SUSD2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>CABIN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>GGT5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>rs4315633</td>\n",
       "      <td>NR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP     GENE\n",
       "0    rs4488811     SYN2\n",
       "1   rs12116456   ZNF326\n",
       "2   rs12116456   BARHL2\n",
       "3    rs9837421   SH3BP5\n",
       "4    rs6785239  FAM19A4\n",
       "..         ...      ...\n",
       "85   rs8141797    SUSD2\n",
       "86   rs8141797    SUSD2\n",
       "87   rs8141797   CABIN1\n",
       "88   rs8141797     GGT5\n",
       "89   rs4315633       NR\n",
       "\n",
       "[90 rows x 2 columns]"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_data_merged_split_na_droped"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Merging the DFs with Genes and SNPs, keeping both occurances "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GENE_Name</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Pathway_Name</th>\n",
       "      <th>SNP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>hsa04950</td>\n",
       "      <td>Maturity onset diabetes of the young</td>\n",
       "      <td>rs7923837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HHEX</td>\n",
       "      <td>hsa05202</td>\n",
       "      <td>Transcriptional misregulation in cancer</td>\n",
       "      <td>rs7923837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa00564</td>\n",
       "      <td>Glycerophospholipid metabolism</td>\n",
       "      <td>rs13167887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa00565</td>\n",
       "      <td>Ether lipid metabolism</td>\n",
       "      <td>rs13167887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>LPCAT1</td>\n",
       "      <td>hsa01100</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "      <td>rs13167887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>GPR179</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs757210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>ARHGAP23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs757210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>BCAS3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11651885</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>CABIN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs8141797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>SUSD2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs8141797</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    GENE_Name Pathway_ID                             Pathway_Name         SNP\n",
       "0        HHEX   hsa04950     Maturity onset diabetes of the young   rs7923837\n",
       "2        HHEX   hsa05202  Transcriptional misregulation in cancer   rs7923837\n",
       "4      LPCAT1   hsa00564           Glycerophospholipid metabolism  rs13167887\n",
       "5      LPCAT1   hsa00565                   Ether lipid metabolism  rs13167887\n",
       "6      LPCAT1   hsa01100                       Metabolic pathways  rs13167887\n",
       "..        ...        ...                                      ...         ...\n",
       "196    GPR179        NaN                                      NaN    rs757210\n",
       "197  ARHGAP23        NaN                                      NaN    rs757210\n",
       "198     BCAS3        NaN                                      NaN  rs11651885\n",
       "199    CABIN1        NaN                                      NaN   rs8141797\n",
       "201     SUSD2        NaN                                      NaN   rs8141797\n",
       "\n",
       "[193 rows x 4 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Genes_rsSNP_Merged = pd.merge(left =Pathway_DF , right = genes_data_merged_split_na_droped,\n",
    "                                   how='outer', left_on='GENE', right_on='GENE').drop_duplicates(keep = 'first')\n",
    "Genes_rsSNP_Merged_renamed =Genes_rsSNP_Merged.rename(columns ={'GENE':'GENE_Name'})\n",
    "Genes_rsSNP_Merged_renamed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Creating Dataframes from the List of dictionaries "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>name</th>\n",
       "      <th>type</th>\n",
       "      <th>link</th>\n",
       "      <th>gene_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>34</td>\n",
       "      <td>hsa:3087</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3087</td>\n",
       "      <td>HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>444</td>\n",
       "      <td>hsa:3087</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3087</td>\n",
       "      <td>HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>131</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>326</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>151</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>634</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>159</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>110</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>140</td>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>116</td>\n",
       "      <td>hsa:3784 hsa:9992</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3784...</td>\n",
       "      <td>KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>71</td>\n",
       "      <td>hsa:3784</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3784</td>\n",
       "      <td>KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>70</td>\n",
       "      <td>hsa:3784</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3784</td>\n",
       "      <td>KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>32</td>\n",
       "      <td>hsa:6445</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6445</td>\n",
       "      <td>SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>33</td>\n",
       "      <td>hsa:6445</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6445</td>\n",
       "      <td>SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>32</td>\n",
       "      <td>hsa:6445</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6445</td>\n",
       "      <td>SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>33</td>\n",
       "      <td>hsa:6445</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6445</td>\n",
       "      <td>SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>77</td>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>89</td>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>129</td>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>228</td>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>988</td>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>161</td>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>428</td>\n",
       "      <td>hsa:31</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>429</td>\n",
       "      <td>hsa:31</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>430</td>\n",
       "      <td>hsa:31</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>118</td>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>18</td>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1450</td>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>305</td>\n",
       "      <td>hsa:31</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>57</td>\n",
       "      <td>hsa:31</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>120</td>\n",
       "      <td>hsa:31</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>30</td>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>168</td>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>115</td>\n",
       "      <td>hsa:9365</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</td>\n",
       "      <td>KL, HFTC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>116</td>\n",
       "      <td>hsa:9365</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</td>\n",
       "      <td>KL, HFTC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>193</td>\n",
       "      <td>hsa:9365</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</td>\n",
       "      <td>KL, HFTC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>114</td>\n",
       "      <td>hsa:9365</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</td>\n",
       "      <td>KL, HFTC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>144</td>\n",
       "      <td>hsa:9365</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</td>\n",
       "      <td>KL, HFTC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>91</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>324</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>39</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>75</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>206</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1426</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>222</td>\n",
       "      <td>hsa:672</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>20</td>\n",
       "      <td>hsa:900 hsa:901</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:900+...</td>\n",
       "      <td>CCNG1, CCNG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>279</td>\n",
       "      <td>hsa:900</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:900</td>\n",
       "      <td>CCNG1, CCNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>319</td>\n",
       "      <td>hsa:900</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:900</td>\n",
       "      <td>CCNG1, CCNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>7</td>\n",
       "      <td>hsa:6928</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6928</td>\n",
       "      <td>HNF1B, FJHN, HNF-1-beta, HNF-1B, HNF1beta, HNF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>243</td>\n",
       "      <td>hsa:3990</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3990</td>\n",
       "      <td>LIPC, HDLCQ12, HL, HTGL, LIPH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>241</td>\n",
       "      <td>hsa:19</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>100</td>\n",
       "      <td>hsa:19</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>205</td>\n",
       "      <td>hsa:19</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>206</td>\n",
       "      <td>hsa:19</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>207</td>\n",
       "      <td>hsa:19</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>22</td>\n",
       "      <td>hsa:5078</td>\n",
       "      <td>gene</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:5078</td>\n",
       "      <td>PAX4, KPD, MODY9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      id               name  type  \\\n",
       "0     34           hsa:3087  gene   \n",
       "1    444           hsa:3087  gene   \n",
       "2    131           hsa:7015  gene   \n",
       "3    326           hsa:7015  gene   \n",
       "4    151           hsa:7015  gene   \n",
       "5    634           hsa:7015  gene   \n",
       "6    159           hsa:7015  gene   \n",
       "7    110           hsa:7015  gene   \n",
       "8    140           hsa:7015  gene   \n",
       "9    116  hsa:3784 hsa:9992  gene   \n",
       "10    71           hsa:3784  gene   \n",
       "11    70           hsa:3784  gene   \n",
       "12    32           hsa:6445  gene   \n",
       "13    33           hsa:6445  gene   \n",
       "14    32           hsa:6445  gene   \n",
       "15    33           hsa:6445  gene   \n",
       "16    77           hsa:1793  gene   \n",
       "17    89           hsa:1793  gene   \n",
       "18   129           hsa:1793  gene   \n",
       "19   228           hsa:1793  gene   \n",
       "20   988           hsa:1793  gene   \n",
       "21    22           hsa:1793  gene   \n",
       "22   161      hsa:31 hsa:32  gene   \n",
       "23   428             hsa:31  gene   \n",
       "24   429             hsa:31  gene   \n",
       "25   430             hsa:31  gene   \n",
       "26   118      hsa:31 hsa:32  gene   \n",
       "27    18      hsa:31 hsa:32  gene   \n",
       "28  1450      hsa:31 hsa:32  gene   \n",
       "29   305             hsa:31  gene   \n",
       "30    57             hsa:31  gene   \n",
       "31   120             hsa:31  gene   \n",
       "32    30      hsa:31 hsa:32  gene   \n",
       "33   168      hsa:31 hsa:32  gene   \n",
       "34   115           hsa:9365  gene   \n",
       "35   116           hsa:9365  gene   \n",
       "36   193           hsa:9365  gene   \n",
       "37   114           hsa:9365  gene   \n",
       "38   144           hsa:9365  gene   \n",
       "39    91            hsa:672  gene   \n",
       "40   324            hsa:672  gene   \n",
       "41    39            hsa:672  gene   \n",
       "42    75            hsa:672  gene   \n",
       "43   206            hsa:672  gene   \n",
       "44  1426            hsa:672  gene   \n",
       "45   222            hsa:672  gene   \n",
       "46    20    hsa:900 hsa:901  gene   \n",
       "47   279            hsa:900  gene   \n",
       "48   319            hsa:900  gene   \n",
       "49     7           hsa:6928  gene   \n",
       "50   243           hsa:3990  gene   \n",
       "51   241             hsa:19  gene   \n",
       "52   100             hsa:19  gene   \n",
       "53   205             hsa:19  gene   \n",
       "54   206             hsa:19  gene   \n",
       "55   207             hsa:19  gene   \n",
       "56    22           hsa:5078  gene   \n",
       "\n",
       "                                                 link  \\\n",
       "0      http://www.kegg.jp/dbget-bin/www_bget?hsa:3087   \n",
       "1      http://www.kegg.jp/dbget-bin/www_bget?hsa:3087   \n",
       "2      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "3      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "4      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "5      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "6      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "7      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "8      http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "9   http://www.kegg.jp/dbget-bin/www_bget?hsa:3784...   \n",
       "10     http://www.kegg.jp/dbget-bin/www_bget?hsa:3784   \n",
       "11     http://www.kegg.jp/dbget-bin/www_bget?hsa:3784   \n",
       "12     http://www.kegg.jp/dbget-bin/www_bget?hsa:6445   \n",
       "13     http://www.kegg.jp/dbget-bin/www_bget?hsa:6445   \n",
       "14     http://www.kegg.jp/dbget-bin/www_bget?hsa:6445   \n",
       "15     http://www.kegg.jp/dbget-bin/www_bget?hsa:6445   \n",
       "16     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "17     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "18     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "19     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "20     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "21     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "22  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "23       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "24       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "25       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "26  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "27  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "28  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "29       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "30       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "31       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "32  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "33  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "34     http://www.kegg.jp/dbget-bin/www_bget?hsa:9365   \n",
       "35     http://www.kegg.jp/dbget-bin/www_bget?hsa:9365   \n",
       "36     http://www.kegg.jp/dbget-bin/www_bget?hsa:9365   \n",
       "37     http://www.kegg.jp/dbget-bin/www_bget?hsa:9365   \n",
       "38     http://www.kegg.jp/dbget-bin/www_bget?hsa:9365   \n",
       "39      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "40      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "41      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "42      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "43      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "44      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "45      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "46  http://www.kegg.jp/dbget-bin/www_bget?hsa:900+...   \n",
       "47      http://www.kegg.jp/dbget-bin/www_bget?hsa:900   \n",
       "48      http://www.kegg.jp/dbget-bin/www_bget?hsa:900   \n",
       "49     http://www.kegg.jp/dbget-bin/www_bget?hsa:6928   \n",
       "50     http://www.kegg.jp/dbget-bin/www_bget?hsa:3990   \n",
       "51       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "52       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "53       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "54       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "55       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "56     http://www.kegg.jp/dbget-bin/www_bget?hsa:5078   \n",
       "\n",
       "                                           gene_names  \n",
       "0              HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX  \n",
       "1              HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX  \n",
       "2   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "3   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "4   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "5   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "6   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "7   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "8   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "9   KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...  \n",
       "10  KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...  \n",
       "11  KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...  \n",
       "12  SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...  \n",
       "13  SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...  \n",
       "14  SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...  \n",
       "15  SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...  \n",
       "16                               DOCK1, DOCK180, ced5  \n",
       "17                               DOCK1, DOCK180, ced5  \n",
       "18                               DOCK1, DOCK180, ced5  \n",
       "19                               DOCK1, DOCK180, ced5  \n",
       "20                               DOCK1, DOCK180, ced5  \n",
       "21                               DOCK1, DOCK180, ced5  \n",
       "22            ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "23               ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "24               ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "25               ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "26            ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "27            ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "28            ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "29               ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "30               ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "31               ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "32            ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "33            ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "34                                          KL, HFTC3  \n",
       "35                                          KL, HFTC3  \n",
       "36                                          KL, HFTC3  \n",
       "37                                          KL, HFTC3  \n",
       "38                                          KL, HFTC3  \n",
       "39  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "40  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "41  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "42  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "43  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "44  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "45  BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "46                                     CCNG1, CCNG...  \n",
       "47                                        CCNG1, CCNG  \n",
       "48                                        CCNG1, CCNG  \n",
       "49  HNF1B, FJHN, HNF-1-beta, HNF-1B, HNF1beta, HNF...  \n",
       "50                      LIPC, HDLCQ12, HL, HTGL, LIPH  \n",
       "51   ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD  \n",
       "52  ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, H...  \n",
       "53   ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD  \n",
       "54   ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD  \n",
       "55   ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD  \n",
       "56                                   PAX4, KPD, MODY9  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#pathways_list = \n",
    "DF_entry = pd.DataFrame(pickle_list_entry_Obj)\n",
    "DF_entry"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Drop Duplicates in the name column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>link</th>\n",
       "      <th>gene_names</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa:3087</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3087</td>\n",
       "      <td>HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa:7015</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</td>\n",
       "      <td>TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa:3784 hsa:9992</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3784...</td>\n",
       "      <td>KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa:3784</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3784</td>\n",
       "      <td>KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa:6445</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6445</td>\n",
       "      <td>SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>hsa:1793</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</td>\n",
       "      <td>DOCK1, DOCK180, ced5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>hsa:31 hsa:32</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>hsa:31</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:31</td>\n",
       "      <td>ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>hsa:9365</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</td>\n",
       "      <td>KL, HFTC3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>hsa:672</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:672</td>\n",
       "      <td>BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>hsa:900 hsa:901</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:900+...</td>\n",
       "      <td>CCNG1, CCNG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>hsa:900</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:900</td>\n",
       "      <td>CCNG1, CCNG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>hsa:6928</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:6928</td>\n",
       "      <td>HNF1B, FJHN, HNF-1-beta, HNF-1B, HNF1beta, HNF...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>hsa:3990</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:3990</td>\n",
       "      <td>LIPC, HDLCQ12, HL, HTGL, LIPH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>hsa:19</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>hsa:19</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:19</td>\n",
       "      <td>ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, H...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>hsa:5078</td>\n",
       "      <td>http://www.kegg.jp/dbget-bin/www_bget?hsa:5078</td>\n",
       "      <td>PAX4, KPD, MODY9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 name                                               link  \\\n",
       "0            hsa:3087     http://www.kegg.jp/dbget-bin/www_bget?hsa:3087   \n",
       "1            hsa:7015     http://www.kegg.jp/dbget-bin/www_bget?hsa:7015   \n",
       "2   hsa:3784 hsa:9992  http://www.kegg.jp/dbget-bin/www_bget?hsa:3784...   \n",
       "3            hsa:3784     http://www.kegg.jp/dbget-bin/www_bget?hsa:3784   \n",
       "4            hsa:6445     http://www.kegg.jp/dbget-bin/www_bget?hsa:6445   \n",
       "5            hsa:1793     http://www.kegg.jp/dbget-bin/www_bget?hsa:1793   \n",
       "6       hsa:31 hsa:32  http://www.kegg.jp/dbget-bin/www_bget?hsa:31+h...   \n",
       "7              hsa:31       http://www.kegg.jp/dbget-bin/www_bget?hsa:31   \n",
       "8            hsa:9365     http://www.kegg.jp/dbget-bin/www_bget?hsa:9365   \n",
       "9             hsa:672      http://www.kegg.jp/dbget-bin/www_bget?hsa:672   \n",
       "10    hsa:900 hsa:901  http://www.kegg.jp/dbget-bin/www_bget?hsa:900+...   \n",
       "11            hsa:900      http://www.kegg.jp/dbget-bin/www_bget?hsa:900   \n",
       "12           hsa:6928     http://www.kegg.jp/dbget-bin/www_bget?hsa:6928   \n",
       "13           hsa:3990     http://www.kegg.jp/dbget-bin/www_bget?hsa:3990   \n",
       "14             hsa:19       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "15             hsa:19       http://www.kegg.jp/dbget-bin/www_bget?hsa:19   \n",
       "16           hsa:5078     http://www.kegg.jp/dbget-bin/www_bget?hsa:5078   \n",
       "\n",
       "                                           gene_names  \n",
       "0              HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX  \n",
       "1   TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1,...  \n",
       "2   KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...  \n",
       "3   KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQ...  \n",
       "4   SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, L...  \n",
       "5                                DOCK1, DOCK180, ced5  \n",
       "6             ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...  \n",
       "7                ACACA, ACAC, ACACAD, ACC, ACC1, ACCA  \n",
       "8                                           KL, HFTC3  \n",
       "9   BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNC...  \n",
       "10                                     CCNG1, CCNG...  \n",
       "11                                        CCNG1, CCNG  \n",
       "12  HNF1B, FJHN, HNF-1-beta, HNF-1B, HNF1beta, HNF...  \n",
       "13                      LIPC, HDLCQ12, HL, HTGL, LIPH  \n",
       "14   ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD  \n",
       "15  ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, H...  \n",
       "16                                   PAX4, KPD, MODY9  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathways_list_Duplicate_removed = DF_entry.drop_duplicates(['gene_names'], \n",
    "                                                                keep= 'first').reset_index(drop =True).drop(['type','id'], axis = 1)\n",
    "\n",
    "pathways_list_Duplicate_removed\n",
    "#pathways_list"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Creating clickable links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style  type=\"text/css\" >\n",
       "</style><table id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7\" ><thead>    <tr>        <th class=\"blank level0\" ></th>        <th class=\"col_heading level0 col0\" >name</th>        <th class=\"col_heading level0 col1\" >link</th>        <th class=\"col_heading level0 col2\" >gene_names</th>    </tr></thead><tbody>\n",
       "                <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row0_col0\" class=\"data row0 col0\" >hsa:3087</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row0_col1\" class=\"data row0 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:3087\">http://www.kegg.jp/dbget-bin/www_bget?hsa:3087</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row0_col2\" class=\"data row0 col2\" >HHEX, HEX, HMPH, HOX11L-PEN, PRH, PRHX</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row1_col0\" class=\"data row1 col0\" >hsa:7015</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row1_col1\" class=\"data row1 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:7015\">http://www.kegg.jp/dbget-bin/www_bget?hsa:7015</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row1_col2\" class=\"data row1 col2\" >TERT, CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1, TP2, TRT, hEST2, hTRT</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row2_col0\" class=\"data row2 col0\" >hsa:3784 hsa:9992</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row2_col1\" class=\"data row2 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:3784+hsa:9992\">http://www.kegg.jp/dbget-bin/www_bget?hsa:3784+hsa:9992</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row2_col2\" class=\"data row2 col2\" >KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQT1, Kv1.9, Kv7.1, LQT, LQT1, RWS, SQT2, WRS...</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row3_col0\" class=\"data row3 col0\" >hsa:3784</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row3_col1\" class=\"data row3 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:3784\">http://www.kegg.jp/dbget-bin/www_bget?hsa:3784</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row3_col2\" class=\"data row3 col2\" >KCNQ1, ATFB1, ATFB3, JLNS1, KCNA8, KCNA9, KVLQT1, Kv1.9, Kv7.1, LQT, LQT1, RWS, SQT2, WRS</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row4\" class=\"row_heading level0 row4\" >4</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row4_col0\" class=\"data row4 col0\" >hsa:6445</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row4_col1\" class=\"data row4 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:6445\">http://www.kegg.jp/dbget-bin/www_bget?hsa:6445</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row4_col2\" class=\"data row4 col2\" >SGCG, 35DAG, A4, DAGA4, DMDA, DMDA1, LGMD2C, LGMDR5, MAM, SCARMD2, SCG3, gamma-SG</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row5\" class=\"row_heading level0 row5\" >5</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row5_col0\" class=\"data row5 col0\" >hsa:1793</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row5_col1\" class=\"data row5 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:1793\">http://www.kegg.jp/dbget-bin/www_bget?hsa:1793</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row5_col2\" class=\"data row5 col2\" >DOCK1, DOCK180, ced5</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row6\" class=\"row_heading level0 row6\" >6</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row6_col0\" class=\"data row6 col0\" >hsa:31 hsa:32</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row6_col1\" class=\"data row6 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:31+hsa:32\">http://www.kegg.jp/dbget-bin/www_bget?hsa:31+hsa:32</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row6_col2\" class=\"data row6 col2\" >ACACA, ACAC, ACACAD, ACC, ACC1, ACCA...</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row7\" class=\"row_heading level0 row7\" >7</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row7_col0\" class=\"data row7 col0\" >hsa:31</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row7_col1\" class=\"data row7 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:31\">http://www.kegg.jp/dbget-bin/www_bget?hsa:31</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row7_col2\" class=\"data row7 col2\" >ACACA, ACAC, ACACAD, ACC, ACC1, ACCA</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row8\" class=\"row_heading level0 row8\" >8</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row8_col0\" class=\"data row8 col0\" >hsa:9365</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row8_col1\" class=\"data row8 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:9365\">http://www.kegg.jp/dbget-bin/www_bget?hsa:9365</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row8_col2\" class=\"data row8 col2\" >KL, HFTC3</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row9\" class=\"row_heading level0 row9\" >9</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row9_col0\" class=\"data row9 col0\" >hsa:672</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row9_col1\" class=\"data row9 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:672\">http://www.kegg.jp/dbget-bin/www_bget?hsa:672</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row9_col2\" class=\"data row9 col2\" >BRCA1, BRCAI, BRCC1, BROVCA1, FANCS, IRIS, PNCA4, PPP1R53, PSCP, RNF53</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row10\" class=\"row_heading level0 row10\" >10</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row10_col0\" class=\"data row10 col0\" >hsa:900 hsa:901</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row10_col1\" class=\"data row10 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:900+hsa:901\">http://www.kegg.jp/dbget-bin/www_bget?hsa:900+hsa:901</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row10_col2\" class=\"data row10 col2\" >CCNG1, CCNG...</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row11\" class=\"row_heading level0 row11\" >11</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row11_col0\" class=\"data row11 col0\" >hsa:900</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row11_col1\" class=\"data row11 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:900\">http://www.kegg.jp/dbget-bin/www_bget?hsa:900</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row11_col2\" class=\"data row11 col2\" >CCNG1, CCNG</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row12\" class=\"row_heading level0 row12\" >12</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row12_col0\" class=\"data row12 col0\" >hsa:6928</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row12_col1\" class=\"data row12 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:6928\">http://www.kegg.jp/dbget-bin/www_bget?hsa:6928</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row12_col2\" class=\"data row12 col2\" >HNF1B, FJHN, HNF-1-beta, HNF-1B, HNF1beta, HNF2, HPC11, LF-B3, LFB3, MODY5, TCF-2, TCF2, VHNF1</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row13\" class=\"row_heading level0 row13\" >13</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row13_col0\" class=\"data row13 col0\" >hsa:3990</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row13_col1\" class=\"data row13 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:3990\">http://www.kegg.jp/dbget-bin/www_bget?hsa:3990</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row13_col2\" class=\"data row13 col2\" >LIPC, HDLCQ12, HL, HTGL, LIPH</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row14\" class=\"row_heading level0 row14\" >14</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row14_col0\" class=\"data row14 col0\" >hsa:19</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row14_col1\" class=\"data row14 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:19\">http://www.kegg.jp/dbget-bin/www_bget?hsa:19</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row14_col2\" class=\"data row14 col2\" >ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, TGD</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row15\" class=\"row_heading level0 row15\" >15</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row15_col0\" class=\"data row15 col0\" >hsa:19</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row15_col1\" class=\"data row15 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:19\">http://www.kegg.jp/dbget-bin/www_bget?hsa:19</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row15_col2\" class=\"data row15 col2\" >ABCA1, ABC-1, ABC1, CERP, HDLCQTL13, HDLDT1, HPALP1, TGD</td>\n",
       "            </tr>\n",
       "            <tr>\n",
       "                        <th id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7level0_row16\" class=\"row_heading level0 row16\" >16</th>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row16_col0\" class=\"data row16 col0\" >hsa:5078</td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row16_col1\" class=\"data row16 col1\" ><a href=\"http://www.kegg.jp/dbget-bin/www_bget?hsa:5078\">http://www.kegg.jp/dbget-bin/www_bget?hsa:5078</a></td>\n",
       "                        <td id=\"T_3fe2ad6e_d7c5_11ea_b943_e470b8fbbbe7row16_col2\" class=\"data row16 col2\" >PAX4, KPD, MODY9</td>\n",
       "            </tr>\n",
       "    </tbody></table>"
      ],
      "text/plain": [
       "<pandas.io.formats.style.Styler at 0x26591b7bb88>"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def make_Link_clickable(val):\n",
    "    return '<a href=\"{}\">{}</a>'.format(val, val)\n",
    "\n",
    "pathways_list_Duplicate_removed.style.format({'link': make_Link_clickable})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Final DataFrame for drawing Manhattan plot "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>...</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>GENE</th>\n",
       "      <th>GENE_Name</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Pathway_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4970420</td>\n",
       "      <td>3640</td>\n",
       "      <td>1790</td>\n",
       "      <td>212</td>\n",
       "      <td>4</td>\n",
       "      <td>12501</td>\n",
       "      <td>6165</td>\n",
       "      <td>732</td>\n",
       "      <td>22</td>\n",
       "      <td>0.917874</td>\n",
       "      <td>...</td>\n",
       "      <td>0.196644</td>\n",
       "      <td>0.196546</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>1096336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs13303016</td>\n",
       "      <td>3472</td>\n",
       "      <td>1925</td>\n",
       "      <td>247</td>\n",
       "      <td>2</td>\n",
       "      <td>11988</td>\n",
       "      <td>6577</td>\n",
       "      <td>850</td>\n",
       "      <td>5</td>\n",
       "      <td>0.793916</td>\n",
       "      <td>...</td>\n",
       "      <td>0.213160</td>\n",
       "      <td>0.213416</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1936451</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2254669</td>\n",
       "      <td>26</td>\n",
       "      <td>664</td>\n",
       "      <td>4956</td>\n",
       "      <td>0</td>\n",
       "      <td>76</td>\n",
       "      <td>2403</td>\n",
       "      <td>16941</td>\n",
       "      <td>0</td>\n",
       "      <td>0.367653</td>\n",
       "      <td>...</td>\n",
       "      <td>0.065783</td>\n",
       "      <td>0.065248</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1</td>\n",
       "      <td>1995968</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs424079</td>\n",
       "      <td>3536</td>\n",
       "      <td>1861</td>\n",
       "      <td>245</td>\n",
       "      <td>4</td>\n",
       "      <td>11987</td>\n",
       "      <td>6479</td>\n",
       "      <td>927</td>\n",
       "      <td>27</td>\n",
       "      <td>0.139649</td>\n",
       "      <td>...</td>\n",
       "      <td>0.214846</td>\n",
       "      <td>0.213381</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>2061200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs7512482</td>\n",
       "      <td>4637</td>\n",
       "      <td>951</td>\n",
       "      <td>58</td>\n",
       "      <td>0</td>\n",
       "      <td>15912</td>\n",
       "      <td>3322</td>\n",
       "      <td>185</td>\n",
       "      <td>1</td>\n",
       "      <td>0.856273</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095061</td>\n",
       "      <td>0.094933</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>1</td>\n",
       "      <td>2136826</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48145</th>\n",
       "      <td>rs11703226</td>\n",
       "      <td>2494</td>\n",
       "      <td>2543</td>\n",
       "      <td>603</td>\n",
       "      <td>6</td>\n",
       "      <td>8787</td>\n",
       "      <td>8502</td>\n",
       "      <td>2098</td>\n",
       "      <td>33</td>\n",
       "      <td>0.332329</td>\n",
       "      <td>...</td>\n",
       "      <td>0.327487</td>\n",
       "      <td>0.328585</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49000551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48146</th>\n",
       "      <td>rs5771242</td>\n",
       "      <td>208</td>\n",
       "      <td>1644</td>\n",
       "      <td>3793</td>\n",
       "      <td>1</td>\n",
       "      <td>632</td>\n",
       "      <td>5913</td>\n",
       "      <td>12869</td>\n",
       "      <td>6</td>\n",
       "      <td>0.565587</td>\n",
       "      <td>...</td>\n",
       "      <td>0.184841</td>\n",
       "      <td>0.184305</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>49003996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48147</th>\n",
       "      <td>rs131816</td>\n",
       "      <td>5172</td>\n",
       "      <td>461</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>17706</td>\n",
       "      <td>1657</td>\n",
       "      <td>52</td>\n",
       "      <td>5</td>\n",
       "      <td>0.318233</td>\n",
       "      <td>...</td>\n",
       "      <td>0.045352</td>\n",
       "      <td>0.044851</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49304328</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48148</th>\n",
       "      <td>rs470119</td>\n",
       "      <td>362</td>\n",
       "      <td>2173</td>\n",
       "      <td>3111</td>\n",
       "      <td>0</td>\n",
       "      <td>1359</td>\n",
       "      <td>7428</td>\n",
       "      <td>10632</td>\n",
       "      <td>1</td>\n",
       "      <td>0.319039</td>\n",
       "      <td>...</td>\n",
       "      <td>0.261239</td>\n",
       "      <td>0.260184</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>22</td>\n",
       "      <td>49313780</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48149</th>\n",
       "      <td>rs9616802</td>\n",
       "      <td>1559</td>\n",
       "      <td>2792</td>\n",
       "      <td>1293</td>\n",
       "      <td>2</td>\n",
       "      <td>5279</td>\n",
       "      <td>9570</td>\n",
       "      <td>4570</td>\n",
       "      <td>1</td>\n",
       "      <td>0.323190</td>\n",
       "      <td>...</td>\n",
       "      <td>0.481745</td>\n",
       "      <td>0.480549</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>49419027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48150 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0       rs4970420    3640    1790     212       4   12501    6165     732   \n",
       "1      rs13303016    3472    1925     247       2   11988    6577     850   \n",
       "2       rs2254669      26     664    4956       0      76    2403   16941   \n",
       "3        rs424079    3536    1861     245       4   11987    6479     927   \n",
       "4       rs7512482    4637     951      58       0   15912    3322     185   \n",
       "...           ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "48145  rs11703226    2494    2543     603       6    8787    8502    2098   \n",
       "48146   rs5771242     208    1644    3793       1     632    5913   12869   \n",
       "48147    rs131816    5172     461      13       0   17706    1657      52   \n",
       "48148    rs470119     362    2173    3111       0    1359    7428   10632   \n",
       "48149   rs9616802    1559    2792    1293       2    5279    9570    4570   \n",
       "\n",
       "       ctrl??         P  ...  maf_ctrl  annotation  alleleA  alleleB  CHR  \\\n",
       "0          22  0.917874  ...  0.196644    0.196546        T        C    1   \n",
       "1           5  0.793916  ...  0.213160    0.213416        A        G    1   \n",
       "2           0  0.367653  ...  0.065783    0.065248        A        G    1   \n",
       "3          27  0.139649  ...  0.214846    0.213381        A        C    1   \n",
       "4           1  0.856273  ...  0.095061    0.094933        T        C    1   \n",
       "...       ...       ...  ...       ...         ...      ...      ...  ...   \n",
       "48145      33  0.332329  ...  0.327487    0.328585        T        C   22   \n",
       "48146       6  0.565587  ...  0.184841    0.184305        A        G   22   \n",
       "48147       5  0.318233  ...  0.045352    0.044851        T        C   22   \n",
       "48148       1  0.319039  ...  0.261239    0.260184        T        C   22   \n",
       "48149       1  0.323190  ...  0.481745    0.480549        A        G   22   \n",
       "\n",
       "             BP GENE GENE_Name  Pathway_ID  Pathway_Name  \n",
       "0       1096336  NaN       NaN         NaN           NaN  \n",
       "1       1936451  NaN       NaN         NaN           NaN  \n",
       "2       1995968  NaN       NaN         NaN           NaN  \n",
       "3       2061200  NaN       NaN         NaN           NaN  \n",
       "4       2136826  NaN       NaN         NaN           NaN  \n",
       "...         ...  ...       ...         ...           ...  \n",
       "48145  49000551  NaN       NaN         NaN           NaN  \n",
       "48146  49003996  NaN       NaN         NaN           NaN  \n",
       "48147  49304328  NaN       NaN         NaN           NaN  \n",
       "48148  49313780  NaN       NaN         NaN           NaN  \n",
       "48149  49419027  NaN       NaN         NaN           NaN  \n",
       "\n",
       "[48150 rows x 24 columns]"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Final_DataFrame = pd.merge(left =SampleData_Merged_Genes, right = Genes_rsSNP_Merged_renamed,\n",
    "                                   how='left', left_on='SNP', right_on='SNP')\n",
    "Final_DataFrame"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>case11</th>\n",
       "      <th>case12</th>\n",
       "      <th>case22</th>\n",
       "      <th>case??</th>\n",
       "      <th>ctrl11</th>\n",
       "      <th>ctrl12</th>\n",
       "      <th>ctrl22</th>\n",
       "      <th>ctrl??</th>\n",
       "      <th>P</th>\n",
       "      <th>...</th>\n",
       "      <th>maf_ctrl</th>\n",
       "      <th>annotation</th>\n",
       "      <th>alleleA</th>\n",
       "      <th>alleleB</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>GENE</th>\n",
       "      <th>GENE_Name</th>\n",
       "      <th>Pathway_ID</th>\n",
       "      <th>Pathway_Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>68</td>\n",
       "      <td>1123</td>\n",
       "      <td>4455</td>\n",
       "      <td>0</td>\n",
       "      <td>197</td>\n",
       "      <td>3574</td>\n",
       "      <td>15649</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102163</td>\n",
       "      <td>0.104265</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>1361552</td>\n",
       "      <td>TERT,CLPTM1L</td>\n",
       "      <td>TERT</td>\n",
       "      <td>hsa05165</td>\n",
       "      <td>Human papillomavirus infection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>68</td>\n",
       "      <td>1123</td>\n",
       "      <td>4455</td>\n",
       "      <td>0</td>\n",
       "      <td>197</td>\n",
       "      <td>3574</td>\n",
       "      <td>15649</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102163</td>\n",
       "      <td>0.104265</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>1361552</td>\n",
       "      <td>TERT,CLPTM1L</td>\n",
       "      <td>TERT</td>\n",
       "      <td>hsa05166</td>\n",
       "      <td>Human T-cell leukemia virus 1 infection</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>68</td>\n",
       "      <td>1123</td>\n",
       "      <td>4455</td>\n",
       "      <td>0</td>\n",
       "      <td>197</td>\n",
       "      <td>3574</td>\n",
       "      <td>15649</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102163</td>\n",
       "      <td>0.104265</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>1361552</td>\n",
       "      <td>TERT,CLPTM1L</td>\n",
       "      <td>TERT</td>\n",
       "      <td>hsa05200</td>\n",
       "      <td>Pathways in cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>68</td>\n",
       "      <td>1123</td>\n",
       "      <td>4455</td>\n",
       "      <td>0</td>\n",
       "      <td>197</td>\n",
       "      <td>3574</td>\n",
       "      <td>15649</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102163</td>\n",
       "      <td>0.104265</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>1361552</td>\n",
       "      <td>TERT,CLPTM1L</td>\n",
       "      <td>TERT</td>\n",
       "      <td>hsa05225</td>\n",
       "      <td>Hepatocellular carcinoma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs4635969</td>\n",
       "      <td>68</td>\n",
       "      <td>1123</td>\n",
       "      <td>4455</td>\n",
       "      <td>0</td>\n",
       "      <td>197</td>\n",
       "      <td>3574</td>\n",
       "      <td>15649</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004226</td>\n",
       "      <td>...</td>\n",
       "      <td>0.102163</td>\n",
       "      <td>0.104265</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>5</td>\n",
       "      <td>1361552</td>\n",
       "      <td>TERT,CLPTM1L</td>\n",
       "      <td>TERT</td>\n",
       "      <td>hsa05226</td>\n",
       "      <td>Gastric cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>4707</td>\n",
       "      <td>903</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>15896</td>\n",
       "      <td>3327</td>\n",
       "      <td>187</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002310</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095338</td>\n",
       "      <td>0.093202</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>22912041</td>\n",
       "      <td>CABIN1,SUSD2</td>\n",
       "      <td>GGT5</td>\n",
       "      <td>hsa01100</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>214</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>4707</td>\n",
       "      <td>903</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>15896</td>\n",
       "      <td>3327</td>\n",
       "      <td>187</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002310</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095338</td>\n",
       "      <td>0.093202</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>22912041</td>\n",
       "      <td>SUSD2,CABIN1,GGT5</td>\n",
       "      <td>GGT5</td>\n",
       "      <td>hsa00430</td>\n",
       "      <td>Taurine and hypotaurine metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>215</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>4707</td>\n",
       "      <td>903</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>15896</td>\n",
       "      <td>3327</td>\n",
       "      <td>187</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002310</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095338</td>\n",
       "      <td>0.093202</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>22912041</td>\n",
       "      <td>SUSD2,CABIN1,GGT5</td>\n",
       "      <td>GGT5</td>\n",
       "      <td>hsa00480</td>\n",
       "      <td>Glutathione metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>4707</td>\n",
       "      <td>903</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>15896</td>\n",
       "      <td>3327</td>\n",
       "      <td>187</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002310</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095338</td>\n",
       "      <td>0.093202</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>22912041</td>\n",
       "      <td>SUSD2,CABIN1,GGT5</td>\n",
       "      <td>GGT5</td>\n",
       "      <td>hsa00590</td>\n",
       "      <td>Arachidonic acid metabolism</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>rs8141797</td>\n",
       "      <td>4707</td>\n",
       "      <td>903</td>\n",
       "      <td>33</td>\n",
       "      <td>3</td>\n",
       "      <td>15896</td>\n",
       "      <td>3327</td>\n",
       "      <td>187</td>\n",
       "      <td>10</td>\n",
       "      <td>0.002310</td>\n",
       "      <td>...</td>\n",
       "      <td>0.095338</td>\n",
       "      <td>0.093202</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>22</td>\n",
       "      <td>22912041</td>\n",
       "      <td>SUSD2,CABIN1,GGT5</td>\n",
       "      <td>GGT5</td>\n",
       "      <td>hsa01100</td>\n",
       "      <td>Metabolic pathways</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>218 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           SNP  case11  case12  case22  case??  ctrl11  ctrl12  ctrl22  \\\n",
       "0    rs4635969      68    1123    4455       0     197    3574   15649   \n",
       "1    rs4635969      68    1123    4455       0     197    3574   15649   \n",
       "2    rs4635969      68    1123    4455       0     197    3574   15649   \n",
       "3    rs4635969      68    1123    4455       0     197    3574   15649   \n",
       "4    rs4635969      68    1123    4455       0     197    3574   15649   \n",
       "..         ...     ...     ...     ...     ...     ...     ...     ...   \n",
       "213  rs8141797    4707     903      33       3   15896    3327     187   \n",
       "214  rs8141797    4707     903      33       3   15896    3327     187   \n",
       "215  rs8141797    4707     903      33       3   15896    3327     187   \n",
       "216  rs8141797    4707     903      33       3   15896    3327     187   \n",
       "217  rs8141797    4707     903      33       3   15896    3327     187   \n",
       "\n",
       "     ctrl??         P  ...  maf_ctrl  annotation  alleleA  alleleB  CHR  \\\n",
       "0         0  0.004226  ...  0.102163    0.104265        T        C    5   \n",
       "1         0  0.004226  ...  0.102163    0.104265        T        C    5   \n",
       "2         0  0.004226  ...  0.102163    0.104265        T        C    5   \n",
       "3         0  0.004226  ...  0.102163    0.104265        T        C    5   \n",
       "4         0  0.004226  ...  0.102163    0.104265        T        C    5   \n",
       "..      ...       ...  ...       ...         ...      ...      ...  ...   \n",
       "213      10  0.002310  ...  0.095338    0.093202        A        G   22   \n",
       "214      10  0.002310  ...  0.095338    0.093202        A        G   22   \n",
       "215      10  0.002310  ...  0.095338    0.093202        A        G   22   \n",
       "216      10  0.002310  ...  0.095338    0.093202        A        G   22   \n",
       "217      10  0.002310  ...  0.095338    0.093202        A        G   22   \n",
       "\n",
       "           BP               GENE GENE_Name  Pathway_ID  \\\n",
       "0     1361552       TERT,CLPTM1L      TERT    hsa05165   \n",
       "1     1361552       TERT,CLPTM1L      TERT    hsa05166   \n",
       "2     1361552       TERT,CLPTM1L      TERT    hsa05200   \n",
       "3     1361552       TERT,CLPTM1L      TERT    hsa05225   \n",
       "4     1361552       TERT,CLPTM1L      TERT    hsa05226   \n",
       "..        ...                ...       ...         ...   \n",
       "213  22912041       CABIN1,SUSD2      GGT5    hsa01100   \n",
       "214  22912041  SUSD2,CABIN1,GGT5      GGT5    hsa00430   \n",
       "215  22912041  SUSD2,CABIN1,GGT5      GGT5    hsa00480   \n",
       "216  22912041  SUSD2,CABIN1,GGT5      GGT5    hsa00590   \n",
       "217  22912041  SUSD2,CABIN1,GGT5      GGT5    hsa01100   \n",
       "\n",
       "                                Pathway_Name  \n",
       "0             Human papillomavirus infection  \n",
       "1    Human T-cell leukemia virus 1 infection  \n",
       "2                         Pathways in cancer  \n",
       "3                   Hepatocellular carcinoma  \n",
       "4                             Gastric cancer  \n",
       "..                                       ...  \n",
       "213                       Metabolic pathways  \n",
       "214       Taurine and hypotaurine metabolism  \n",
       "215                   Glutathione metabolism  \n",
       "216              Arachidonic acid metabolism  \n",
       "217                       Metabolic pathways  \n",
       "\n",
       "[218 rows x 24 columns]"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kk2 = Final_DataFrame.dropna().reset_index(drop = True)\n",
    "kk2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Final_DataFrame.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#fig4 = ManhattanPlot(Final_DataFrame, title='Manhattan plot', annotation = 'Pathway_Name', gene= 'GENE')\n",
    "#fig4.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Creating Sliders in Dash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!pip install jupyter-dash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "app = JupyterDash(__name__)\n",
    "#n_chr = 23  # number of chromosome pairs in humans\n",
    "assert 'CHR' in Final_DataFrame.columns\n",
    "#assert Final_DataFrame['CHR'].max() == n_chr\n",
    "\n",
    "# # Trim down the data\n",
    "# DATASET = df.groupby('CHR').apply(lambda u: u.head(50))\n",
    "# DATASET = DATASET.droplevel('CHR').reset_index(drop=True)\n",
    "\n",
    "manhattanplot = ManhattanPlot(\n",
    "    dataframe=Final_DataFrame,\n",
    "    suggestiveline_color='#AA00AA',\n",
    "    genomewideline_color='#AA5500'\n",
    ")\n",
    "app.layout = html.Div([\n",
    "    ' - log10(P-Value Threshold)',\n",
    "    dcc.Slider(\n",
    "        id='manhattanplot-input',\n",
    "        min=1,\n",
    "        max=10,\n",
    "        marks={\n",
    "            i: 'P-Value{}'.format(i) for i in range(10)\n",
    "        },\n",
    "        value=6\n",
    "    ),\n",
    "     html.Br(),\n",
    "    html.Div(\n",
    "        dcc.Graph(\n",
    "            id='my-dashbio-manhattanplot',\n",
    "            figure=ManhattanPlot(\n",
    "                dataframe=Final_DataFrame\n",
    "            )\n",
    "        )\n",
    "    )\n",
    "])\n",
    "\n",
    "@app.callback(\n",
    "    dash.dependencies.Output('my-dashbio-manhattanplot', 'figure'),\n",
    "    [dash.dependencies.Input('manhattanplot-input', 'value')]\n",
    ")\n",
    "def update_manhattanplot(threshold):\n",
    "\n",
    "    return ManhattanPlot(\n",
    "        dataframe=Final_DataFrame,\n",
    "        genomewideline_value=threshold\n",
    "    )\n",
    "\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    app.run_server(mode = 'inline')  #Turn off reloader if inside Jupyter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Load Data\n",
    "# df = px.data.tips()\n",
    "# # Build App\n",
    "# app = JupyterDash(__name__)\n",
    "# app.layout = html.Div([\n",
    "#     html.H1(\"JupyterDash Demo\"),\n",
    "#     dcc.Graph(id='graph'),\n",
    "#     html.Label([\n",
    "#         \"colorscale\",\n",
    "#         dcc.Dropdown(\n",
    "#             id='colorscale-dropdown', clearable=False,\n",
    "#             value='plasma', options=[\n",
    "#                 {'label': c, 'value': c}\n",
    "#                 for c in px.colors.named_colorscales()\n",
    "#             ])\n",
    "#     ]),\n",
    "# ])\n",
    "# # Define callback to update graph\n",
    "# @app.callback(\n",
    "#     Output('graph', 'figure'),\n",
    "#     [Input(\"colorscale-dropdown\", \"value\")]\n",
    "# )\n",
    "# def update_figure(colorscale):\n",
    "#     return px.scatter(\n",
    "#         df, x=\"total_bill\", y=\"tip\", color=\"size\",\n",
    "#         color_continuous_scale=colorscale,\n",
    "#         render_mode=\"webgl\", title=\"Tips\"\n",
    "#     )\n",
    "# # Run app and display result inline in the notebook\n",
    "# app.run_server(mode='inline')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('GWAS_rename_DF.pickle', 'rb') as GWAS_rename_DF_Obj:\n",
    "\n",
    "    GWAS_rename_DF_Ob = pickle.load(GWAS_rename_DF_Obj)\n",
    "\n",
    "    GWAS_rename_DF_Obj.close()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#GWAS_rename_DF_Ob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get the snps as a list\n",
    "snps_lists = GWAS_rename_DF_Ob['SNP']\n",
    "\n",
    "# remove duplicates if any\n",
    "snps_lists_duplicate_removed = list(set(snps_lists))\n",
    "#pickling the Snps for use in biomartRt\n",
    "with open('snps_list.pickle', 'wb') as fsnps_list_Obj: # Creating pickle file object\n",
    "    pickle.dump(snps_lists_duplicate_removed,fsnps_list_Obj) # Dumping the object into the pickle file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# grouped_dataFrame2 = pd.DataFrame()\n",
    "\n",
    "# for num, (name, group) in enumerate(DATASET_trimgroup):\n",
    "#     #px.scatter(group, x='ind', y='p_adj', color=\"Chr\")\n",
    "#     grouped_dataFrame2 = grouped_dataFrame2.append(group)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## .Second Datasets just in Case"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_DF4 = pd.read_csv('GWASData/GWAS/DIAGRAMv3.2012DEC17.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_DF4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_DF4.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_DF4_sorted_DF_Not_rs = gwas_DF4[gwas_DF4['SNP'].str.startswith('rs')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_DF4_sorted_DF_Not_rs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_DF4_sorted_DF_Not_rs.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATASET_trim4 = gwas_DF4_sorted_DF_Not_rs.groupby(['CHROMOSOME', 'POSITION']).apply(lambda u: u.head(1500))\n",
    "DATASET_trim4 = DATASET_trim4.droplevel(['CHROMOSOME','POSITION']).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATASET_trim4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATASET_trim4 = DATASET_trim4.rename(columns={\n",
    "                        \"CHROMOSOME\": \"CHR\",\n",
    "                        \"POSITION\":\"BP\",\n",
    "                       \"P_VALUE\":\"P\"\n",
    "                        \n",
    "                       }, errors=\"raise\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Groupby Testings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [],
   "source": [
    "aList = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10,\n",
    "'one', 'two', 'three', 'four', 'five',\n",
    "'six', 'seven', 'eight', 'nine', 'ten']\n",
    "\n",
    "# Open the file for writing\n",
    "dataFile = open('writetest.txt', 'w')\n",
    "\n",
    "# Loop through each item in the list\n",
    "# and write it to the output file.\n",
    "for eachitem in aList:\n",
    "    dataFile.write(str(eachitem)+'\\n')\n",
    "\n",
    "# Close the output file\n",
    "dataFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [],
   "source": [
    "kf = pd.read_csv('writetest.txt')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Coverting list of dicts to dictionary "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import ChainMap\n",
    "#data = dict(ChainMap(*data))\n",
    "dicts = dict(ChainMap(*list_pathways))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import plotly.graph_objs as go\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "layout1= go.Layout(title=go.layout.Title(text=\"A graph\",x=0.5),\n",
    "        xaxis={'title':'x[m]'},\n",
    "        yaxis={'title':'y[m]','range':[-10,10]})\n",
    "\n",
    "point_plot=[\n",
    "            go.Scatter(x=[1,2,3,3,4],y=[1,1,np.nan, 1,2],name=\"V0\"),\n",
    "            go.Scatter(x=[5,6],y=[2,3],name=\"GT\")\n",
    "    ]\n",
    "      \n",
    "      \n",
    "      \n",
    "go.Figure(data=point_plot, layout=layout1).show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "lake = pd.DataFrame({'co tp': ['DE Lake', 'Forest', 'FR Lake', 'Forest'], \n",
    "                 'area': [10, 20, 30, 40], \n",
    "                 'count': [7, 5, 2, 3]})\n",
    "lake.set_index('co tp', inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>area</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>co tp</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DE Lake</th>\n",
       "      <td>10</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Forest</th>\n",
       "      <td>20</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FR Lake</th>\n",
       "      <td>30</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Forest</th>\n",
       "      <td>40</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         area  count\n",
       "co tp               \n",
       "DE Lake    10      7\n",
       "Forest     20      5\n",
       "FR Lake    30      2\n",
       "Forest     40      3"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lake"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{10: 7, 20: 5, 30: 2, 40: 3}\n"
     ]
    }
   ],
   "source": [
    "# to get key value using pandas\n",
    "area_dict = lake.set_index('area').to_dict()['count']#('records')[0]\n",
    "print(area_dict)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{10: 7, 20: 5, 30: 2, 40: 3}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "area_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(np.random.randint(32, 120, 100000).reshape(50000,2),columns=list('AB'))\n",
    "df['A'] = df['A'].apply(chr)\n",
    "\n",
    "# %timeit dict(zip(df.A,df.B))\n",
    "# %timeit pd.Series(df.A.values,index=df.B).to_dict()\n",
    "# %timeit df.set_index('A').to_dict()['B']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>B</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>B</td>\n",
       "      <td>75</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>U</td>\n",
       "      <td>112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>`</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>s</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   A    B\n",
       "0  B   75\n",
       "1  U  112\n",
       "2  `  107\n",
       "3  P   50\n",
       "4  s   46"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "k = dict(zip(df.A,df.B))\n",
    "\n",
    "k1= pd.Series(df.A.values,index=df.B).to_dict()\n",
    "\n",
    "k2= df.set_index('A').to_dict()['B']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'B': 118, 'U': 56, '`': 68, 'P': 113, 's': 39, 'A': 117, 'W': 60, '^': 96, '1': 57, '5': 75, 'j': 98, 'f': 110, '(': 51, '*': 119, 'G': 37, '+': 38, 'L': 61, 't': 70, '?': 94, 'b': 90, 'S': 43, '.': 113, 'r': 54, 'd': 36, '4': 33, 'l': 78, '<': 73, 'i': 104, '%': 74, 'n': 53, 'J': 99, '$': 119, 'D': 84, 'k': 44, 'K': 83, '=': 64, 'h': 35, '!': 105, 'a': 54, 'N': 107, 'p': 89, 'E': 76, 'C': 69, \"'\": 82, 'v': 45, '\\\\': 68, 'M': 32, '[': 74, '6': 71, ' ': 47, 'T': 45, '_': 44, 'X': 60, '-': 69, 'u': 94, '8': 72, '\"': 51, 'o': 96, ';': 114, 'V': 114, 'R': 32, '2': 47, '0': 49, 'g': 47, 'e': 92, '&': 115, 'H': 54, 'Z': 113, '/': 58, '#': 62, '@': 40, ':': 113, 'O': 86, 'm': 47, 'Q': 115, ']': 48, '9': 70, ')': 111, '>': 37, 'F': 67, 'I': 85, ',': 101, '3': 119, '7': 43, 'Y': 61, 'q': 71, 'c': 110, 'w': 102}\n"
     ]
    }
   ],
   "source": [
    "print(k2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "22\n",
      "24\n",
      "46\n",
      "70\n",
      "116\n",
      "186\n"
     ]
    }
   ],
   "source": [
    "a, b = 20, 2\n",
    "while a < 150:\n",
    "    \n",
    "    a, b = b, a+b\n",
    "    print(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "range(0, 5)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "range(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n"
     ]
    }
   ],
   "source": [
    "for i in range(5):\n",
    "    print (i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample data\n",
    "d={'A':[3,3,2,1,5],\n",
    "   'B':[4,4,1,4,7],\n",
    "   'A_info':['nothing', '', '', 'bad', 'good'],\n",
    "   'B_info':['', '', 'bad', 'better', 'best']}\n",
    "\n",
    "# pandas dataframe\n",
    "df=pd.DataFrame(d, index=[10,11,12,13,14])\n",
    "\n",
    "# set up plotly figure\n",
    "fig = go.Figure()\n",
    "\n",
    "# add line / trace 1 to figure\n",
    "fig.add_trace(go.Scatter(\n",
    "    x=df.index,\n",
    "    y=df['A'],\n",
    "    hovertext=df['A_info'],\n",
    "    hoverinfo=\"text\",\n",
    "    marker=dict(\n",
    "        color=\"blue\"\n",
    "    ),\n",
    "    showlegend=False\n",
    "))\n",
    "\n",
    "# add line / trace 2 to figure\n",
    "fig.add_trace(go.Scatter(\n",
    "    x=df.index,\n",
    "    y=df['B'],\n",
    "    hovertext=df['B_info'],\n",
    "    hoverinfo=\"text\",\n",
    "    marker=dict(\n",
    "        color=\"green\"\n",
    "    ),\n",
    "    showlegend=False\n",
    "))\n",
    "\n",
    "fig.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A</th>\n",
       "      <th>B</th>\n",
       "      <th>A_info</th>\n",
       "      <th>B_info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>nothing</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td></td>\n",
       "      <td>bad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>bad</td>\n",
       "      <td>better</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>good</td>\n",
       "      <td>best</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    A  B   A_info  B_info\n",
       "10  3  4  nothing        \n",
       "11  3  4                 \n",
       "12  2  1              bad\n",
       "13  1  4      bad  better\n",
       "14  5  7     good    best"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
